
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://github.com/alliance-genome/agr_curation_schema/DiseaseAnnotation/" rel="canonical"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.1, mkdocs-material-9.6.14" name="generator"/>
<title>DiseaseAnnotation - Alliance Persistent Store</title>
<link href="../assets/stylesheets/main.342714a4.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#diseaseannotation">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Alliance Persistent Store" class="md-header__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Alliance Persistent Store
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              DiseaseAnnotation
            
          </span>
</div>
</div>
</div>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Alliance Persistent Store" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Alliance Persistent Store
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
<span class="md-ellipsis">
    Browse the Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../developing-the-model/">
<span class="md-ellipsis">
    Developing the Alliance Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="https://linkml.io/linkml/intro/tutorial.html">
<span class="md-ellipsis">
    LinkML Tutorial
    
  </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#parent-classes">
<span class="md-ellipsis">
      Parent Classes
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#children">
<span class="md-ellipsis">
      Children
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
<span class="md-ellipsis">
      Slots
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
<span class="md-ellipsis">
      Mappings
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-specification">
<span class="md-ellipsis">
      LinkML Specification
    </span>
</a>
<nav aria-label="LinkML Specification" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
<span class="md-ellipsis">
      Direct
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
<span class="md-ellipsis">
      Induced
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="diseaseannotation">DiseaseAnnotation</h1>
<p>An annotation asserting an association between a biological entity and a disease supported by evidence.</p>
<div class="mermaid"> classDiagram
      Annotation &lt;|-- DiseaseAnnotation

      DiseaseAnnotation : annotation_type
      DiseaseAnnotation : condition_relations
      DiseaseAnnotation : created_by
      DiseaseAnnotation : curie
      DiseaseAnnotation : data_provider
      DiseaseAnnotation : data_provider_cross_reference
      DiseaseAnnotation : date_created
      DiseaseAnnotation : date_updated
      DiseaseAnnotation : db_date_created
      DiseaseAnnotation : db_date_updated
      DiseaseAnnotation : disease_annotation_object
      DiseaseAnnotation : disease_annotation_subject
      DiseaseAnnotation : disease_genetic_modifier_agms
      DiseaseAnnotation : disease_genetic_modifier_alleles
      DiseaseAnnotation : disease_genetic_modifier_genes
      DiseaseAnnotation : disease_genetic_modifier_relation
      DiseaseAnnotation : disease_qualifiers
      DiseaseAnnotation : evidence_codes
      DiseaseAnnotation : evidence_item
      DiseaseAnnotation : genetic_sex
      DiseaseAnnotation : internal
      DiseaseAnnotation : mod_internal_id
      DiseaseAnnotation : negated
      DiseaseAnnotation : obsolete
      DiseaseAnnotation : primary_external_id
      DiseaseAnnotation : related_notes
      DiseaseAnnotation : relation
      DiseaseAnnotation : secondary_data_provider
      DiseaseAnnotation : secondary_data_provider_cross_reference
      DiseaseAnnotation : unique_id
      DiseaseAnnotation : updated_by
      DiseaseAnnotation : with_or_from


</div>
<p>URI: <a href="http://alliancegenome.org/DiseaseAnnotation">alliance:DiseaseAnnotation</a></p>
<h2 id="parent-classes">Parent Classes</h2>
<ul>
<li><a href="../AuditedObject/">AuditedObject</a><ul>
<li><a href="../Association/">Association</a><ul>
<li><a href="../SingleReferenceAssociation/">SingleReferenceAssociation</a><ul>
<li><a href="../Annotation/">Annotation</a><ul>
<li><strong>DiseaseAnnotation</strong><ul>
<li><a href="../GeneDiseaseAnnotation/">GeneDiseaseAnnotation</a></li>
<li><a href="../AlleleDiseaseAnnotation/">AlleleDiseaseAnnotation</a></li>
<li><a href="../AGMDiseaseAnnotation/">AGMDiseaseAnnotation</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="children">Children</h2>
<ul>
<li><a href="../AuditedObject/">AuditedObject</a><ul>
<li><a href="../Association/">Association</a><ul>
<li><a href="../SingleReferenceAssociation/">SingleReferenceAssociation</a><ul>
<li><a href="../Annotation/">Annotation</a><ul>
<li><strong>DiseaseAnnotation</strong><ul>
<li><a href="../GeneDiseaseAnnotation/">GeneDiseaseAnnotation</a></li>
<li><a href="../AlleleDiseaseAnnotation/">AlleleDiseaseAnnotation</a></li>
<li><a href="../AGMDiseaseAnnotation/">AGMDiseaseAnnotation</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../annotation_type/">annotation_type</a></td>
<td>The type of annotation classified according to curation method. Submitted value should be a vocabulary term from the 'Annotation types' vocabulary</td>
</tr>
<tr>
<td><a href="../condition_relations/">condition_relations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../created_by/">created_by</a></td>
<td>The individual that created the entity.</td>
</tr>
<tr>
<td><a href="../curie/">curie</a></td>
<td>The Alliance-minted ID for the disease annotation. The ID is of the format AGRKB:100000000000001, where the first three digits represent the DiseaseAnnotation class code of "100", followed by a 12-digit identifier</td>
</tr>
<tr>
<td><a href="../data_provider/">data_provider</a></td>
<td>The organization (e.g. MOD) from which the data was sourced</td>
</tr>
<tr>
<td><a href="../data_provider_cross_reference/">data_provider_cross_reference</a></td>
<td>CrossReference to the organization from which the data was sourced</td>
</tr>
<tr>
<td><a href="../date_created/">date_created</a></td>
<td>The date on which an entity was created. This can be applied to nodes or edges.</td>
</tr>
<tr>
<td><a href="../date_updated/">date_updated</a></td>
<td>Date on which an entity was last modified.</td>
</tr>
<tr>
<td><a href="../db_date_created/">db_date_created</a></td>
<td>The date on which an entity was created in the Alliance database.  This is distinct from date_created, which represents the date when the entity was originally created (i.e. at the MOD for imported data).</td>
</tr>
<tr>
<td><a href="../db_date_updated/">db_date_updated</a></td>
<td>Date on which an entity was last modified in the Alliance database.  This is distinct from date_updated, which represents the date when the entity was last modified and may predate import into the Alliance database.</td>
</tr>
<tr>
<td><a href="../disease_annotation_object/">disease_annotation_object</a></td>
<td>The Disease Ontology term</td>
</tr>
<tr>
<td><a href="../disease_annotation_subject/">disease_annotation_subject</a></td>
<td>The biological entity to which the disease ontology term is associated.</td>
</tr>
<tr>
<td><a href="../disease_genetic_modifier_agms/">disease_genetic_modifier_agms</a></td>
<td>Specifies an affected genomic model (e.g. genotype) that modifies a disease model.</td>
</tr>
<tr>
<td><a href="../disease_genetic_modifier_alleles/">disease_genetic_modifier_alleles</a></td>
<td>Specifies an allele that modifies a disease model.</td>
</tr>
<tr>
<td><a href="../disease_genetic_modifier_genes/">disease_genetic_modifier_genes</a></td>
<td>Specifies a gene that modifies a disease model.</td>
</tr>
<tr>
<td><a href="../disease_genetic_modifier_relation/">disease_genetic_modifier_relation</a></td>
<td>A relation describing how the genetic modifier modifies the disease model. Submitted value should be a vocabulary term from the 'Disease genetic modifiers' vocabulary</td>
</tr>
<tr>
<td><a href="../disease_qualifiers/">disease_qualifiers</a></td>
<td>Submitted values should be vocabulary terms from the 'Disease qualifiers' Vocabulary</td>
</tr>
<tr>
<td><a href="../evidence_codes/">evidence_codes</a></td>
<td>ECO term IDs</td>
</tr>
<tr>
<td><a href="../evidence_item/">evidence_item</a></td>
<td>Single item of evidence that supports some assertion.</td>
</tr>
<tr>
<td><a href="../genetic_sex/">genetic_sex</a></td>
<td>Submitted value should be a vocabulary term from the 'Genetic sexes' vocabulary</td>
</tr>
<tr>
<td><a href="../internal/">internal</a></td>
<td>Classifies the entity as private (for internal use) or not (for public use).</td>
</tr>
<tr>
<td><a href="../mod_internal_id/">mod_internal_id</a></td>
<td>The model organism database (MOD) internal identifier for the disease annotation. When available, this is used for determination of distinctness instead of the Unique ID field.</td>
</tr>
<tr>
<td><a href="../negated/">negated</a></td>
<td>The negative qualifier for the annotation.</td>
</tr>
<tr>
<td><a href="../obsolete/">obsolete</a></td>
<td>Entity is no longer current.</td>
</tr>
<tr>
<td><a href="../primary_external_id/">primary_external_id</a></td>
<td>The primary external (non-AGRKB) identifier/curie for the disease annotation. Currently only used by WormBase for disease annotations, e.g. "WBDOannot00000907"</td>
</tr>
<tr>
<td><a href="../related_notes/">related_notes</a></td>
<td>Valid note types are available for viewing in the A-Team curation tool Controlled Vocabulary Terms Table (in the "Disease annotation note types" vocabulary) on the production environment (curation.alliancegenome.org). New terms can be added as needed.</td>
</tr>
<tr>
<td><a href="../relation/">relation</a></td>
<td>The relation between the subject biological entity and disease in the annotation. Permissible values are terms from the CV 'Disease Relation'</td>
</tr>
<tr>
<td><a href="../secondary_data_provider/">secondary_data_provider</a></td>
<td>Object representing organization (e.g. MOD) that provided the data directly to the Alliance, but not the original source</td>
</tr>
<tr>
<td><a href="../secondary_data_provider_cross_reference/">secondary_data_provider_cross_reference</a></td>
<td>CrossReference to the organization that provided the data to the Alliance</td>
</tr>
<tr>
<td><a href="../unique_id/">unique_id</a></td>
<td>Unique identifier for the disease annotation.  Will be generated at AGR if not submitted by the MOD.</td>
</tr>
<tr>
<td><a href="../updated_by/">updated_by</a></td>
<td>The individual that last modified the entity.</td>
</tr>
<tr>
<td><a href="../with_or_from/">with_or_from</a></td>
<td>A field for disease annotations that captures the human gene the implicated MOD gene is homologous or orthologous to when using the ISS (inferred from sequence similarity) evidence code or the ISO (inferred from sequence orthology) evidence code. The assertion would state that the MOD gene is implicated in the indicated disease based on sequence-similarity/sequence-orthology "with" some human gene. Currently used by SGD.</td>
</tr>
</tbody>
</table>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>['alliance:DiseaseAnnotation']</td>
</tr>
<tr>
<td>native</td>
<td>['alliance:DiseaseAnnotation']</td>
</tr>
</tbody>
</table>
<h2 id="linkml-specification">LinkML Specification</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: DiseaseAnnotation
description: An annotation asserting an association between a biological entity and
  a disease supported by evidence.
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: Annotation
abstract: true
slots:
- disease_annotation_subject
- negated
- evidence_codes
- annotation_type
- with_or_from
- disease_qualifiers
- genetic_sex
- secondary_data_provider
- secondary_data_provider_cross_reference
- disease_genetic_modifier_genes
- disease_genetic_modifier_alleles
- disease_genetic_modifier_agms
- disease_genetic_modifier_relation
slot_usage:
  curie:
    name: curie
    description: The Alliance-minted ID for the disease annotation. The ID is of the
      format AGRKB:100000000000001, where the first three digits represent the DiseaseAnnotation
      class code of "100", followed by a 12-digit identifier
  unique_id:
    name: unique_id
    description: Unique identifier for the disease annotation.  Will be generated
      at AGR if not submitted by the MOD.
    required: false
  primary_external_id:
    name: primary_external_id
    description: The primary external (non-AGRKB) identifier/curie for the disease
      annotation. Currently only used by WormBase for disease annotations, e.g. "WBDOannot00000907"
    required: false
  mod_internal_id:
    name: mod_internal_id
    description: The model organism database (MOD) internal identifier for the disease
      annotation. When available, this is used for determination of distinctness instead
      of the Unique ID field.
    required: false
  negated:
    name: negated
    description: The negative qualifier for the annotation.
  relation:
    name: relation
    description: The relation between the subject biological entity and disease in
      the annotation. Permissible values are terms from the CV 'Disease Relation'
    required: true
  annotation_type:
    name: annotation_type
    required: false
  with_or_from:
    name: with_or_from
    description: A field for disease annotations that captures the human gene the
      implicated MOD gene is homologous or orthologous to when using the ISS (inferred
      from sequence similarity) evidence code or the ISO (inferred from sequence orthology)
      evidence code. The assertion would state that the MOD gene is implicated in
      the indicated disease based on sequence-similarity/sequence-orthology "with"
      some human gene. Currently used by SGD.
    required: false
  single_reference:
    name: single_reference
    description: The reference in which the disease association was asserted/reported.
    required: true
  evidence_codes:
    name: evidence_codes
    required: true
    multivalued: true
  genetic_sex:
    name: genetic_sex
    description: Submitted value should be a vocabulary term from the 'Genetic sexes'
      vocabulary
    range: VocabularyTerm
    required: false
  related_notes:
    name: related_notes
    description: Valid note types are available for viewing in the A-Team curation
      tool Controlled Vocabulary Terms Table (in the "Disease annotation note types"
      vocabulary) on the production environment (curation.alliancegenome.org). New
      terms can be added as needed.
attributes:
  disease_annotation_object:
    name: disease_annotation_object
    description: The Disease Ontology term
    from_schema: https://github.com/alliance-genome/agr_persistent_schema/phenotypeAndDiseaseAnnotation.yaml
    domain_of:
    - DiseaseAnnotation
    range: DOTerm
    required: true

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: DiseaseAnnotation
description: An annotation asserting an association between a biological entity and
  a disease supported by evidence.
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: Annotation
abstract: true
slot_usage:
  curie:
    name: curie
    description: The Alliance-minted ID for the disease annotation. The ID is of the
      format AGRKB:100000000000001, where the first three digits represent the DiseaseAnnotation
      class code of "100", followed by a 12-digit identifier
  unique_id:
    name: unique_id
    description: Unique identifier for the disease annotation.  Will be generated
      at AGR if not submitted by the MOD.
    required: false
  primary_external_id:
    name: primary_external_id
    description: The primary external (non-AGRKB) identifier/curie for the disease
      annotation. Currently only used by WormBase for disease annotations, e.g. "WBDOannot00000907"
    required: false
  mod_internal_id:
    name: mod_internal_id
    description: The model organism database (MOD) internal identifier for the disease
      annotation. When available, this is used for determination of distinctness instead
      of the Unique ID field.
    required: false
  negated:
    name: negated
    description: The negative qualifier for the annotation.
  relation:
    name: relation
    description: The relation between the subject biological entity and disease in
      the annotation. Permissible values are terms from the CV 'Disease Relation'
    required: true
  annotation_type:
    name: annotation_type
    required: false
  with_or_from:
    name: with_or_from
    description: A field for disease annotations that captures the human gene the
      implicated MOD gene is homologous or orthologous to when using the ISS (inferred
      from sequence similarity) evidence code or the ISO (inferred from sequence orthology)
      evidence code. The assertion would state that the MOD gene is implicated in
      the indicated disease based on sequence-similarity/sequence-orthology "with"
      some human gene. Currently used by SGD.
    required: false
  single_reference:
    name: single_reference
    description: The reference in which the disease association was asserted/reported.
    required: true
  evidence_codes:
    name: evidence_codes
    required: true
    multivalued: true
  genetic_sex:
    name: genetic_sex
    description: Submitted value should be a vocabulary term from the 'Genetic sexes'
      vocabulary
    range: VocabularyTerm
    required: false
  related_notes:
    name: related_notes
    description: Valid note types are available for viewing in the A-Team curation
      tool Controlled Vocabulary Terms Table (in the "Disease annotation note types"
      vocabulary) on the production environment (curation.alliancegenome.org). New
      terms can be added as needed.
attributes:
  disease_annotation_object:
    name: disease_annotation_object
    description: The Disease Ontology term
    from_schema: https://github.com/alliance-genome/agr_persistent_schema/phenotypeAndDiseaseAnnotation.yaml
    alias: disease_annotation_object
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: DOTerm
    required: true
  disease_annotation_subject:
    name: disease_annotation_subject
    description: The biological entity to which the disease ontology term is associated.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: DiseaseAnnotation
    alias: disease_annotation_subject
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: BiologicalEntity
    required: true
  negated:
    name: negated
    description: The negative qualifier for the annotation.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: negated
    owner: DiseaseAnnotation
    domain_of:
    - ExpressionAnnotation
    - PhenotypeAnnotation
    - PhenotypeAnnotationDTO
    - DiseaseAnnotation
    - DiseaseAnnotationDTO
    range: boolean
  evidence_codes:
    name: evidence_codes
    description: ECO term IDs
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    values_from:
    - ECO
    alias: evidence_codes
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: ECOTerm
    required: true
    multivalued: true
  annotation_type:
    name: annotation_type
    description: The type of annotation classified according to curation method. Submitted
      value should be a vocabulary term from the 'Annotation types' vocabulary
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: annotation_type
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: VocabularyTerm
    required: false
  with_or_from:
    name: with_or_from
    description: A field for disease annotations that captures the human gene the
      implicated MOD gene is homologous or orthologous to when using the ISS (inferred
      from sequence similarity) evidence code or the ISO (inferred from sequence orthology)
      evidence code. The assertion would state that the MOD gene is implicated in
      the indicated disease based on sequence-similarity/sequence-orthology "with"
      some human gene. Currently used by SGD.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: with_or_from
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: Gene
    required: false
    multivalued: true
  disease_qualifiers:
    name: disease_qualifiers
    description: Submitted values should be vocabulary terms from the 'Disease qualifiers'
      Vocabulary
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: DiseaseAnnotation
    alias: disease_qualifiers
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: VocabularyTerm
    multivalued: true
  genetic_sex:
    name: genetic_sex
    description: Submitted value should be a vocabulary term from the 'Genetic sexes'
      vocabulary
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: genetic_sex
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    - HTPExpressionDatasetSampleAnnotation
    range: VocabularyTerm
    required: false
  secondary_data_provider:
    name: secondary_data_provider
    description: Object representing organization (e.g. MOD) that provided the data
      directly to the Alliance, but not the original source
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: secondary_data_provider
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: Organization
    multivalued: false
  secondary_data_provider_cross_reference:
    name: secondary_data_provider_cross_reference
    description: CrossReference to the organization that provided the data to the
      Alliance
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: secondary_data_provider_cross_reference
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: CrossReference
    multivalued: false
  disease_genetic_modifier_genes:
    name: disease_genetic_modifier_genes
    description: Specifies a gene that modifies a disease model.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: disease_genetic_modifier_genes
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: Gene
    required: false
    multivalued: true
  disease_genetic_modifier_alleles:
    name: disease_genetic_modifier_alleles
    description: Specifies an allele that modifies a disease model.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: disease_genetic_modifier_alleles
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: Allele
    required: false
    multivalued: true
  disease_genetic_modifier_agms:
    name: disease_genetic_modifier_agms
    description: Specifies an affected genomic model (e.g. genotype) that modifies
      a disease model.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: disease_genetic_modifier_agms
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: AffectedGenomicModel
    required: false
    multivalued: true
  disease_genetic_modifier_relation:
    name: disease_genetic_modifier_relation
    description: A relation describing how the genetic modifier modifies the disease
      model. Submitted value should be a vocabulary term from the 'Disease genetic
      modifiers' vocabulary
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: disease_genetic_modifier_relation
    owner: DiseaseAnnotation
    domain_of:
    - DiseaseAnnotation
    range: VocabularyTerm
    multivalued: false
  curie:
    name: curie
    description: The Alliance-minted ID for the disease annotation. The ID is of the
      format AGRKB:100000000000001, where the first three digits represent the DiseaseAnnotation
      class code of "100", followed by a 12-digit identifier
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: curie
    owner: DiseaseAnnotation
    domain_of:
    - Laboratory
    - CurieObject
    - Identifier
    - GeneNomenclatureSet
    - OntologyTerm
    - Annotation
    - ExternalDataBaseEntityDTO
    range: uriorcurie
    multivalued: false
  unique_id:
    name: unique_id
    description: Unique identifier for the disease annotation.  Will be generated
      at AGR if not submitted by the MOD.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: unique_id
    owner: DiseaseAnnotation
    domain_of:
    - Person
    - Exon
    - CodingSequence
    - ExpressionExperiment
    - GeneInteraction
    - Reagent
    - Annotation
    - ExperimentalCondition
    - ConditionRelation
    range: string
    required: false
    multivalued: false
  primary_external_id:
    name: primary_external_id
    description: The primary external (non-AGRKB) identifier/curie for the disease
      annotation. Currently only used by WormBase for disease annotations, e.g. "WBDOannot00000907"
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: primary_external_id
    owner: DiseaseAnnotation
    domain_of:
    - Person
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
    required: false
  mod_internal_id:
    name: mod_internal_id
    description: The model organism database (MOD) internal identifier for the disease
      annotation. When available, this is used for determination of distinctness instead
      of the Unique ID field.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: mod_internal_id
    owner: DiseaseAnnotation
    domain_of:
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
    required: false
  condition_relations:
    name: condition_relations
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: condition_relations
    owner: DiseaseAnnotation
    domain_of:
    - ExpressionExperiment
    - Annotation
    range: ConditionRelation
    multivalued: true
  related_notes:
    name: related_notes
    description: Valid note types are available for viewing in the A-Team curation
      tool Controlled Vocabulary Terms Table (in the "Disease annotation note types"
      vocabulary) on the production environment (curation.alliancegenome.org). New
      terms can be added as needed.
    notes:
    - The original name suggested for this slot was simply notes, but I think that
      label is reserved in LinkML for internal descriptions of objects, hence the
      related_notes label used here.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: related_notes
    owner: DiseaseAnnotation
    domain_of:
    - SubmittedObject
    - Gene
    - GeneChangeEventSlotAnnotation
    - ExpressionExperiment
    - ConstructComponentSlotAnnotation
    - ConstructGenomicEntityAssociation
    - ConstructCassetteAssociation
    - CassetteComponentSlotAnnotation
    - CassetteGenomicEntityAssociation
    - CassetteTransgenicToolAssociation
    - CassetteStrAssociation
    - Annotation
    - HTPExpressionDatasetSampleAnnotation
    range: Note
    multivalued: true
  data_provider:
    name: data_provider
    description: The organization (e.g. MOD) from which the data was sourced
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: data_provider
    owner: DiseaseAnnotation
    domain_of:
    - SubmittedObject
    - Chromosome
    - Species
    - Annotation
    - HTPExpressionDatasetAnnotation
    range: Organization
    required: true
    multivalued: false
  data_provider_cross_reference:
    name: data_provider_cross_reference
    description: CrossReference to the organization from which the data was sourced
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: data_provider_cross_reference
    owner: DiseaseAnnotation
    domain_of:
    - SubmittedObject
    - Chromosome
    - Species
    - Annotation
    - HTPExpressionDatasetAnnotation
    range: CrossReference
    required: false
    multivalued: false
  evidence_item:
    name: evidence_item
    description: Single item of evidence that supports some assertion.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: evidence_item
    owner: DiseaseAnnotation
    domain_of:
    - SingleReferenceAssociation
    range: InformationContentEntity
  relation:
    name: relation
    description: The relation between the subject biological entity and disease in
      the annotation. Permissible values are terms from the CV 'Disease Relation'
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    exact_mappings:
    - biolink:predicate
    is_a: association_slot
    alias: relation
    owner: DiseaseAnnotation
    domain_of:
    - Association
    - ConstructComponentSlotAnnotation
    - CassetteComponentSlotAnnotation
    range: string
    required: true
  created_by:
    name: created_by
    description: The individual that created the entity.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObject
    alias: created_by
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    range: Person
    multivalued: false
  date_created:
    name: date_created
    description: The date on which an entity was created. This can be applied to nodes
      or edges.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - creation_date
    exact_mappings:
    - dct:createdOn
    - WIKIDATA_PROPERTY:P577
    alias: date_created
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  updated_by:
    name: updated_by
    description: The individual that last modified the entity.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObject
    alias: updated_by
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    range: Person
    multivalued: false
  date_updated:
    name: date_updated
    description: Date on which an entity was last modified.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - date_last_modified
    alias: date_updated
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_created:
    name: db_date_created
    description: The date on which an entity was created in the Alliance database.  This
      is distinct from date_created, which represents the date when the entity was
      originally created (i.e. at the MOD for imported data).
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_created
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_updated:
    name: db_date_updated
    description: Date on which an entity was last modified in the Alliance database.  This
      is distinct from date_updated, which represents the date when the entity was
      last modified and may predate import into the Alliance database.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_updated
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  internal:
    name: internal
    description: Classifies the entity as private (for internal use) or not (for public
      use).
    notes:
    - Default value is true.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: internal
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean
    required: true
  obsolete:
    name: obsolete
    description: Entity is no longer current.
    notes:
    - Obsolete entities are preserved in the database for posterity but should not
      be publicly displayed.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: obsolete
    owner: DiseaseAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean

</code></pre>
</details>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
<script src="../assets/javascripts/bundle.13a4f30d.min.js"></script>
<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>