
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://github.com/alliance-genome/agr_curation_schema/">
      
      
      
        <link rel="next" href="developing-the-model/">
      
      
      <link rel="icon" href="assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>Alliance Persistent Store</title>
      
    
    
      <link rel="stylesheet" href="assets/stylesheets/main.342714a4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL(".",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#alliancemodel" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="." title="Alliance Persistent Store" class="md-header__button md-logo" aria-label="Alliance Persistent Store" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Alliance Persistent Store
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Browse the Model
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="." title="Alliance Persistent Store" class="md-nav__button md-logo" aria-label="Alliance Persistent Store" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Alliance Persistent Store
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Browse the Model
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="." class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Browse the Model
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="developing-the-model/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Developing the Alliance Model
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="https://linkml.io/linkml/intro/tutorial.html" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    LinkML Tutorial
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="alliancemodel">AllianceModel</h1>
<p>None</p>
<p>URI: https://github.com/alliance-genome/agr_curation_schema/alliance_schema</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AffectedGenomicModel/">AffectedGenomicModel</a></td>
<td>Includes inbred strains, stocks, disease models and mutant genotypes</td>
</tr>
<tr>
<td><a href="AffectedGenomicModelDTO/">AffectedGenomicModelDTO</a></td>
<td>Ingest class for AGMs</td>
</tr>
<tr>
<td><a href="Agent/">Agent</a></td>
<td>An individual, group, organization or project that provides information and/or materials.</td>
</tr>
<tr>
<td><a href="AgmAgmAssociation/">AgmAgmAssociation</a></td>
<td>Association (without evidence) between two AffectedGenomicModels</td>
</tr>
<tr>
<td><a href="AgmAgmAssociationDTO/">AgmAgmAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AgmAlleleAssociation/">AgmAlleleAssociation</a></td>
<td>Association (without evidence) between two AffectedGenomicModels</td>
</tr>
<tr>
<td><a href="AgmAlleleAssociationDTO/">AgmAlleleAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AGMDiseaseAnnotation/">AGMDiseaseAnnotation</a></td>
<td>An annotation asserting an association between an AGM and a disease supported by evidence.</td>
</tr>
<tr>
<td><a href="AGMDiseaseAnnotationDTO/">AGMDiseaseAnnotationDTO</a></td>
<td>Ingest class for an association between an AGM and a disease</td>
</tr>
<tr>
<td><a href="AgmFullNameSlotAnnotation/">AgmFullNameSlotAnnotation</a></td>
<td>The one current full name for the AGM</td>
</tr>
<tr>
<td><a href="AGMPhenotypeAnnotation/">AGMPhenotypeAnnotation</a></td>
<td>An annotation asserting an association between an AGM and a phenotype supported by evidence.</td>
</tr>
<tr>
<td><a href="AGMPhenotypeAnnotationDTO/">AGMPhenotypeAnnotationDTO</a></td>
<td>Ingest class for an association between an AGM and a phenotype</td>
</tr>
<tr>
<td><a href="AgmSecondaryIdSlotAnnotation/">AgmSecondaryIdSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AgmStrAssociation/">AgmStrAssociation</a></td>
<td>Association (without evidence) between two AffectedGenomicModels</td>
</tr>
<tr>
<td><a href="AgmStrAssociationDTO/">AgmStrAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AgmSynonymSlotAnnotation/">AgmSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the AGM. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="Allele/">Allele</a></td>
<td>One of multiple possible forms of a functional genomic element (most often described as a locus or gene), differing from the reference DNA sequence.  The genomic element can be endogenous or constructed.</td>
</tr>
<tr>
<td><a href="AlleleAlleleAssociation/">AlleleAlleleAssociation</a></td>
<td>Association between an allele and another allele</td>
</tr>
<tr>
<td><a href="AlleleAlleleAssociationDTO/">AlleleAlleleAssociationDTO</a></td>
<td>Association between an allele and another allele</td>
</tr>
<tr>
<td><a href="AlleleAssociation/">AlleleAssociation</a></td>
<td>Base class for all Allele Associations</td>
</tr>
<tr>
<td><a href="AlleleCellLineAssociation/">AlleleCellLineAssociation</a></td>
<td>The relationship between an allele and a cell line.  Includes mutant/ embryonic stem cell lines known to carry the allele, and parental cell line of alleles made in embryonic stem cells.</td>
</tr>
<tr>
<td><a href="AlleleCellLineAssociationDTO/">AlleleCellLineAssociationDTO</a></td>
<td>The relationship between an allele and a cell line.  Includes mutant/ embryonic stem cell lines known to carry the allele, and parental cell line of alleles made in embryonic stem cells.</td>
</tr>
<tr>
<td><a href="AlleleConstructAssociation/">AlleleConstructAssociation</a></td>
<td>The relationship between an allele and constructs contained in that allele.</td>
</tr>
<tr>
<td><a href="AlleleConstructAssociationDTO/">AlleleConstructAssociationDTO</a></td>
<td>The relationship between an allele and constructs contained in that allele.</td>
</tr>
<tr>
<td><a href="AlleleDatabaseStatusSlotAnnotation/">AlleleDatabaseStatusSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleDatabaseStatusSlotAnnotationDTO/">AlleleDatabaseStatusSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleDiseaseAnnotation/">AlleleDiseaseAnnotation</a></td>
<td>An annotation asserting an association between an allele and a disease supported by evidence.</td>
</tr>
<tr>
<td><a href="AlleleDiseaseAnnotationDTO/">AlleleDiseaseAnnotationDTO</a></td>
<td>Ingest class for an association between an allele and a disease</td>
</tr>
<tr>
<td><a href="AlleleDTO/">AlleleDTO</a></td>
<td>Ingest class for an Allele object</td>
</tr>
<tr>
<td><a href="AlleleFullNameSlotAnnotation/">AlleleFullNameSlotAnnotation</a></td>
<td>The one current full name for the allele.</td>
</tr>
<tr>
<td><a href="AlleleFunctionalImpactSlotAnnotation/">AlleleFunctionalImpactSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleFunctionalImpactSlotAnnotationDTO/">AlleleFunctionalImpactSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleGeneAssociation/">AlleleGeneAssociation</a></td>
<td>Association between an allele and a gene</td>
</tr>
<tr>
<td><a href="AlleleGeneAssociationDTO/">AlleleGeneAssociationDTO</a></td>
<td>Association between an allele and a gene</td>
</tr>
<tr>
<td><a href="AlleleGenerationMethodAssociation/">AlleleGenerationMethodAssociation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleGenerationMethodAssociationDTO/">AlleleGenerationMethodAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleGenomicEntityAssociation/">AlleleGenomicEntityAssociation</a></td>
<td>Association between an allele and a genomic entity</td>
</tr>
<tr>
<td><a href="AlleleGenomicEntityAssociationDTO/">AlleleGenomicEntityAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleGermlineTransmissionStatusSlotAnnotation/">AlleleGermlineTransmissionStatusSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleGermlineTransmissionStatusSlotAnnotationDTO/">AlleleGermlineTransmissionStatusSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleImageAssociation/">AlleleImageAssociation</a></td>
<td>The relationship between an allele and an image.</td>
</tr>
<tr>
<td><a href="AlleleImageAssociationDTO/">AlleleImageAssociationDTO</a></td>
<td>The relationship between an allele and an image.</td>
</tr>
<tr>
<td><a href="AlleleInheritanceModeSlotAnnotation/">AlleleInheritanceModeSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleInheritanceModeSlotAnnotationDTO/">AlleleInheritanceModeSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleMutationTypeSlotAnnotation/">AlleleMutationTypeSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleMutationTypeSlotAnnotationDTO/">AlleleMutationTypeSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleNomenclatureEventSlotAnnotation/">AlleleNomenclatureEventSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleNomenclatureEventSlotAnnotationDTO/">AlleleNomenclatureEventSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleOriginAssociation/">AlleleOriginAssociation</a></td>
<td>The relationship between an allele and the origin of the allele.</td>
</tr>
<tr>
<td><a href="AlleleOriginAssociationDTO/">AlleleOriginAssociationDTO</a></td>
<td>The relationship between an allele and the AGM origin of the allele.</td>
</tr>
<tr>
<td><a href="AllelePhenotypeAnnotation/">AllelePhenotypeAnnotation</a></td>
<td>An annotation asserting an association between an allele and a phenotype supported by evidence.</td>
</tr>
<tr>
<td><a href="AllelePhenotypeAnnotationDTO/">AllelePhenotypeAnnotationDTO</a></td>
<td>Ingest class for an association between an allele and a phenotype</td>
</tr>
<tr>
<td><a href="AlleleProteinAssociation/">AlleleProteinAssociation</a></td>
<td>Association between an allele and a protein</td>
</tr>
<tr>
<td><a href="AlleleProteinAssociationDTO/">AlleleProteinAssociationDTO</a></td>
<td>Association between an allele and a protein</td>
</tr>
<tr>
<td><a href="AlleleSecondaryIdSlotAnnotation/">AlleleSecondaryIdSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="AlleleSymbolSlotAnnotation/">AlleleSymbolSlotAnnotation</a></td>
<td>The one current symbol for the allele.</td>
</tr>
<tr>
<td><a href="AlleleSynonymSlotAnnotation/">AlleleSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the allele. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="AlleleTranscriptAssociation/">AlleleTranscriptAssociation</a></td>
<td>Association between an allele and a transcript</td>
</tr>
<tr>
<td><a href="AlleleTranscriptAssociationDTO/">AlleleTranscriptAssociationDTO</a></td>
<td>Association between an allele and a transcript</td>
</tr>
<tr>
<td><a href="AlleleVariantAssociation/">AlleleVariantAssociation</a></td>
<td>The relationship between an allele and a variant is many to many. An Allele may have many variants and a variant can be present in many alleles.</td>
</tr>
<tr>
<td><a href="AlleleVariantAssociationDTO/">AlleleVariantAssociationDTO</a></td>
<td>The relationship between an allele and a variant is many to many. An Allele may have many variants and a variant can be present in many alleles.</td>
</tr>
<tr>
<td><a href="AllianceMember/">AllianceMember</a></td>
<td>An organization that is a member of the Alliance of Genome Resources.</td>
</tr>
<tr>
<td><a href="AnatomicalSite/">AnatomicalSite</a></td>
<td>The set of terms that define a specific anatomical site within a specimen.</td>
</tr>
<tr>
<td><a href="AnatomicalSiteDTO/">AnatomicalSiteDTO</a></td>
<td>The set of terms that define a specific anatomical site within a specimen.</td>
</tr>
<tr>
<td><a href="AnatomicalTerm/">AnatomicalTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Annotation/">Annotation</a></td>
<td>An annotation describing the link between a "first class citizen" of the database (e.g. Gene / Allele / AGM) and an ontology term or description, supported by a single reference and an optional list of condition relations.</td>
</tr>
<tr>
<td><a href="AnnotationDTO/">AnnotationDTO</a></td>
<td>Ingest class for an annotation</td>
</tr>
<tr>
<td><a href="Antibody/">Antibody</a></td>
<td>Immunoglobulin proteins that bind specific molecule(s). Can be used experimentally for the purposes of detection or purification.</td>
</tr>
<tr>
<td><a href="AntibodyDTO/">AntibodyDTO</a></td>
<td>Ingest class for immunoglobulin proteins that bind specific molecule(s). Can be used experimentally for the purposes of detection or purification.</td>
</tr>
<tr>
<td><a href="APOTerm/">APOTerm</a></td>
<td>A phenotype ontology term from the Ascomycete Phenotype Ontology (APO). PURL: http://purl.obolibrary.org/obo/APO.owl</td>
</tr>
<tr>
<td><a href="AssemblyComponent/">AssemblyComponent</a></td>
<td>A scaffold component of a genome assembly, including chromosomal pseudo-molecule sequences, unplaced/unlocalized scaffolds and contigs, and organelle genomes.</td>
</tr>
<tr>
<td><a href="AssemblyComponentDTO/">AssemblyComponentDTO</a></td>
<td>Ingest class for assembly components</td>
</tr>
<tr>
<td><a href="Association/">Association</a></td>
<td>A typed association between two entities, supported by evidence.  Associations have three base slots: subject, object, and relation, but they can have any number of additional attributes that help qualify the relationship between the subject and the object.  The subject is the curie (or identifier) of the class that is the subject of the association, and likewise the object is the curie (or identifier of the class that is the object. The relationship between subject and object is defined by the relation slot (which can also be constrained using the range of the relation).</td>
</tr>
<tr>
<td><a href="ATPTerm/">ATPTerm</a></td>
<td>An ontology term from the Alliance Tags for Papers ontology (ATP)</td>
</tr>
<tr>
<td><a href="AuditedObject/">AuditedObject</a></td>
<td>Base class for all other LinkML classes. Some entity for which changes are tracked, including date/time of change, the agent responsible for the change, and whether the entity is internal (private).</td>
</tr>
<tr>
<td><a href="AuditedObjectDTO/">AuditedObjectDTO</a></td>
<td>Base class for all other LinkML DTO classes.</td>
</tr>
<tr>
<td><a href="AuthorReference/">AuthorReference</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="BiologicalEntity/">BiologicalEntity</a></td>
<td>An entity of biological origin that can be unambiguously attributed to a single species.</td>
</tr>
<tr>
<td><a href="BiologicalEntityDTO/">BiologicalEntityDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="BioSampleAge/">BioSampleAge</a></td>
<td>An object that holds the temporal aspect of a sample.</td>
</tr>
<tr>
<td><a href="BioSampleAgeDTO/">BioSampleAgeDTO</a></td>
<td>An object that holds the temporal aspect of a sample.</td>
</tr>
<tr>
<td><a href="BioSampleGenomicInformation/">BioSampleGenomicInformation</a></td>
<td>A basic entry for defining a biosample's genomic information</td>
</tr>
<tr>
<td><a href="BioSampleGenomicInformationDTO/">BioSampleGenomicInformationDTO</a></td>
<td>A basic (DTO) entry for defining a biosample's genomic information</td>
</tr>
<tr>
<td><a href="BSPOTerm/">BSPOTerm</a></td>
<td>An ontology term from the Biological Spatial Ontology (BSPO)</td>
</tr>
<tr>
<td><a href="BulkFMSLoad/">BulkFMSLoad</a></td>
<td>This bulk load automatically pulls files from the BulkFMSLoad</td>
</tr>
<tr>
<td><a href="BulkLoad/">BulkLoad</a></td>
<td>Base class for all loads</td>
</tr>
<tr>
<td><a href="BulkLoadFile/">BulkLoadFile</a></td>
<td>This class is used to hold version of the files being loaded</td>
</tr>
<tr>
<td><a href="BulkLoadFileHistory/">BulkLoadFileHistory</a></td>
<td>Object used to describe the indiviual run of this BulkLoadFile</td>
</tr>
<tr>
<td><a href="BulkLoadGroup/">BulkLoadGroup</a></td>
<td>This class is use to group together bulk load_files</td>
</tr>
<tr>
<td><a href="BulkManualLoad/">BulkManualLoad</a></td>
<td>This bulk load is used by DQM's to submit their files to the curation system</td>
</tr>
<tr>
<td><a href="BulkScheduledLoad/">BulkScheduledLoad</a></td>
<td>This a bulk load that includes a cron scheuld</td>
</tr>
<tr>
<td><a href="BulkURLLoad/">BulkURLLoad</a></td>
<td>This bulk load automatically pulls files from a defined BulkURLLoad</td>
</tr>
<tr>
<td><a href="Cassette/">Cassette</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CassetteAssociation/">CassetteAssociation</a></td>
<td>Base class for cassette associations</td>
</tr>
<tr>
<td><a href="CassetteComponentSlotAnnotation/">CassetteComponentSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CassetteComponentSlotAnnotationDTO/">CassetteComponentSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CassetteDTO/">CassetteDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CassetteFullNameSlotAnnotation/">CassetteFullNameSlotAnnotation</a></td>
<td>The one current full name for the cassette.</td>
</tr>
<tr>
<td><a href="CassetteGenomicEntityAssociation/">CassetteGenomicEntityAssociation</a></td>
<td>An association class to capture the identity of a cassette component that is a known genomic entity with a curie and any accompanying meta data about the relationship between the cassette and that genomic entity component</td>
</tr>
<tr>
<td><a href="CassetteGenomicEntityAssociationDTO/">CassetteGenomicEntityAssociationDTO</a></td>
<td>Ingest class for association between a cassette component that is a known genomic entity with a curie and any accompanying meta data about the relationship between the cassette and that genomic entity component</td>
</tr>
<tr>
<td><a href="CassetteStrAssociation/">CassetteStrAssociation</a></td>
<td>An association class to capture the identity of a sequence targeting reagent (STR) that is a component of the Cassette and any accompanying meta data about the relationship between the cassette and that STR component</td>
</tr>
<tr>
<td><a href="CassetteStrAssociationDTO/">CassetteStrAssociationDTO</a></td>
<td>An association class to capture the identity of a sequence targeting reagent (STR) that is a component of the Cassette and any accompanying meta data about the relationship between the cassette and that STR component</td>
</tr>
<tr>
<td><a href="CassetteSymbolSlotAnnotation/">CassetteSymbolSlotAnnotation</a></td>
<td>The one current symbol for the cassette.</td>
</tr>
<tr>
<td><a href="CassetteSynonymSlotAnnotation/">CassetteSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the cassette. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="CassetteTransgenicToolAssociation/">CassetteTransgenicToolAssociation</a></td>
<td>An association class to capture the identity of a cassette component that is a known transgenic tool and any accompanying meta data about the relationship between the cassette and that transgenic tool component</td>
</tr>
<tr>
<td><a href="CassetteTransgenicToolAssociationDTO/">CassetteTransgenicToolAssociationDTO</a></td>
<td>Ingest class for association between a cassette component that is a known transgenic tool and any accompanying meta data about the relationship between the cassette and that transgenic tool component</td>
</tr>
<tr>
<td><a href="CassetteUseSlotAnnotation/">CassetteUseSlotAnnotation</a></td>
<td>A slot annotation to capture the use(s) (e.g. enhancer trap, protein trap) of a particular Cassette with associated evidence.</td>
</tr>
<tr>
<td><a href="CassetteUseSlotAnnotationDTO/">CassetteUseSlotAnnotationDTO</a></td>
<td>Ingest class for a slot annotation to capture the use(s) (e.g. enhancer trap, protein trap) of a particular Cassette with associated evidence.</td>
</tr>
<tr>
<td><a href="CellLine/">CellLine</a></td>
<td>Dummy cell line class</td>
</tr>
<tr>
<td><a href="CellLineDTO/">CellLineDTO</a></td>
<td>Dummy cell line DTO class</td>
</tr>
<tr>
<td><a href="CHEBITerm/">CHEBITerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ChemicalTerm/">ChemicalTerm</a></td>
<td>An ontology term representing a chemical or molecule</td>
</tr>
<tr>
<td><a href="Chromosome/">Chromosome</a></td>
<td>A structure composed of a single DNA molecule and associated proteins that carries here hereditary information from cell to cell. In the context of this model, it is used as a landmark for cytogenetic and genetic mapping of genes, alleles and chromosomal aberrations; this entity ties together related chromosome sequences from different genome assemblies.</td>
</tr>
<tr>
<td><a href="ChromosomeDTO/">ChromosomeDTO</a></td>
<td>Ingest class for chromosomes</td>
</tr>
<tr>
<td><a href="CLTerm/">CLTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CMOTerm/">CMOTerm</a></td>
<td>An ontology term from the Clinical Measurement Ontology (CMO)</td>
</tr>
<tr>
<td><a href="CodingSequence/">CodingSequence</a></td>
<td>The coding sequence (CDS) is the portion of an exon that is translated  into a protein</td>
</tr>
<tr>
<td><a href="CodingSequenceDTO/">CodingSequenceDTO</a></td>
<td>Ingest class for transcripts</td>
</tr>
<tr>
<td><a href="CodingSequenceGenomicLocationAssociation/">CodingSequenceGenomicLocationAssociation</a></td>
<td>Location(s) of the CDS on the genome.</td>
</tr>
<tr>
<td><a href="Company/">Company</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ConditionRelation/">ConditionRelation</a></td>
<td>A pairing of an experimental condition relation (i.e. has_condition) with a list of 1 or more ExperimentalCondition objects. Annotation objects can connect directly to a set of 0 or more of these ConditionRelation objects via a 'condition_relations' slot to express the experimental conditions relevant to the annotation.</td>
</tr>
<tr>
<td><a href="ConditionRelationDTO/">ConditionRelationDTO</a></td>
<td>Ingest class for the pairing of an experimental condition relation with a list of one or more conditions</td>
</tr>
<tr>
<td><a href="Construct/">Construct</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ConstructAssociation/">ConstructAssociation</a></td>
<td>Base class for construct associations</td>
</tr>
<tr>
<td><a href="ConstructCassetteAssociation/">ConstructCassetteAssociation</a></td>
<td>An association class to capture the identity of a construct cassette that is part of the construct</td>
</tr>
<tr>
<td><a href="ConstructCassetteAssociationDTO/">ConstructCassetteAssociationDTO</a></td>
<td>Ingest class for the association class to capture the identity of a construct cassette that is part of the construct</td>
</tr>
<tr>
<td><a href="ConstructComponentSlotAnnotation/">ConstructComponentSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ConstructComponentSlotAnnotationDTO/">ConstructComponentSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ConstructDTO/">ConstructDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ConstructFullNameSlotAnnotation/">ConstructFullNameSlotAnnotation</a></td>
<td>The one current full name for the construct.</td>
</tr>
<tr>
<td><a href="ConstructGenomicEntityAssociation/">ConstructGenomicEntityAssociation</a></td>
<td>An association class to capture the identity of a construct component that is a known genomic entity with a curie and any accompanying meta data about the relationship between the construct and that genomic entity component</td>
</tr>
<tr>
<td><a href="ConstructGenomicEntityAssociationDTO/">ConstructGenomicEntityAssociationDTO</a></td>
<td>Ingest class for association between a construct component that is a known genomic entity with a curie and any accompanying meta data about the relationship between the construct and that genomic entity component</td>
</tr>
<tr>
<td><a href="ConstructSymbolSlotAnnotation/">ConstructSymbolSlotAnnotation</a></td>
<td>The one current symbol for the construct.</td>
</tr>
<tr>
<td><a href="ConstructSynonymSlotAnnotation/">ConstructSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the construct. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="CrossReference/">CrossReference</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CrossReferenceDTO/">CrossReferenceDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantGenomicLocationAssociation/">CuratedVariantGenomicLocationAssociation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantGenomicLocationAssociationDTO/">CuratedVariantGenomicLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantPolypeptideLocationAssociation/">CuratedVariantPolypeptideLocationAssociation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantPolypeptideLocationAssociationDTO/">CuratedVariantPolypeptideLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantTranscriptLocationAssociation/">CuratedVariantTranscriptLocationAssociation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CuratedVariantTranscriptLocationAssociationDTO/">CuratedVariantTranscriptLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="CurationReport/">CurationReport</a></td>
<td>Base class for all curation reports</td>
</tr>
<tr>
<td><a href="CurationReportGroup/">CurationReportGroup</a></td>
<td>This class is use to group together curation reports</td>
</tr>
<tr>
<td><a href="CurationReportHistory/">CurationReportHistory</a></td>
<td>Object used to describe the individual run of this curation report</td>
</tr>
<tr>
<td><a href="CurieObject/">CurieObject</a></td>
<td>Parent class for LinkML classes where all child classes have a curie (or will have one generated at the Alliance)</td>
</tr>
<tr>
<td><a href="DAOTerm/">DAOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="DataProviderDTO/">DataProviderDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="DiseaseAnnotation/">DiseaseAnnotation</a></td>
<td>An annotation asserting an association between a biological entity and a disease supported by evidence.</td>
</tr>
<tr>
<td><a href="DiseaseAnnotationDTO/">DiseaseAnnotationDTO</a></td>
<td>Ingest class for association between a biological entity and a disease</td>
</tr>
<tr>
<td><a href="DNAClone/">DNAClone</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="DOTerm/">DOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="DPOTerm/">DPOTerm</a></td>
<td>A phenotype ontology term from the Drosophila Phenotype Ontology (DPO). PURL: http://purl.obolibrary.org/obo/dpo.owl</td>
</tr>
<tr>
<td><a href="ECOTerm/">ECOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="EDAMTerm/">EDAMTerm</a></td>
<td>An ontology term from the EMBRACE Data and Methods Ontology (EDAM)</td>
</tr>
<tr>
<td><a href="EMAPATerm/">EMAPATerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="EvidenceAssociation/">EvidenceAssociation</a></td>
<td>An association supported by any number of InformationContentEntity objects</td>
</tr>
<tr>
<td><a href="EvidenceAssociationDTO/">EvidenceAssociationDTO</a></td>
<td>Ingest class for an association supported by any number of InformationContentEntity objects</td>
</tr>
<tr>
<td><a href="Exon/">Exon</a></td>
<td>A part of a gene that will form a part of the final mature RNA produced after introns have been removed by RNA splicing</td>
</tr>
<tr>
<td><a href="ExonDTO/">ExonDTO</a></td>
<td>Ingest class for exons</td>
</tr>
<tr>
<td><a href="ExonGenomicLocationAssociation/">ExonGenomicLocationAssociation</a></td>
<td>Location(s) of the exon on the genome.</td>
</tr>
<tr>
<td><a href="ExperimentalCondition/">ExperimentalCondition</a></td>
<td>The environmental context in which an experiment is carried out. This may (e.g. drug treatment) or may not (e.g. standard conditions) directly influence the outcome of the experiment.</td>
</tr>
<tr>
<td><a href="ExperimentalConditionDTO/">ExperimentalConditionDTO</a></td>
<td>Ingest class for describing the environmental context in which an experiment is carried out</td>
</tr>
<tr>
<td><a href="ExperimentalConditionOntologyTerm/">ExperimentalConditionOntologyTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ExpressionAnnotation/">ExpressionAnnotation</a></td>
<td>An annotation asserting an association between some entity and an expression pattern, supported by a single reference. Uniquely identifying aspects of this object comprise all slots (possibly excepting figure and image information).</td>
</tr>
<tr>
<td><a href="ExpressionAnnotationAssociation/">ExpressionAnnotationAssociation</a></td>
<td>Base class for all ExpressionAnnotation associations</td>
</tr>
<tr>
<td><a href="ExpressionAnnotationImagePaneAssociation/">ExpressionAnnotationImagePaneAssociation</a></td>
<td>An association between an ExpressionAnnotation and an image (in part or in full).</td>
</tr>
<tr>
<td><a href="ExpressionExperiment/">ExpressionExperiment</a></td>
<td>The experimental design used to determine the expression of some entity, defined by the entity_assayed, the specimen, the assay, the reagents (e.g., Antibody, Probe), the environmental conditions, and the reference. It groups ExpressionAnnotations; it may have its own notes.</td>
</tr>
<tr>
<td><a href="ExpressionPattern/">ExpressionPattern</a></td>
<td>An expression pattern, as defined by spatial and temporal attributes, including all qualifiers.</td>
</tr>
<tr>
<td><a href="ExternalDataBaseEntity/">ExternalDataBaseEntity</a></td>
<td>A local database entry representing an external database entry; used currently for datasets or dataset samples; htpId.json in agr_schemas json</td>
</tr>
<tr>
<td><a href="ExternalDataBaseEntityDTO/">ExternalDataBaseEntityDTO</a></td>
<td>DTO entity for a local database entry representing an external database entry; used currently for datasets or dataset samples; htpId.json in agr_schemas json</td>
</tr>
<tr>
<td><a href="ExternalDatabaseLink/">ExternalDatabaseLink</a></td>
<td>Base relation that holds the identifier prefix, base url and url suffix to help in generating URLs in crossReferences.</td>
</tr>
<tr>
<td><a href="ExternalDatabaseReference/">ExternalDatabaseReference</a></td>
<td>A reference to an entity in an external database</td>
</tr>
<tr>
<td><a href="FBCVTerm/">FBCVTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="FBDVTerm/">FBDVTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Figure/">Figure</a></td>
<td>An entity representing a figure or table in a publication.</td>
</tr>
<tr>
<td><a href="File/">File</a></td>
<td>A dummy object.</td>
</tr>
<tr>
<td><a href="FullNameSlotAnnotationDTO/">FullNameSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="FunctionalGeneSet/">FunctionalGeneSet</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Gene/">Gene</a></td>
<td>A DNA genomic entity from which one or more functional* RNA transcript molecules are transcribed, along with cis-regulatory elements responsible for regulating expression (transcription) of the gene. * A functional RNA molecule here can mean one that is directly responsible for the gene's function (e.g. catalysis, structure, etc.) or one that is translated to produce a functional polypeptide/protein. A pseudogene may be considered a gene under this definition, albeit no longer functional.</td>
</tr>
<tr>
<td><a href="GeneChangeEventSlotAnnotation/">GeneChangeEventSlotAnnotation</a></td>
<td>A major change to an entity: e.g., new, merge, split, rename, deprecation. Proposed event_type vocabulary below. new - The entity was created (excluding new entities from split/merges). merge - The entity subsumed other entities in a merge; objects subsumed in the merge get deprecation type ChangeEvent annotations. split - The entity was split from another entity; objects deprecated in the split get deprecation type ChangeEvent annotations. deprecation - The entity was made obsolete. This may include objects deprecated by some merge/split event. rename - A current accepted name of the entity was changed. type_change - The entity type was changed. Proposed status vocabulary: provisional/approved.</td>
</tr>
<tr>
<td><a href="GeneChromosomalLocationAssociation/">GeneChromosomalLocationAssociation</a></td>
<td>Localization of the gene to a chromosome, in some cases in reference to specific markers.</td>
</tr>
<tr>
<td><a href="GeneCluster/">GeneCluster</a></td>
<td>A gene cluster is a set of genes which have a biological significance, and which are probably clustered together on the genome, like histones and miRNAs</td>
</tr>
<tr>
<td><a href="GeneCollection/">GeneCollection</a></td>
<td>A gene collection is a set of genes which have been grouped based on experimental evidence, for example a set of interacting genes, genes in expression cluster, or a set of ChIP binding peaks</td>
</tr>
<tr>
<td><a href="GeneDiseaseAnnotation/">GeneDiseaseAnnotation</a></td>
<td>An annotation asserting an association between a gene and a disease supported by evidence.</td>
</tr>
<tr>
<td><a href="GeneDiseaseAnnotationDTO/">GeneDiseaseAnnotationDTO</a></td>
<td>Ingest class for an association between a gene and a disease</td>
</tr>
<tr>
<td><a href="GeneDTO/">GeneDTO</a></td>
<td>Ingest class for genes</td>
</tr>
<tr>
<td><a href="GeneExpressionAnnotation/">GeneExpressionAnnotation</a></td>
<td>An annotation describing an expression pattern for a gene.</td>
</tr>
<tr>
<td><a href="GeneExpressionExperiment/">GeneExpressionExperiment</a></td>
<td>An ExpressionExperiment for a Gene.</td>
</tr>
<tr>
<td><a href="GeneFullNameSlotAnnotation/">GeneFullNameSlotAnnotation</a></td>
<td>The one current full name for the gene.</td>
</tr>
<tr>
<td><a href="GeneGeneAssociation/">GeneGeneAssociation</a></td>
<td>abstract parent class for different kinds of gene-gene or gene product to gene product relationships. Includes homology and interaction.</td>
</tr>
<tr>
<td><a href="GeneGeneAssociationDTO/">GeneGeneAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GeneGeneticInteraction/">GeneGeneticInteraction</a></td>
<td>A genetic interaction between genes (e.g. phenotypic suppression)</td>
</tr>
<tr>
<td><a href="GeneGeneticInteractionDTO/">GeneGeneticInteractionDTO</a></td>
<td>Ingest class for genetic interaction between genes (e.g. phenotypic suppression)</td>
</tr>
<tr>
<td><a href="GeneGeneticMapAssociation/">GeneGeneticMapAssociation</a></td>
<td>The mapping of a gene along a chromosome using recombination frequency relative to other visible markers to calculate the genetic distance in centiMorgan (cM) units.</td>
</tr>
<tr>
<td><a href="GeneGenomicLocationAssociation/">GeneGenomicLocationAssociation</a></td>
<td>Location(s) of the gene on the genome.</td>
</tr>
<tr>
<td><a href="GeneInteraction/">GeneInteraction</a></td>
<td>An interaction between two genes or gene products; this may be a physical molecular interaction between gene products (e.g. protein-protein interactions), or may be a genetic interaction between genes (e.g. phenotypic suppression)</td>
</tr>
<tr>
<td><a href="GeneInteractionDTO/">GeneInteractionDTO</a></td>
<td>Ingest class for an interaction between two genes or gene products; this may be a physical molecular interaction between gene products (e.g. protein-protein interactions), or may be a genetic interaction between genes (e.g. phenotypic suppression)</td>
</tr>
<tr>
<td><a href="GeneMolecularInteraction/">GeneMolecularInteraction</a></td>
<td>A physical molecular interaction between gene products (e.g. protein-protein interactions or protein-RNA interactions) or between genes and DNA-binding factors (e.g. protein-DNA interactions)</td>
</tr>
<tr>
<td><a href="GeneMolecularInteractionDTO/">GeneMolecularInteractionDTO</a></td>
<td>Ingest class for a physical molecular interaction between gene products (e.g. protein-protein interactions or protein-RNA interactions) or between genes and DNA-binding factors (e.g. protein-DNA interactions)</td>
</tr>
<tr>
<td><a href="GeneNomenclatureSet/">GeneNomenclatureSet</a></td>
<td>WB specific. A gene class is a set of genes which share nomenclature, belonging to the same gene class.</td>
</tr>
<tr>
<td><a href="GeneOntologyAnnotation/">GeneOntologyAnnotation</a></td>
<td>An annotation associating a gene to a Gene Ontology (GO) term</td>
</tr>
<tr>
<td><a href="GeneOntologyAnnotationDTO/">GeneOntologyAnnotationDTO</a></td>
<td>An annotation associating a gene to a Gene Ontology (GO) term (DTO class)</td>
</tr>
<tr>
<td><a href="GenePathwayAssociation/">GenePathwayAssociation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GenePhenotypeAnnotation/">GenePhenotypeAnnotation</a></td>
<td>An annotation asserting an association between a gene and a phenotype supported by evidence.</td>
</tr>
<tr>
<td><a href="GenePhenotypeAnnotationDTO/">GenePhenotypeAnnotationDTO</a></td>
<td>Ingest class for an association between a gene and a phenotype</td>
</tr>
<tr>
<td><a href="GenerationMethod/">GenerationMethod</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GenerationMethodDTO/">GenerationMethodDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GeneSecondaryIdSlotAnnotation/">GeneSecondaryIdSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GeneSymbolSlotAnnotation/">GeneSymbolSlotAnnotation</a></td>
<td>The one current symbol for the gene.</td>
</tr>
<tr>
<td><a href="GeneSynonymSlotAnnotation/">GeneSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the gene. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="GeneSystematicNameSlotAnnotation/">GeneSystematicNameSlotAnnotation</a></td>
<td>The one current systematic name for the gene.</td>
</tr>
<tr>
<td><a href="GeneToGeneOrthology/">GeneToGeneOrthology</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GeneToGeneOrthologyCurated/">GeneToGeneOrthologyCurated</a></td>
<td>Class that holds the properties necessary to store a curated orthology record.</td>
</tr>
<tr>
<td><a href="GeneToGeneOrthologyGenerated/">GeneToGeneOrthologyGenerated</a></td>
<td>Class that holds the properties necessary to record an orthology record from DIOPT</td>
</tr>
<tr>
<td><a href="GeneToGeneParalogy/">GeneToGeneParalogy</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GenomeAssembly/">GenomeAssembly</a></td>
<td>A set of chromosome assemblies, scaffolds and contigs that represent the haploid genome sequence of a species or strain.</td>
</tr>
<tr>
<td><a href="GenomeAssemblyDTO/">GenomeAssemblyDTO</a></td>
<td>Ingest class for genome assemblies</td>
</tr>
<tr>
<td><a href="GenomicEntity/">GenomicEntity</a></td>
<td>An entity that is part of a genome (i.e. segment of the DNA molecule), is derived directly from the genome (i.e. RNA transcript molecule), or is derived indirectly from the genome (i.e. polypeptide or protein via RNA transcript translation).</td>
</tr>
<tr>
<td><a href="GenomicEntityDTO/">GenomicEntityDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="GENOTerm/">GENOTerm</a></td>
<td>An ontology term from the GENO ontology. PURL: http://purl.obolibrary.org/obo/geno.owl</td>
</tr>
<tr>
<td><a href="GOTerm/">GOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="HPTerm/">HPTerm</a></td>
<td>A phenotype ontology term from the Human Phenotype (HP) Ontology. PURL: http://purl.obolibrary.org/obo/hp.owl</td>
</tr>
<tr>
<td><a href="HTPExpressionDatasetAnnotation/">HTPExpressionDatasetAnnotation</a></td>
<td>An entry for a published high-throughput expression dataset annotation.</td>
</tr>
<tr>
<td><a href="HTPExpressionDatasetAnnotationDTO/">HTPExpressionDatasetAnnotationDTO</a></td>
<td>A DTO entry for a published high-throughput expression dataset annotation.</td>
</tr>
<tr>
<td><a href="HTPExpressionDatasetSampleAnnotation/">HTPExpressionDatasetSampleAnnotation</a></td>
<td>An entry with dataset sample object associations.</td>
</tr>
<tr>
<td><a href="HTPExpressionDatasetSampleAnnotationDTO/">HTPExpressionDatasetSampleAnnotationDTO</a></td>
<td>An entry with dataset sample object associations.</td>
</tr>
<tr>
<td><a href="Identifier/">Identifier</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="IdentifiersRange/">IdentifiersRange</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Image/">Image</a></td>
<td>The set of files and metadata that constitute an image.</td>
</tr>
<tr>
<td><a href="ImagePane/">ImagePane</a></td>
<td>Part of an Image that is relevant to some annotation. An annotation may reference many panes of an Image.</td>
</tr>
<tr>
<td><a href="InformationContentEntity/">InformationContentEntity</a></td>
<td>a piece of information that typically is used as support for an assertion or annotation.</td>
</tr>
<tr>
<td><a href="Ingest/">Ingest</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Laboratory/">Laboratory</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="LocationAssociation/">LocationAssociation</a></td>
<td>The mapping of a entity relative to another entity.</td>
</tr>
<tr>
<td><a href="LocationAssociationDTO/">LocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="LoggedInPerson/">LoggedInPerson</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="MATerm/">MATerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="MeshDetail/">MeshDetail</a></td>
<td>Medical Subject Headings information coming from PubMed.</td>
</tr>
<tr>
<td><a href="MicroarraySampleDetails/">MicroarraySampleDetails</a></td>
<td>An entry with microarray-specific metadata</td>
</tr>
<tr>
<td><a href="MITerm/">MITerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="MMOTerm/">MMOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="MMUSDVTerm/">MMUSDVTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ModCorpusAssociation/">ModCorpusAssociation</a></td>
<td>For a given reference and Mod, whether it is inside corpus, outside corpus, or needs review, as well as where this sorting came from.</td>
</tr>
<tr>
<td><a href="MODTerm/">MODTerm</a></td>
<td>An ontology term from the Protein Modification Ontology (MOD)</td>
</tr>
<tr>
<td><a href="Molecule/">Molecule</a></td>
<td>Molecules as described by WormBase</td>
</tr>
<tr>
<td><a href="MPATHTerm/">MPATHTerm</a></td>
<td>An ontology term from the Mouse Pathology Ontology (MPATH)</td>
</tr>
<tr>
<td><a href="MPTerm/">MPTerm</a></td>
<td>A phenotype ontology term from the Mammalian Phenotype (MP) Ontology. PURL: http://purl.obolibrary.org/obo/mp.owl</td>
</tr>
<tr>
<td><a href="NameSlotAnnotation/">NameSlotAnnotation</a></td>
<td>Some symbol or name for an object, including current names as well as aliases, with accompanying metadata. The entity to which the symbol/name applies is specified in objects that inherit from this object.</td>
</tr>
<tr>
<td><a href="NameSlotAnnotationDTO/">NameSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="NCBITaxonTerm/">NCBITaxonTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Note/">Note</a></td>
<td>Note object for capturing free-text describing some attribute of an entity, coupled with a 'note type', internal boolean, and an optional list of references. Permissible values for note_type can be viewed and managed in the A-Team curation UI Controlled Vocabulary Terms Table.</td>
</tr>
<tr>
<td><a href="NoteDTO/">NoteDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="OBITerm/">OBITerm</a></td>
<td>An ontology term from the Ontology for Biomedical Investigations (OBI). PURL: http://purl.obolibrary.org/obo/obi.owl</td>
</tr>
<tr>
<td><a href="OntologyTerm/">OntologyTerm</a></td>
<td>A concept or class in an ontology, vocabulary or thesaurus. Note, ontology terms can be found in multiple ontologies (ie: mireoting). defining_slots helps define an alternative key for ontology terms.</td>
</tr>
<tr>
<td><a href="OntologyTermClosure/">OntologyTermClosure</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Operon/">Operon</a></td>
<td>The DNA region of a group of adjacent genes whose transcription is coordinated on one or several mutually overlapping transcription units transcribed in the same direction and sharing at least one gene.</td>
</tr>
<tr>
<td><a href="Organization/">Organization</a></td>
<td>An organization that provides information and/or materials to the Alliance. This includes Alliance member organizations (see AllianceMember subclass).</td>
</tr>
<tr>
<td><a href="Pathway/">Pathway</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="PATOTerm/">PATOTerm</a></td>
<td>An ontology term from the Phenotype and Trait Ontology (PATO). PURL: http://purl.obolibrary.org/obo/PATO.owl</td>
</tr>
<tr>
<td><a href="Person/">Person</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="PersonalCommunication/">PersonalCommunication</a></td>
<td>a piece of information that is used to support an assertion or annotation, where the information comes from a person other than the author of the assertion or annotation, or the author of the reference.</td>
</tr>
<tr>
<td><a href="PhenotypeAnnotation/">PhenotypeAnnotation</a></td>
<td>An annotation asserting an association between a biological entity and a phenotype supported by evidence.</td>
</tr>
<tr>
<td><a href="PhenotypeAnnotationDTO/">PhenotypeAnnotationDTO</a></td>
<td>Ingest class for association between a biological entity and a phenotype</td>
</tr>
<tr>
<td><a href="PhenotypeTerm/">PhenotypeTerm</a></td>
<td>An ontology term representing a characteristic of an organism. This may or may not be expressed as a difference in comparison to a reference organism.</td>
</tr>
<tr>
<td><a href="PredictedVariantConsequence/">PredictedVariantConsequence</a></td>
<td>Class for predicted consequences and associated data from VEP analysis of a VariantGenomicLocationAssociation</td>
</tr>
<tr>
<td><a href="Protein/">Protein</a></td>
<td>A polypeptide of amino acids, often the result of protein translation of an RNA molecule (transcript) into its corresponding polypeptide by the ribosome complex</td>
</tr>
<tr>
<td><a href="ProteinComplex/">ProteinComplex</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ProteinDTO/">ProteinDTO</a></td>
<td>Ingest class for proteins</td>
</tr>
<tr>
<td><a href="PROTerm/">PROTerm</a></td>
<td>An ontology term from the Protein Ontology (PRO)</td>
</tr>
<tr>
<td><a href="PWTerm/">PWTerm</a></td>
<td>An ontology term from the Pathway Ontology (PW)</td>
</tr>
<tr>
<td><a href="Reagent/">Reagent</a></td>
<td>A material entity used in experiments.</td>
</tr>
<tr>
<td><a href="ReagentDTO/">ReagentDTO</a></td>
<td>Ingest class for material entities used in experiments.</td>
</tr>
<tr>
<td><a href="Reference/">Reference</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Resource/">Resource</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ResourceDescriptor/">ResourceDescriptor</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ResourceDescriptorPage/">ResourceDescriptorPage</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="RNAClone/">RNAClone</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ROTerm/">ROTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="RSTerm/">RSTerm</a></td>
<td>An ontology term from the Rat Strain Ontology (RS)</td>
</tr>
<tr>
<td><a href="SecondaryIdSlotAnnotation/">SecondaryIdSlotAnnotation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SecondaryIdSlotAnnotationDTO/">SecondaryIdSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SequenceTargetingReagent/">SequenceTargetingReagent</a></td>
<td></td>
</tr>
<tr>
<td><a href="SequenceTargetingReagentDTO/">SequenceTargetingReagentDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SingleReferenceAssociation/">SingleReferenceAssociation</a></td>
<td>An association supported by a single InformationContentEntity object</td>
</tr>
<tr>
<td><a href="SingleReferenceAssociationDTO/">SingleReferenceAssociationDTO</a></td>
<td>Ingest class for an association supported by a single InformationContentEntity object</td>
</tr>
<tr>
<td><a href="SlotAnnotation/">SlotAnnotation</a></td>
<td>SlotAnnotation classes should be used when we need to attach metadata (in particular evidence and provenance) to a slot in the context of its referencing class, that can not be fully captured using an Association between the full class itself, and an InformationContentEntity. Evidence and provenance can exist here in the form of an evidence code, a publication, a personal communication or any other kind of InformationContentEntity. SlotAnnotation classes are used where the slot is not referencing a class in and of itself, and often has a scalar range.</td>
</tr>
<tr>
<td><a href="SlotAnnotationDTO/">SlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SOTerm/">SOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SourceVariantGenomicLocationAssociation/">SourceVariantGenomicLocationAssociation</a></td>
<td>Links a paper to the variant genomic location described in that paper</td>
</tr>
<tr>
<td><a href="SourceVariantGenomicLocationAssociationDTO/">SourceVariantGenomicLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SourceVariantPolypeptideLocationAssociation/">SourceVariantPolypeptideLocationAssociation</a></td>
<td>Links a paper to the variant polypeptide location described in that paper</td>
</tr>
<tr>
<td><a href="SourceVariantPolypeptideLocationAssociationDTO/">SourceVariantPolypeptideLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SourceVariantTranscriptLocationAssociation/">SourceVariantTranscriptLocationAssociation</a></td>
<td>Links a paper to the variant transcript location described in that paper</td>
</tr>
<tr>
<td><a href="SourceVariantTranscriptLocationAssociationDTO/">SourceVariantTranscriptLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Species/">Species</a></td>
<td>This class is initially intended to capture information about Alliance "core" model species, canonical species for each MOD member. The information in each record pertains to logistics of loading, processing, and displaying data and rendering Alliance pages for related entities.</td>
</tr>
<tr>
<td><a href="StageTerm/">StageTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="StrAssociation/">StrAssociation</a></td>
<td>Base class for STR associations</td>
</tr>
<tr>
<td><a href="StrAssociationDTO/">StrAssociationDTO</a></td>
<td>Base class for STR association DTOs</td>
</tr>
<tr>
<td><a href="StrGeneAssociation/">StrGeneAssociation</a></td>
<td>The relationship between a Sequence Targeting Reagent and its targeted genes. The relation should be a VocabularyTerm with one of the following values - targets</td>
</tr>
<tr>
<td><a href="StrGeneAssociationDTO/">StrGeneAssociationDTO</a></td>
<td>The DTO class for the relationship between a Sequence Targeting Reagent and its targeted genes. The relation_name should be a VocabularyTerm with one of the following values - targets</td>
</tr>
<tr>
<td><a href="SubmittedObject/">SubmittedObject</a></td>
<td>Parent class for LinkML classes of submitted entities that may have a public or internal MOD ID/curie</td>
</tr>
<tr>
<td><a href="SubmittedObjectDTO/">SubmittedObjectDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SymbolSlotAnnotationDTO/">SymbolSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SystematicNameSlotAnnotationDTO/">SystematicNameSlotAnnotationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="TemporalContext/">TemporalContext</a></td>
<td>The developmental stage and/or age of the specimen in an annotation. The developmental_stage_stop slot is optional.</td>
</tr>
<tr>
<td><a href="TemporalContextDTO/">TemporalContextDTO</a></td>
<td>The developmental stage and/or age of the specimen in an annotation. The developmental_stage_stop slot is optional.</td>
</tr>
<tr>
<td><a href="Transcript/">Transcript</a></td>
<td>A contiguous RNA molecule representing a sequence of a genome, often the result of a DNA-dependent RNA polymerase gene transcription event</td>
</tr>
<tr>
<td><a href="TranscriptCodingSequenceAssociation/">TranscriptCodingSequenceAssociation</a></td>
<td>Association between a transcript a CDS of that transcript</td>
</tr>
<tr>
<td><a href="TranscriptDTO/">TranscriptDTO</a></td>
<td>Ingest class for transcripts</td>
</tr>
<tr>
<td><a href="TranscriptExonAssociation/">TranscriptExonAssociation</a></td>
<td>Association between a transcript and an exon of that transcript</td>
</tr>
<tr>
<td><a href="TranscriptGeneAssociation/">TranscriptGeneAssociation</a></td>
<td>Association between a transcript and the gene that encodes it</td>
</tr>
<tr>
<td><a href="TranscriptGenomicLocationAssociation/">TranscriptGenomicLocationAssociation</a></td>
<td>Location(s) of the transcript on the genome.</td>
</tr>
<tr>
<td><a href="TransgenicTool/">TransgenicTool</a></td>
<td>A Transgenic Tool is defined as a commonly used sequence that forms part of a Construct or Transgene, and which has useful properties that are exploited to study the biological function of another gene product or a biological process.</td>
</tr>
<tr>
<td><a href="TransgenicToolAssociation/">TransgenicToolAssociation</a></td>
<td>Base class for TransgenicTool associations</td>
</tr>
<tr>
<td><a href="TransgenicToolDTO/">TransgenicToolDTO</a></td>
<td>An ingest class for a Transgenic Tool; a Transgenic Tool is defined as a commonly used sequence that forms part of a Construct or Transgene, and which has useful properties that are exploited to study the biological function of another gene product or a biological process.</td>
</tr>
<tr>
<td><a href="TransgenicToolFullNameSlotAnnotation/">TransgenicToolFullNameSlotAnnotation</a></td>
<td>The one current full name for the TransgenicTool.</td>
</tr>
<tr>
<td><a href="TransgenicToolSymbolSlotAnnotation/">TransgenicToolSymbolSlotAnnotation</a></td>
<td>The one current symbol for the TransgenicTool.</td>
</tr>
<tr>
<td><a href="TransgenicToolSynonymSlotAnnotation/">TransgenicToolSynonymSlotAnnotation</a></td>
<td>All aliases (non-preferred names) for the TransgenicTool. Any type of synonym is acceptable.</td>
</tr>
<tr>
<td><a href="TransgenicToolTransgenicToolAssociation/">TransgenicToolTransgenicToolAssociation</a></td>
<td>Association (without evidence) between two compatible TransgenicTools. The associatied TransgenicTool is any transgenic tool that is compatible for use in combination with the tool, either as part of a multi-component experimental system (eg. binary or split systems), or because the tools form an enzyme/target site pair (eg. FLP/FRT).</td>
</tr>
<tr>
<td><a href="TransgenicToolTransgenicToolAssociationDTO/">TransgenicToolTransgenicToolAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="UBERONTerm/">UBERONTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="UniGeneSet/">UniGeneSet</a></td>
<td>XenBase-specific. A set of three genes from X. tropicalis and X. laevis (S and L forms) representing a single unigene.</td>
</tr>
<tr>
<td><a href="Variant/">Variant</a></td>
<td>A DNA, RNA or protein/polypeptide sequence that differs relative to a designated reference sequence.  The sequence occurs at a single position or in a range of contiguous nucleotides or amino acids.</td>
</tr>
<tr>
<td><a href="VariantDTO/">VariantDTO</a></td>
<td>Ingest class for a Variant object</td>
</tr>
<tr>
<td><a href="VariantGenomicLocationAssociation/">VariantGenomicLocationAssociation</a></td>
<td>Links a variant to a genomic position and the resulting consequence to the sequence and/or function. In practice, functional consequences for variants which overlap genes are not generally provided at the genome level but rather are calculated and annotated relative to a specific transcript or protein isoform.</td>
</tr>
<tr>
<td><a href="VariantGenomicLocationAssociationDTO/">VariantGenomicLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="VariantLocationAssociation/">VariantLocationAssociation</a></td>
<td>Base class linking a variant to a position on a genomic entity and the resulting consequence to the sequence and/or function of that genomic entity. Slots are provided for data taken from a source publication or data load and for data resulting from manual curation. Where the values are the same, the curator has confirmed the information from the source.  In other cases, the curator's analysis has resulted in different values, for instance, if the assembly is different, the source did not specify the transcript or protein isoform, the definition of the transcript or protein isoform used by the source has changed, or if there was an error in the source data.</td>
</tr>
<tr>
<td><a href="VariantLocationAssociationDTO/">VariantLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="VariantPolypeptideLocationAssociation/">VariantPolypeptideLocationAssociation</a></td>
<td>Links a variant to a position on a specified polypeptide and the resulting consequence to the sequence and/or function of that polypeptide.</td>
</tr>
<tr>
<td><a href="VariantPolypeptideLocationAssociationDTO/">VariantPolypeptideLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="VariantSourceGeneralConsequenceSlotAnnotation/">VariantSourceGeneralConsequenceSlotAnnotation</a></td>
<td>The consequence of the variant, as stated in the source reference when no transcript ID is provided.</td>
</tr>
<tr>
<td><a href="VariantSourceGeneralConsequenceSlotAnnotationDTO/">VariantSourceGeneralConsequenceSlotAnnotationDTO</a></td>
<td>The consequence of the variant, as stated in the source reference when no transcript ID is provided.</td>
</tr>
<tr>
<td><a href="VariantTranscriptLocationAssociation/">VariantTranscriptLocationAssociation</a></td>
<td>Links a variant to a position on a specified transcript and the resulting consequence to the sequence and/or function of that transcript.</td>
</tr>
<tr>
<td><a href="VariantTranscriptLocationAssociationDTO/">VariantTranscriptLocationAssociationDTO</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="Vocabulary/">Vocabulary</a></td>
<td>A set of VocabularyTerm objects.</td>
</tr>
<tr>
<td><a href="VocabularyTerm/">VocabularyTerm</a></td>
<td>A concept or class in a simple vocabulary.</td>
</tr>
<tr>
<td><a href="VocabularyTermSet/">VocabularyTermSet</a></td>
<td>A subset of terms from a Vocabulary that are valid for particular applications</td>
</tr>
<tr>
<td><a href="VTTerm/">VTTerm</a></td>
<td>An ontology term from the Vertebrate Trait Ontologt (VT). PURL: http://purl.obolibrary.org/obo/vt.owl</td>
</tr>
<tr>
<td><a href="WBBTTerm/">WBBTTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="WBLSTerm/">WBLSTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="WBPhenotypeTerm/">WBPhenotypeTerm</a></td>
<td>A phenotype ontology term from the (WormBase) C. elegans Phenotype Ontology (WBPhenotype). PURL: http://purl.obolibrary.org/obo/wbphenotype.owl</td>
</tr>
<tr>
<td><a href="XBATerm/">XBATerm</a></td>
<td>The Xenbase anatomy ontology XAO is home to Xenopus anatomy terms as well as Xenopus developmental life stage terms, differentiated by namespace. The anatomy term class in LinkML will be named 'XBATerm' for Xenbase Anatomy Term and life stage terms will be named 'XBSTerm' for Xenbase Stage Term.</td>
</tr>
<tr>
<td><a href="XBEDTerm/">XBEDTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="XBSTerm/">XBSTerm</a></td>
<td>The Xenbase anatomy ontology XAO is home to Xenopus anatomy terms as well as Xenopus developmental life stage terms, differentiated by namespace. The anatomy term class in LinkML will be named 'XBATerm' for Xenbase Anatomy Term and life stage terms will be named 'XBSTerm' for Xenbase Stage Term.</td>
</tr>
<tr>
<td><a href="XCOTerm/">XCOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="XPOTerm/">XPOTerm</a></td>
<td>A phenotype ontology term from the Xenopus Phenotype Ontology (XPO). PURL: http://purl.obolibrary.org/obo/xpo.owl</td>
</tr>
<tr>
<td><a href="XSMOTerm/">XSMOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ZECOTerm/">ZECOTerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ZFATerm/">ZFATerm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ZFSTerm/">ZFSTerm</a></td>
<td>None</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Property</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="abbreviation/">abbreviation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="aberration/">aberration</a></td>
<td>Associated deficiency (etc.) whose breakpoint causes the mutation</td>
</tr>
<tr>
<td><a href="abstract/">abstract</a></td>
<td>The author summary of the publication. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="abundance/">abundance</a></td>
<td>The free text describing the amount of sample used</td>
</tr>
<tr>
<td><a href="acquires_in_merge/">acquires_in_merge</a></td>
<td>For an entity that persists through a merge, the list of entities that it subsumes in a given NomenclatureEvent.</td>
</tr>
<tr>
<td><a href="affiliated_alliance_member/">affiliated_alliance_member</a></td>
<td>The Alliance Member the person is affiliated with</td>
</tr>
<tr>
<td><a href="age/">age</a></td>
<td>The age at which an annotated event was observed.</td>
</tr>
<tr>
<td><a href="aggregation_database/">aggregation_database</a></td>
<td>The database that collected the interaction annotation from another resource and aggregated it into a consolidated resource, e.g. IMEx.  An MITerm from the AGR aggregation database subset of MITerms.</td>
</tr>
<tr>
<td><a href="aggregation_database_curie/">aggregation_database_curie</a></td>
<td>The database that collected the interaction annotation from another resource and aggregated it into a consolidated resource, e.g. IMEx.  Value is the curie of an MITerm from the AGR aggregation database subset of MITerms.</td>
</tr>
<tr>
<td><a href="agm_agm_association_ingest_set/">agm_agm_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_agm_association_object/">agm_agm_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_allele_association_ingest_set/">agm_allele_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_allele_association_object/">agm_allele_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_association_subject/">agm_association_subject</a></td>
<td>Subject slot to be used for AGM associations</td>
</tr>
<tr>
<td><a href="agm_full_name/">agm_full_name</a></td>
<td>The one current full name for an AGM.</td>
</tr>
<tr>
<td><a href="agm_full_name_dto/">agm_full_name_dto</a></td>
<td>The one current full name for an AGM.</td>
</tr>
<tr>
<td><a href="agm_identifier/">agm_identifier</a></td>
<td>Curie or internal ID of AGM</td>
</tr>
<tr>
<td><a href="agm_ingest_set/">agm_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_object_identifier/">agm_object_identifier</a></td>
<td>Curie or internal ID of AGM that is the object of an association</td>
</tr>
<tr>
<td><a href="agm_secondary_id_dtos/">agm_secondary_id_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_secondary_ids/">agm_secondary_ids</a></td>
<td>Secondary IDs of a given AGM</td>
</tr>
<tr>
<td><a href="agm_str_association_ingest_set/">agm_str_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_str_association_object/">agm_str_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_str_associations/">agm_str_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_subject_identifier/">agm_subject_identifier</a></td>
<td>Curie or internal ID of AGM that is the subject of an association</td>
</tr>
<tr>
<td><a href="agm_synonym_dtos/">agm_synonym_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="agm_synonyms/">agm_synonyms</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_allele_association_object/">allele_allele_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_allele_associations/">allele_allele_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_association_subject/">allele_association_subject</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_cell_line_association_ingest_set/">allele_cell_line_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_cell_line_association_object/">allele_cell_line_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_cell_line_associations/">allele_cell_line_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_construct_association_ingest_set/">allele_construct_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_construct_association_object/">allele_construct_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_construct_associations/">allele_construct_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_database_status/">allele_database_status</a></td>
<td>Database status of a given allele</td>
</tr>
<tr>
<td><a href="allele_database_status_dto/">allele_database_status_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_full_name/">allele_full_name</a></td>
<td>The one current full name for an allele: e.g., Sternopleural.</td>
</tr>
<tr>
<td><a href="allele_full_name_dto/">allele_full_name_dto</a></td>
<td>The one current full name for an allele: e.g., Sternopleural.</td>
</tr>
<tr>
<td><a href="allele_functional_impact_dtos/">allele_functional_impact_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_functional_impacts/">allele_functional_impacts</a></td>
<td>Functional impacts of a given allele</td>
</tr>
<tr>
<td><a href="allele_gene_association_ingest_set/">allele_gene_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_gene_association_object/">allele_gene_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_gene_associations/">allele_gene_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_generation_method_association_ingest_set/">allele_generation_method_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_generation_method_association_object/">allele_generation_method_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_generation_method_associations/">allele_generation_method_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_germline_transmission_status/">allele_germline_transmission_status</a></td>
<td>Germline transmission status for a given allele</td>
</tr>
<tr>
<td><a href="allele_germline_transmission_status_dto/">allele_germline_transmission_status_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_identifier/">allele_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the allele</td>
</tr>
<tr>
<td><a href="allele_image_association_ingest_set/">allele_image_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_image_association_object/">allele_image_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_image_associations/">allele_image_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_ingest_set/">allele_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_inheritance_mode_dtos/">allele_inheritance_mode_dtos</a></td>
<td>One or more allele inheritance mode DTO objects to be submitted</td>
</tr>
<tr>
<td><a href="allele_inheritance_modes/">allele_inheritance_modes</a></td>
<td>Inheritance modes for an allele</td>
</tr>
<tr>
<td><a href="allele_mutation_type_dtos/">allele_mutation_type_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_mutation_types/">allele_mutation_types</a></td>
<td>Mutation types for a given allele</td>
</tr>
<tr>
<td><a href="allele_nomenclature_event_dtos/">allele_nomenclature_event_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_nomenclature_events/">allele_nomenclature_events</a></td>
<td>Nomenclature events of a given allele</td>
</tr>
<tr>
<td><a href="allele_origin_association_ingest_set/">allele_origin_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_origin_association_object/">allele_origin_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_origin_associations/">allele_origin_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_protein_association_ingest_set/">allele_protein_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_protein_association_object/">allele_protein_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_protein_associations/">allele_protein_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_secondary_id_dtos/">allele_secondary_id_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_secondary_ids/">allele_secondary_ids</a></td>
<td>Secondary IDs of a given allele</td>
</tr>
<tr>
<td><a href="allele_symbol/">allele_symbol</a></td>
<td>The one current accepted symbol for the allele: e.g., wg<sup>Sp-1</sup>.</td>
</tr>
<tr>
<td><a href="allele_symbol_dto/">allele_symbol_dto</a></td>
<td>The one current accepted symbol for the allele: e.g., wg<sup>Sp-1</sup>.</td>
</tr>
<tr>
<td><a href="allele_synonym_dtos/">allele_synonym_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_synonyms/">allele_synonyms</a></td>
<td>Holds between an Allele and a synonym.</td>
</tr>
<tr>
<td><a href="allele_transcript_association_ingest_set/">allele_transcript_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_transcript_association_object/">allele_transcript_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_transcript_associations/">allele_transcript_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_variant_association_ingest_set/">allele_variant_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_variant_association_object/">allele_variant_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="allele_variant_associations/">allele_variant_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="alliance_member_id/">alliance_member_id</a></td>
<td>An integer referring to an AllianceMember object in the AllianceMember/MOD database table. It's a primary key in the AllianceMember/MOD table, a foreign key if used in other tables.</td>
</tr>
<tr>
<td><a href="alliance_member_release_version/">alliance_member_release_version</a></td>
<td>Version of Alliance member/MOD release used for submitted file</td>
</tr>
<tr>
<td><a href="amino_acid_reference/">amino_acid_reference</a></td>
<td>reference genome amino acid sequence at variant position</td>
</tr>
<tr>
<td><a href="amino_acid_variant/">amino_acid_variant</a></td>
<td>variant amino acid sequence at variant position</td>
</tr>
<tr>
<td><a href="analyses/">analyses</a></td>
<td>TODO: added as a result of 'origin' class - please advise on a more descriptive slot/class definition.</td>
</tr>
<tr>
<td><a href="anatomical_structure/">anatomical_structure</a></td>
<td>The primary anatomical structure referred to within the context of an AnatomicalSite object.</td>
</tr>
<tr>
<td><a href="anatomical_structure_curie/">anatomical_structure_curie</a></td>
<td>The primary anatomical structure referred to within the context of an AnatomicalSite object. The range is any string representing an AnatomicalTerm ontology term</td>
</tr>
<tr>
<td><a href="anatomical_structure_qualifier_curies/">anatomical_structure_qualifier_curies</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="anatomical_structure_qualifiers/">anatomical_structure_qualifiers</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="anatomical_structure_uberon_term_curies/">anatomical_structure_uberon_term_curies</a></td>
<td>UBERON slim terms that apply to the anatomical structure.</td>
</tr>
<tr>
<td><a href="anatomical_structure_uberon_term_other/">anatomical_structure_uberon_term_other</a></td>
<td>Is set to true when other categorized uberon term is submitted in anatomical_structure_uberon_terms</td>
</tr>
<tr>
<td><a href="anatomical_structure_uberon_terms/">anatomical_structure_uberon_terms</a></td>
<td>UBERON slim terms that apply to the anatomical structure.</td>
</tr>
<tr>
<td><a href="anatomical_substructure/">anatomical_substructure</a></td>
<td>Primary anatomical substructure referenced within the context of an AnatomicalSite.</td>
</tr>
<tr>
<td><a href="anatomical_substructure_curie/">anatomical_substructure_curie</a></td>
<td>Primary anatomical substructure referenced within the context of an AnatomicalSiteDTO. The range is any string representing an AnatomicalTerm ontology term.</td>
</tr>
<tr>
<td><a href="anatomical_substructure_qualifier_curies/">anatomical_substructure_qualifier_curies</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="anatomical_substructure_qualifiers/">anatomical_substructure_qualifiers</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="anatomical_substructure_uberon_term_curies/">anatomical_substructure_uberon_term_curies</a></td>
<td>String representations of UBERON slim terms that apply to the anatomical substructure.</td>
</tr>
<tr>
<td><a href="anatomical_substructure_uberon_term_other/">anatomical_substructure_uberon_term_other</a></td>
<td>Is set to true when other categorized uberon term is submitted in anatomical_substructure_uberon_terms</td>
</tr>
<tr>
<td><a href="anatomical_substructure_uberon_terms/">anatomical_substructure_uberon_terms</a></td>
<td>UBERON slim terms that apply to the anatomical substructure.</td>
</tr>
<tr>
<td><a href="anatomy_function/">anatomy_function</a></td>
<td>WB specific. Allow the connection between Anatomy_term, Phenotype and gene Eg WBbtf0001</td>
</tr>
<tr>
<td><a href="ancestors/">ancestors</a></td>
<td>The ancestors of this term in the ontology, including the term itself. This language works well for the majority of use cases, however for a relationship like "develops_from", ancestors are the terms from which this term develops (not a true parent/child or ancestor/descendant relationship).</td>
</tr>
<tr>
<td><a href="annotation_type/">annotation_type</a></td>
<td>The type of annotation classified according to curation method. Submitted value should be a vocabulary term from the 'Annotation types' vocabulary</td>
</tr>
<tr>
<td><a href="annotation_type_name/">annotation_type_name</a></td>
<td>Name of the VocabularyTerm describing the annotation type selected from the 'Annotation types' Vocabulary</td>
</tr>
<tr>
<td><a href="antibody_ingest_set/">antibody_ingest_set</a></td>
<td>An ingest set for Antibody objects</td>
</tr>
<tr>
<td><a href="antibody_target_gene_identifiers/">antibody_target_gene_identifiers</a></td>
<td>Identifiers (curie/primary_external_id/mod_internal_id) of the genes whose gene products are recognized by the antibody.</td>
</tr>
<tr>
<td><a href="antibody_target_genes/">antibody_target_genes</a></td>
<td>The genes whose gene products are recognized by the antibody.</td>
</tr>
<tr>
<td><a href="antigen_taxon/">antigen_taxon</a></td>
<td>Holds between an Antibody and the species from which the antigen originates (as represented by taxon).</td>
</tr>
<tr>
<td><a href="antigen_taxon_curie/">antigen_taxon_curie</a></td>
<td>Holds between an Antibody and the species from which the antigen originates (as represented by taxon).</td>
</tr>
<tr>
<td><a href="api_token/">api_token</a></td>
<td>The API token of the LoggedInPerson for the curation system</td>
</tr>
<tr>
<td><a href="assembly_component_ingest_set/">assembly_component_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="assembly_versions/">assembly_versions</a></td>
<td>single or paired end sequencing; might be better in experiment info</td>
</tr>
<tr>
<td><a href="asserted_allele_identifiers/">asserted_allele_identifiers</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the allele to which something is manually asserted to be associated</td>
</tr>
<tr>
<td><a href="asserted_alleles/">asserted_alleles</a></td>
<td>The allele to which something is manually asserted to be associated.</td>
</tr>
<tr>
<td><a href="asserted_gene_identifiers/">asserted_gene_identifiers</a></td>
<td>Identifiers (curie/primary_external_id/mod_internal_id) of the gene(s) to which something is manually asserted to be associated</td>
</tr>
<tr>
<td><a href="asserted_genes/">asserted_genes</a></td>
<td>The gene(s) to which something is manually asserted to be associated.</td>
</tr>
<tr>
<td><a href="associated_transcript_identifiers/">associated_transcript_identifiers</a></td>
<td>Identifiers (curie/primary_external_id/mod_internal_id) of transcript(s) associated with polypeptide to which variant is aligned.</td>
</tr>
<tr>
<td><a href="associated_transcripts/">associated_transcripts</a></td>
<td>Transcript(s) associated with polypeptide to which variant is aligned.</td>
</tr>
<tr>
<td><a href="association_slot/">association_slot</a></td>
<td>any slot that relates an association to another entity</td>
</tr>
<tr>
<td><a href="authors/">authors</a></td>
<td>Ordered author entities for this publication.  An Author is associated with only one publication.  A Person can be associated with multiple publications.</td>
</tr>
<tr>
<td><a href="backend_bulk_load_type/">backend_bulk_load_type</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="backend_load_type/">backend_load_type</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="biosample_agm/">biosample_agm</a></td>
<td>An affected genomic model (AGM) from which a biosample has been derived</td>
</tr>
<tr>
<td><a href="biosample_agm_type/">biosample_agm_type</a></td>
<td>Indicates the type of AGM biosample; enum: fish, genotype, strain</td>
</tr>
<tr>
<td><a href="biosample_agm_type_name/">biosample_agm_type_name</a></td>
<td>Indicates the type of AGM biosample; enum: fish, genotype, strain</td>
</tr>
<tr>
<td><a href="biosample_allele/">biosample_allele</a></td>
<td>An allele of focus for a particular biosample</td>
</tr>
<tr>
<td><a href="biosample_text/">biosample_text</a></td>
<td>The free text of the biosample genomic information if no biosample is available</td>
</tr>
<tr>
<td><a href="birt_report_file_path/">birt_report_file_path</a></td>
<td>File path to where the BIRT file has been saved</td>
</tr>
<tr>
<td><a href="bulk_load/">bulk_load</a></td>
<td>A link back to the parent bulk load</td>
</tr>
<tr>
<td><a href="bulkload_group/">bulkload_group</a></td>
<td>Bulk load group designed to group together bulk loads</td>
</tr>
<tr>
<td><a href="bulkload_status/">bulkload_status</a></td>
<td>Status used to capture the progress of the load</td>
</tr>
<tr>
<td><a href="bulkload_url/">bulkload_url</a></td>
<td>The URL that this bulk load will download for ingest</td>
</tr>
<tr>
<td><a href="calculated_cdna_end/">calculated_cdna_end</a></td>
<td>end position of variation in cDNA coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="calculated_cdna_start/">calculated_cdna_start</a></td>
<td>start position of variation in cDNA coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="calculated_cds_end/">calculated_cds_end</a></td>
<td>end position of variation in CDS coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="calculated_cds_start/">calculated_cds_start</a></td>
<td>start position of variation in CDS coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="calculated_protein_end/">calculated_protein_end</a></td>
<td>end position of variation in protein amino acid coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="calculated_protein_start/">calculated_protein_start</a></td>
<td>start position of variation in protein amino acid coordinates as calculated by VEP from input VCF, GFF and BAM files</td>
</tr>
<tr>
<td><a href="caption/">caption</a></td>
<td>Text describing the contents of a figure or table in a publication.</td>
</tr>
<tr>
<td><a href="cassette_association_subject/">cassette_association_subject</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_component_dtos/">cassette_component_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_component_symbol/">cassette_component_symbol</a></td>
<td>The free text of a symbol or name of a cassette component for which no curie exists or is available</td>
</tr>
<tr>
<td><a href="cassette_components/">cassette_components</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_full_name/">cassette_full_name</a></td>
<td>Holds between a Cassette and a full name.</td>
</tr>
<tr>
<td><a href="cassette_full_name_dto/">cassette_full_name_dto</a></td>
<td>Holds between a Cassette and a full name.</td>
</tr>
<tr>
<td><a href="cassette_genomic_entity_association_ingest_set/">cassette_genomic_entity_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_genomic_entity_association_object/">cassette_genomic_entity_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_genomic_entity_associations/">cassette_genomic_entity_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_identifier/">cassette_identifier</a></td>
<td>Either the MOD curie or internal ID that was used in the submission of the associated cassette</td>
</tr>
<tr>
<td><a href="cassette_ingest_set/">cassette_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_str_association_ingest_set/">cassette_str_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_str_association_object/">cassette_str_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_str_associations/">cassette_str_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_symbol/">cassette_symbol</a></td>
<td>Holds between a Cassette and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="cassette_symbol_dto/">cassette_symbol_dto</a></td>
<td>Holds between a Cassette and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="cassette_synonym_dtos/">cassette_synonym_dtos</a></td>
<td>Holds between a Cassette and a list of synonyms.</td>
</tr>
<tr>
<td><a href="cassette_synonyms/">cassette_synonyms</a></td>
<td>Holds between a Cassette and a list of synonyms.</td>
</tr>
<tr>
<td><a href="cassette_transgenic_tool_association_ingest_set/">cassette_transgenic_tool_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_transgenic_tool_association_object/">cassette_transgenic_tool_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_transgenic_tool_associations/">cassette_transgenic_tool_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cassette_use_dtos/">cassette_use_dtos</a></td>
<td>Holds between a Cassette and a list of uses.</td>
</tr>
<tr>
<td><a href="cassette_uses/">cassette_uses</a></td>
<td>Holds between a Cassette and a list of uses.</td>
</tr>
<tr>
<td><a href="category/">category</a></td>
<td>The alliance category type.  Only relevant at Alliance.</td>
</tr>
<tr>
<td><a href="category_tag_names/">category_tag_names</a></td>
<td>A list of controlled vocabulary terms to describe the experiment</td>
</tr>
<tr>
<td><a href="category_tags/">category_tags</a></td>
<td>A list of controlled vocabulary terms to describe the experiment</td>
</tr>
<tr>
<td><a href="cds_association_subject/">cds_association_subject</a></td>
<td>Subject slot to be used for CDS associations</td>
</tr>
<tr>
<td><a href="cds_genomic_location_association_object/">cds_genomic_location_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cds_genomic_location_associations/">cds_genomic_location_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cds_type/">cds_type</a></td>
<td>SOTerm describing CDS type</td>
</tr>
<tr>
<td><a href="cds_type_curie/">cds_type_curie</a></td>
<td>Curie of SOTerm describing CDS type</td>
</tr>
<tr>
<td><a href="cell_line_identifier/">cell_line_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the cell line</td>
</tr>
<tr>
<td><a href="cellular_component/">cellular_component</a></td>
<td>The primary cellular component referred to within the context of an AnatomicalSite object.</td>
</tr>
<tr>
<td><a href="cellular_component_curie/">cellular_component_curie</a></td>
<td>The primary cellular component referred to within the context of an AnatomicalSite object. The range is any string representing a GOTerm ontology term</td>
</tr>
<tr>
<td><a href="cellular_component_other/">cellular_component_other</a></td>
<td>Is set to true when no cellular_component_ribbon_term is available</td>
</tr>
<tr>
<td><a href="cellular_component_qualifier_curies/">cellular_component_qualifier_curies</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cellular_component_qualifiers/">cellular_component_qualifiers</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cellular_component_ribbon_term/">cellular_component_ribbon_term</a></td>
<td>The primary cellular component slim term referred to within the context of an AnatomicalSite object and used by the expression ribbon.</td>
</tr>
<tr>
<td><a href="channel_id/">channel_id</a></td>
<td>for microarray experiments, channel ID</td>
</tr>
<tr>
<td><a href="channel_number/">channel_number</a></td>
<td>for micorarray data, which channel this sample is in, 1 or 2</td>
</tr>
<tr>
<td><a href="chemical_mutagen/">chemical_mutagen</a></td>
<td>The chemical used to generate the mutation through mutagenesis</td>
</tr>
<tr>
<td><a href="chemical_mutagen_name/">chemical_mutagen_name</a></td>
<td>The name of the chemical used to generate the mutation through mutagenesis</td>
</tr>
<tr>
<td><a href="child_count/">child_count</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="chromosome_curie/">chromosome_curie</a></td>
<td>The NCBI chromosome accession number for the chromosome.</td>
</tr>
<tr>
<td><a href="chromosome_ingest_set/">chromosome_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="clonality/">clonality</a></td>
<td>The clonality of the antibody: e.g., monoclonal.</td>
</tr>
<tr>
<td><a href="clonality_name/">clonality_name</a></td>
<td>Name of the VocabularyTerm representing the clonality of the antibody: e.g., monoclonal. From the Antibody Clonality CV.</td>
</tr>
<tr>
<td><a href="codon_reference/">codon_reference</a></td>
<td>reference sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cTa)</td>
</tr>
<tr>
<td><a href="codon_variant/">codon_variant</a></td>
<td>variant sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cAa)</td>
</tr>
<tr>
<td><a href="common_names/">common_names</a></td>
<td>The list of common names used for a species</td>
</tr>
<tr>
<td><a href="completed_records/">completed_records</a></td>
<td>The number of completed records</td>
</tr>
<tr>
<td><a href="component_symbol/">component_symbol</a></td>
<td>The free text of a symbol or name of a construct component for which no curie exists or is available</td>
</tr>
<tr>
<td><a href="component_type_curies/">component_type_curies</a></td>
<td>Curies of SOTerms representing the nature of Construct Cassette components</td>
</tr>
<tr>
<td><a href="component_types/">component_types</a></td>
<td>The list of SO Terms to describe the nature of the Construct Cassette component, as used by FlyBase et al.</td>
</tr>
<tr>
<td><a href="components/">components</a></td>
<td>Collection of genomic components that make up a model, i.e. 'allele', each with a zygosity</td>
</tr>
<tr>
<td><a href="condition_anatomy/">condition_anatomy</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_anatomy_curie/">condition_anatomy_curie</a></td>
<td>Curie of AnatomicalTerm associated with condition</td>
</tr>
<tr>
<td><a href="condition_chemical/">condition_chemical</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_chemical_curie/">condition_chemical_curie</a></td>
<td>Curie of ChemicalTerm associated with condition</td>
</tr>
<tr>
<td><a href="condition_class/">condition_class</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_class_curie/">condition_class_curie</a></td>
<td>Curie of ZECOTerm describing condition class</td>
</tr>
<tr>
<td><a href="condition_dtos/">condition_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_free_text/">condition_free_text</a></td>
<td>Free-text description of the experimental condition</td>
</tr>
<tr>
<td><a href="condition_gene_ontology/">condition_gene_ontology</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_gene_ontology_curie/">condition_gene_ontology_curie</a></td>
<td>Curie of GOTerm associated with condition</td>
</tr>
<tr>
<td><a href="condition_id/">condition_id</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_id_curie/">condition_id_curie</a></td>
<td>Curie of ExperimentalConditionOntologyTerm describing condition</td>
</tr>
<tr>
<td><a href="condition_quantity/">condition_quantity</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_relation_dtos/">condition_relation_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_relation_type/">condition_relation_type</a></td>
<td>Submitted value should be a vocabulary term from the 'Condition relation types' vocabulary</td>
</tr>
<tr>
<td><a href="condition_relation_type_name/">condition_relation_type_name</a></td>
<td>Name of VocabularyTerm from 'Condition relation types' vocabulary</td>
</tr>
<tr>
<td><a href="condition_relations/">condition_relations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_summary/">condition_summary</a></td>
<td>A human-readable summary of the experimental condition accounting for all ontology terms and free-text components of the ExperimentalCondition object. Not generated or submitted by DQMs, but generated at the Alliance after ingest only (or generated/updated in the curation interface).</td>
</tr>
<tr>
<td><a href="condition_taxon/">condition_taxon</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="condition_taxon_curie/">condition_taxon_curie</a></td>
<td>Curie of NCBITaxonTerm associated with condition</td>
</tr>
<tr>
<td><a href="conditions/">conditions</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="consequence/">consequence</a></td>
<td>SOTerm (child of SO:0001576 - transcript_variant) that describes the consequence of the variant, as stated in the source reference. In practice source consequence will be associated with locations at any or all of VariantGenomicLocationAssociation, VariantTranscriptLocationAssociation, and VariantPolypeptideLocationAssociation.</td>
</tr>
<tr>
<td><a href="consequence_curie/">consequence_curie</a></td>
<td>Curie of SOTerm (child of SO:0001576 - transcript_variant) that describes the consequence of the variant, as stated in the source reference. In practice source consequence will be associated with locations at any or all of VariantLocationAssociation, VariantTranscriptLocationAssociation, and VariantPolypeptideLocationAssociation.</td>
</tr>
<tr>
<td><a href="construct_association_subject/">construct_association_subject</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_cassette_association_ingest_set/">construct_cassette_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_cassette_association_object/">construct_cassette_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_cassette_associations/">construct_cassette_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_component_dtos/">construct_component_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_components/">construct_components</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_full_name/">construct_full_name</a></td>
<td>Holds between a Construct and a full name.</td>
</tr>
<tr>
<td><a href="construct_full_name_dto/">construct_full_name_dto</a></td>
<td>Holds between a Construct and a full name.</td>
</tr>
<tr>
<td><a href="construct_genomic_entity_association_ingest_set/">construct_genomic_entity_association_ingest_set</a></td>
<td>An ingest set for ConstructGenomicEntityAssociationDTO</td>
</tr>
<tr>
<td><a href="construct_genomic_entity_association_object/">construct_genomic_entity_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_genomic_entity_associations/">construct_genomic_entity_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="construct_identifier/">construct_identifier</a></td>
<td>Either the MOD curie or internal ID that was used in the submission of the associated construct</td>
</tr>
<tr>
<td><a href="construct_ingest_set/">construct_ingest_set</a></td>
<td>An ingest set for Constructs</td>
</tr>
<tr>
<td><a href="construct_symbol/">construct_symbol</a></td>
<td>Holds between a Construct and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="construct_symbol_dto/">construct_symbol_dto</a></td>
<td>Holds between a Construct and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="construct_synonym_dtos/">construct_synonym_dtos</a></td>
<td>Holds between a Construct and a list of synonyms.</td>
</tr>
<tr>
<td><a href="construct_synonyms/">construct_synonyms</a></td>
<td>Holds between a Construct and a list of synonyms.</td>
</tr>
<tr>
<td><a href="contig/">contig</a></td>
<td>Contig or clone this gene is located to</td>
</tr>
<tr>
<td><a href="copyright_date/">copyright_date</a></td>
<td></td>
</tr>
<tr>
<td><a href="corpus/">corpus</a></td>
<td>in-out-review corpus status of the publication at the mod. null means needs_review, false means outside_corpus,  true means inside_corpus.</td>
</tr>
<tr>
<td><a href="corresponding_author/">corresponding_author</a></td>
<td>Indicates if the author is a corresponding author.</td>
</tr>
<tr>
<td><a href="counter/">counter</a></td>
<td>a number to identify an alliance resource in a subdomain</td>
</tr>
<tr>
<td><a href="created_by/">created_by</a></td>
<td>The individual that created the entity.</td>
</tr>
<tr>
<td><a href="created_by_curie/">created_by_curie</a></td>
<td>Curie of the Person object representing the individual that created the entity</td>
</tr>
<tr>
<td><a href="cron_schedule/">cron_schedule</a></td>
<td>A string describing the cron syntax for the schedule</td>
</tr>
<tr>
<td><a href="cropped_from/">cropped_from</a></td>
<td>Another larger image from which this image was cropped.</td>
</tr>
<tr>
<td><a href="cross_reference/">cross_reference</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cross_reference_dto/">cross_reference_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cross_reference_dtos/">cross_reference_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="cross_references/">cross_references</a></td>
<td>Holds between an object and its CrossReferences.</td>
</tr>
<tr>
<td><a href="curated_consequence/">curated_consequence</a></td>
<td>SOTerm that describes the consequence of the variant, as determined by the curator. In practice curated consequence will be associated with locations at any or all of VariantGenomicLocationAssociation, VariantTranscriptLocationAssociation, and VariantPolypeptideLocationAssociation.</td>
</tr>
<tr>
<td><a href="curated_consequence_curie/">curated_consequence_curie</a></td>
<td>Curie of SOTerm that describes the consequence of the variant, as determined by the curator. In practice curated consequence will be associated with locations at any or all of VariantGenomicLocationAssociation, VariantTranscriptLocationAssociation, and VariantPolypeptideLocationAssociation.</td>
</tr>
<tr>
<td><a href="curated_variant_genomic_locations/">curated_variant_genomic_locations</a></td>
<td>Location of the variant in a genomic context.  multiple variant_genome_locations represent multiple assemblies.  because we want the variant to stay consistent between assemblies, we must have a single  variant object apart from its location.</td>
</tr>
<tr>
<td><a href="curated_variant_polypeptide_locations/">curated_variant_polypeptide_locations</a></td>
<td>Location of the variant within polypeptide entities.</td>
</tr>
<tr>
<td><a href="curated_variant_transcript_locations/">curated_variant_transcript_locations</a></td>
<td>Location of the variant within transcript entities.</td>
</tr>
<tr>
<td><a href="curation_report/">curation_report</a></td>
<td>Links a report history to its associated report</td>
</tr>
<tr>
<td><a href="curation_report_group/">curation_report_group</a></td>
<td>Links a curation report to its report group</td>
</tr>
<tr>
<td><a href="curation_report_status/">curation_report_status</a></td>
<td>Describes the status of the curation report</td>
</tr>
<tr>
<td><a href="curation_report_timestamp/">curation_report_timestamp</a></td>
<td>Timestamp for creation of BIRT report file</td>
</tr>
<tr>
<td><a href="curator_comment/">curator_comment</a></td>
<td>A publicly visible explanatory note from curators about some entity.</td>
</tr>
<tr>
<td><a href="curie/">curie</a></td>
<td>A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI</td>
</tr>
<tr>
<td><a href="current/">current</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="current_version/">current_version</a></td>
<td>The current version of the entity, generated by this nomenclature event. It starts at one, and is incremented by one with every major change.</td>
</tr>
<tr>
<td><a href="data_provider/">data_provider</a></td>
<td>The organization (e.g. MOD) from which the data was sourced</td>
</tr>
<tr>
<td><a href="data_provider_cross_reference/">data_provider_cross_reference</a></td>
<td>CrossReference to the organization from which the data was sourced</td>
</tr>
<tr>
<td><a href="data_provider_dto/">data_provider_dto</a></td>
<td>Ingest object representing the organization (e.g. MOD) from which the data was sourced and a CrossReference to that organisation's site</td>
</tr>
<tr>
<td><a href="database_status/">database_status</a></td>
<td>Database status of the allele</td>
</tr>
<tr>
<td><a href="database_status_name/">database_status_name</a></td>
<td>Name of the VocabularyTerm describing the database status</td>
</tr>
<tr>
<td><a href="dataset_ids/">dataset_ids</a></td>
<td>list of primary dataset ids that a sample is associated with</td>
</tr>
<tr>
<td><a href="date_arrived_in_pubmed/">date_arrived_in_pubmed</a></td>
<td>Day in which a reference or resource was created in PubMed. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="date_created/">date_created</a></td>
<td>The date on which an entity was created. This can be applied to nodes or edges.</td>
</tr>
<tr>
<td><a href="date_last_loaded/">date_last_loaded</a></td>
<td>Indicates when the file was last loaded</td>
</tr>
<tr>
<td><a href="date_last_modified_in_pubmed/">date_last_modified_in_pubmed</a></td>
<td>Date on which entity was last updated at PubMed. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="date_published/">date_published</a></td>
<td>Date on which an entity was published.  From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="date_updated/">date_updated</a></td>
<td>Date on which an entity was last modified.</td>
</tr>
<tr>
<td><a href="db_date_created/">db_date_created</a></td>
<td>The date on which an entity was created in the Alliance database.  This is distinct from date_created, which represents the date when the entity was originally created (i.e. at the MOD for imported data).</td>
</tr>
<tr>
<td><a href="db_date_updated/">db_date_updated</a></td>
<td>Date on which an entity was last modified in the Alliance database.  This is distinct from date_updated, which represents the date when the entity was last modified and may predate import into the Alliance database.</td>
</tr>
<tr>
<td><a href="dbkey/">dbkey</a></td>
<td>Typically the primary key on the table.  Should be a global sequence in the database to insure uniqueness over the entire suite of tables.  Alternatively, could be a serial8 identifier. Tables with a dbkey should have an alternate key to establish uniqueness based on the data in the table.</td>
</tr>
<tr>
<td><a href="default_url_template/">default_url_template</a></td>
<td>Default URL template for the resource</td>
</tr>
<tr>
<td><a href="definition/">definition</a></td>
<td>The explanation of the meaning of a term.</td>
</tr>
<tr>
<td><a href="definition_urls/">definition_urls</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="deleted_sequence/">deleted_sequence</a></td>
<td>The sequence of DNA base pairs removed from the reference sequence by the variant.</td>
</tr>
<tr>
<td><a href="descendant_count/">descendant_count</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="descendants/">descendants</a></td>
<td>The descendants of this term in the ontology, including the term itself. This language works well for the majority of use cases, however for a relationship like "develops_from", ancestors are the terms to which this term develops into (not a true parent/child or ancestor/descendant relationship).</td>
</tr>
<tr>
<td><a href="description/">description</a></td>
<td>a human-readable description of an entity</td>
</tr>
<tr>
<td><a href="designating_laboratories/">designating_laboratories</a></td>
<td>A laboratory, rarely laboratories, which designated this gene</td>
</tr>
<tr>
<td><a href="designating_laboratory/">designating_laboratory</a></td>
<td>A laboratory which designated this gene class</td>
</tr>
<tr>
<td><a href="designating_persons/">designating_persons</a></td>
<td>A person/persons who designated this gene</td>
</tr>
<tr>
<td><a href="detection_method/">detection_method</a></td>
<td>The experimental method used to identify the physical interaction between two molecules. An MITerm from the AGR detection method subset of MITerms.</td>
</tr>
<tr>
<td><a href="detection_method_curie/">detection_method_curie</a></td>
<td>The experimental method used to identify the physical interaction between two molecules. Value is the curie of an MITerm from the AGR detection method subset of MITerms.</td>
</tr>
<tr>
<td><a href="detection_reagents/">detection_reagents</a></td>
<td>Reagents used in the expression assay: e.g., construct, antibodies, probes.</td>
</tr>
<tr>
<td><a href="developmental_stage_start/">developmental_stage_start</a></td>
<td>The beginning developmental stage at which an annotated event was observed.</td>
</tr>
<tr>
<td><a href="developmental_stage_start_curie/">developmental_stage_start_curie</a></td>
<td>The beginning developmental stage at which an annotated event was observed.</td>
</tr>
<tr>
<td><a href="developmental_stage_stop/">developmental_stage_stop</a></td>
<td>The end developmental stage at which an annotated event was observed. This is not required if the annotation applies to a single stage.</td>
</tr>
<tr>
<td><a href="developmental_stage_stop_curie/">developmental_stage_stop_curie</a></td>
<td>The end developmental stage at which an annotated event was observed. This is not required if the annotation applies to a single stage.</td>
</tr>
<tr>
<td><a href="disease_agm_ingest_set/">disease_agm_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="disease_allele_ingest_set/">disease_allele_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="disease_annotation_object/">disease_annotation_object</a></td>
<td>The Disease Ontology term</td>
</tr>
<tr>
<td><a href="disease_annotation_subject/">disease_annotation_subject</a></td>
<td>The biological entity to which the disease ontology term is associated.</td>
</tr>
<tr>
<td><a href="disease_gene_ingest_set/">disease_gene_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_agms/">disease_genetic_modifier_agms</a></td>
<td>Specifies an affected genomic model (e.g. genotype) that modifies a disease model.</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_alleles/">disease_genetic_modifier_alleles</a></td>
<td>Specifies an allele that modifies a disease model.</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_genes/">disease_genetic_modifier_genes</a></td>
<td>Specifies a gene that modifies a disease model.</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_identifiers/">disease_genetic_modifier_identifiers</a></td>
<td>Identifiers (curie/primary_external_id/mod_internal_id) of Gene, Allele or AGM that modifies the disease model</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_relation/">disease_genetic_modifier_relation</a></td>
<td>A relation describing how the genetic modifier modifies the disease model. Submitted value should be a vocabulary term from the 'Disease genetic modifiers' vocabulary</td>
</tr>
<tr>
<td><a href="disease_genetic_modifier_relation_name/">disease_genetic_modifier_relation_name</a></td>
<td>Name of the VocabularyTerm that describes how the genetic modifier modifies the disease model, selected from the 'Disease genetic modifier relations' Vocabulary.</td>
</tr>
<tr>
<td><a href="disease_qualifier_names/">disease_qualifier_names</a></td>
<td>Names of terms from the 'Disease qualifiers' vocabulary</td>
</tr>
<tr>
<td><a href="disease_qualifiers/">disease_qualifiers</a></td>
<td>Submitted values should be vocabulary terms from the 'Disease qualifiers' Vocabulary</td>
</tr>
<tr>
<td><a href="disease_relation_name/">disease_relation_name</a></td>
<td>Name of term from 'Disease Relation Vocabulary' vocabulary</td>
</tr>
<tr>
<td><a href="display_name/">display_name</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="display_synonym/">display_synonym</a></td>
<td>Placeholder.</td>
</tr>
<tr>
<td><a href="display_text/">display_text</a></td>
<td>A version of a synonym string for display. Any UTF8 character is permitted.</td>
</tr>
<tr>
<td><a href="distance_between/">distance_between</a></td>
<td>distance_between is zero for reflexive–transitive closure each node has an ancestor or descendant of itself</td>
</tr>
<tr>
<td><a href="dna_mutation_type/">dna_mutation_type</a></td>
<td>The type of DNA mutation, for example, insertion, deletion, substitution, or indel.</td>
</tr>
<tr>
<td><a href="dna_mutation_type_curie/">dna_mutation_type_curie</a></td>
<td>Curie of SOTerm describing the type of DNA mutation, for example, insertion, deletion, substitution, or indel.</td>
</tr>
<tr>
<td><a href="do_term_curie/">do_term_curie</a></td>
<td>Curie of DOTerm describing the disease</td>
</tr>
<tr>
<td><a href="editors/">editors</a></td>
<td>holds between a resource and a editor_resource</td>
</tr>
<tr>
<td><a href="emails/">emails</a></td>
<td>list of emails for a person</td>
</tr>
<tr>
<td><a href="enables/">enables</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="end/">end</a></td>
<td>The end of the feature in positive 1-based integer coordinates relative to the reference landmark.</td>
</tr>
<tr>
<td><a href="entity_assayed/">entity_assayed</a></td>
<td>The subject of the ExpressionExperiment. This will usually be a Gene, but may be a sequence feature, or a placeholder if the exact subject of the expression pattern is unknown (e.g., unknown epitope of an antibody).</td>
</tr>
<tr>
<td><a href="error_message/">error_message</a></td>
<td>Error message string if an error occurs</td>
</tr>
<tr>
<td><a href="event_status/">event_status</a></td>
<td>Status for an event change: provisional or approved.</td>
</tr>
<tr>
<td><a href="event_type/">event_type</a></td>
<td>Category of major change event for an entity.</td>
</tr>
<tr>
<td><a href="evidence/">evidence</a></td>
<td>The evidence that supports some assertion.</td>
</tr>
<tr>
<td><a href="evidence_code/">evidence_code</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="evidence_code_curie/">evidence_code_curie</a></td>
<td>Curie of ECOTerm</td>
</tr>
<tr>
<td><a href="evidence_code_curies/">evidence_code_curies</a></td>
<td>List of ECOTerm curies</td>
</tr>
<tr>
<td><a href="evidence_codes/">evidence_codes</a></td>
<td>ECO term IDs</td>
</tr>
<tr>
<td><a href="evidence_curie/">evidence_curie</a></td>
<td>Curie of InformationContentEntity object given as evidence</td>
</tr>
<tr>
<td><a href="evidence_curies/">evidence_curies</a></td>
<td>Curies of InformationContentEntity objects given as evidence</td>
</tr>
<tr>
<td><a href="evidence_item/">evidence_item</a></td>
<td>Single item of evidence that supports some assertion.</td>
</tr>
<tr>
<td><a href="exon_association_subject/">exon_association_subject</a></td>
<td>Subject slot to be used for exon associations</td>
</tr>
<tr>
<td><a href="exon_genomic_location_association_object/">exon_genomic_location_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="exon_genomic_location_associations/">exon_genomic_location_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="exon_number/">exon_number</a></td>
<td>The exon number of the variant location.  If the variant is in an intron, this slot is not populated. Primarily used for zebrafish.</td>
</tr>
<tr>
<td><a href="exon_type/">exon_type</a></td>
<td>SOTerm describing exon type</td>
</tr>
<tr>
<td><a href="exon_type_curie/">exon_type_curie</a></td>
<td>Curie of SOTerm describing exon type</td>
</tr>
<tr>
<td><a href="experiment_type/">experiment_type</a></td>
<td>Type of experiment by which these genes were collated eg chip-seq, interaction, expression</td>
</tr>
<tr>
<td><a href="expression_annotation_image_associations/">expression_annotation_image_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="expression_annotation_image_pane_association_object/">expression_annotation_image_pane_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="expression_annotation_subject/">expression_annotation_subject</a></td>
<td>The entity for which an expression pattern is being described; the exact type will be constrained by the type of ExpressionAnnotation.</td>
</tr>
<tr>
<td><a href="expression_annotations/">expression_annotations</a></td>
<td>The list of ExpressionAnnotations for a given ExpressionExperiment.</td>
</tr>
<tr>
<td><a href="expression_assay_curie/">expression_assay_curie</a></td>
<td>The assay used to experimentally determine gene expression.</td>
</tr>
<tr>
<td><a href="expression_assay_used/">expression_assay_used</a></td>
<td>The assay used to experimentally determine gene expression.</td>
</tr>
<tr>
<td><a href="expression_pattern/">expression_pattern</a></td>
<td>The spatiotemporal expression pattern of the expression annotation subject.</td>
</tr>
<tr>
<td><a href="expression_qualifiers/">expression_qualifiers</a></td>
<td>Qualifiers that describe additional characteristics of gene expression. For example: faint, intense, restricted.</td>
</tr>
<tr>
<td><a href="failed_records/">failed_records</a></td>
<td>The number of failed records</td>
</tr>
<tr>
<td><a href="file_size/">file_size</a></td>
<td>The size of the file</td>
</tr>
<tr>
<td><a href="first/">first</a></td>
<td>first identifier in a range</td>
</tr>
<tr>
<td><a href="first_author/">first_author</a></td>
<td>Indicates if the author is a first author.</td>
</tr>
<tr>
<td><a href="first_name/">first_name</a></td>
<td>first name of a person</td>
</tr>
<tr>
<td><a href="fms_data_sub_type/">fms_data_sub_type</a></td>
<td>The dataSubType paramater in the FMS</td>
</tr>
<tr>
<td><a href="fms_data_type/">fms_data_type</a></td>
<td>The dataType paramater in the FMS</td>
</tr>
<tr>
<td><a href="format_text/">format_text</a></td>
<td>A version of a synonym string using only ASCII characters, which is easier to type (for searches), print and parse. For example, Greek characters are transliterated.</td>
</tr>
<tr>
<td><a href="formula/">formula</a></td>
<td>Formula of the molecule</td>
</tr>
<tr>
<td><a href="free_text/">free_text</a></td>
<td>A free text string that describes some aspect of an entity.</td>
</tr>
<tr>
<td><a href="from_image/">from_image</a></td>
<td>Holds between an ImagePane and an Image.</td>
</tr>
<tr>
<td><a href="full_name/">full_name</a></td>
<td>The full name of the entity</td>
</tr>
<tr>
<td><a href="full_name_renamed_from/">full_name_renamed_from</a></td>
<td>The previous full_name for a renamed entity.</td>
</tr>
<tr>
<td><a href="full_name_renamed_to/">full_name_renamed_to</a></td>
<td>The new full_name for a renamed entity.</td>
</tr>
<tr>
<td><a href="functional_impact_names/">functional_impact_names</a></td>
<td>Name of the VocabularyTerm describing the experimentally assessed functional impact of the allele, e.g. knockout / amorphic</td>
</tr>
<tr>
<td><a href="functional_impacts/">functional_impacts</a></td>
<td>Experimentally assessed functional impact of the allele, e.g. knockout / amorphic</td>
</tr>
<tr>
<td><a href="gcrp_cross_reference/">gcrp_cross_reference</a></td>
<td>The single UniProtKB Gene-Centric Reference Proteome (GCRP) UniProtKB identifier and cross reference for the gene</td>
</tr>
<tr>
<td><a href="gcrp_cross_reference_dto/">gcrp_cross_reference_dto</a></td>
<td>The single UniProtKB Gene-Centric Reference Proteome (GCRP) UniProtKB identifier and cross reference for the gene, data transfer object for submission</td>
</tr>
<tr>
<td><a href="gene_association_subject/">gene_association_subject</a></td>
<td>Subject slot to be used for gene associations</td>
</tr>
<tr>
<td><a href="gene_change_events/">gene_change_events</a></td>
<td>Change events for a given gene.</td>
</tr>
<tr>
<td><a href="gene_chromosomal_location_association_object/">gene_chromosomal_location_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_chromosomal_location_associations/">gene_chromosomal_location_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_full_name/">gene_full_name</a></td>
<td>The one current full name for a gene: e.g., wingless, paired box 2a.</td>
</tr>
<tr>
<td><a href="gene_full_name_dto/">gene_full_name_dto</a></td>
<td>The one current full name for a gene: e.g., wingless, paired box 2a.</td>
</tr>
<tr>
<td><a href="gene_gene_association_object/">gene_gene_association_object</a></td>
<td>the object gene in the association. If the relation is symmetric, subject vs object is arbitrary. We allow a gene product to stand as a proxy for the gene or vice versa.</td>
</tr>
<tr>
<td><a href="gene_genetic_interaction_ingest_set/">gene_genetic_interaction_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_genetic_map_association_object/">gene_genetic_map_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_genetic_map_positions/">gene_genetic_map_positions</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_genomic_location_association_object/">gene_genomic_location_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_genomic_location_associations/">gene_genomic_location_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_identifier/">gene_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the gene</td>
</tr>
<tr>
<td><a href="gene_ingest_set/">gene_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_level_consequence/">gene_level_consequence</a></td>
<td>Boolean indicating whether transcript consequence is the most severe consequence for the corresponding gene.  In the case of equally severe consequences, a single transcript consequence will be designated as the  gene-level consequence by the VEP</td>
</tr>
<tr>
<td><a href="gene_localization_type/">gene_localization_type</a></td>
<td>The type of gene localization, for example, intronic, intergenic, or intragenic.</td>
</tr>
<tr>
<td><a href="gene_localization_type_curie/">gene_localization_type_curie</a></td>
<td>Curie of SOTerm describing the type of gene localization, for example, intronic, intergenic, or intragenic.</td>
</tr>
<tr>
<td><a href="gene_molecular_interaction_ingest_set/">gene_molecular_interaction_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_pathway_association_object/">gene_pathway_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_relation_name/">gene_relation_name</a></td>
<td>Name of VocabularyTerm describing relationship between a subject entity and an object Gene</td>
</tr>
<tr>
<td><a href="gene_secondary_id_dtos/">gene_secondary_id_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_secondary_ids/">gene_secondary_ids</a></td>
<td>Secondary IDs of a given gene</td>
</tr>
<tr>
<td><a href="gene_symbol/">gene_symbol</a></td>
<td>The one current accepted symbol for a gene: e.g., wg, pax2a.</td>
</tr>
<tr>
<td><a href="gene_symbol_dto/">gene_symbol_dto</a></td>
<td>The one current accepted symbol for a gene: e.g., wg, pax2a.</td>
</tr>
<tr>
<td><a href="gene_synonym_dtos/">gene_synonym_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="gene_synonyms/">gene_synonyms</a></td>
<td>Holds between a Gene and a synonym.</td>
</tr>
<tr>
<td><a href="gene_systematic_name/">gene_systematic_name</a></td>
<td>The one current systematic name for a gene: e.g., YHR084W, R09F10.2.</td>
</tr>
<tr>
<td><a href="gene_systematic_name_dto/">gene_systematic_name_dto</a></td>
<td>The one current systematic name for a gene: e.g., YHR084W, R09F10.2.</td>
</tr>
<tr>
<td><a href="gene_type/">gene_type</a></td>
<td>SOTerm describing gene type</td>
</tr>
<tr>
<td><a href="gene_type_curie/">gene_type_curie</a></td>
<td>Curie of SOTerm describing gene type</td>
</tr>
<tr>
<td><a href="gene_types_secondary/">gene_types_secondary</a></td>
<td>SOTerm describing alternate gene types</td>
</tr>
<tr>
<td><a href="generated_by/">generated_by</a></td>
<td>Holds between a material entity and an Agent that generated it: e.g., Thomas Blumenthal, Kornberg Laboratory.</td>
</tr>
<tr>
<td><a href="generated_by_identifier/">generated_by_identifier</a></td>
<td>Identifier for Reagent generated_by field.</td>
</tr>
<tr>
<td><a href="generation_method_dto/">generation_method_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="genes/">genes</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="genetic_map_band/">genetic_map_band</a></td>
<td>Genetic map predicted chromosome location eg 10q12</td>
</tr>
<tr>
<td><a href="genetic_map_position_centimorgan/">genetic_map_position_centimorgan</a></td>
<td>Genetic map distance calculated along a chromosome in centiMorgans: e.g., 22.3366 cM.</td>
</tr>
<tr>
<td><a href="genetic_map_position_centimorgan_error/">genetic_map_position_centimorgan_error</a></td>
<td>Error in calculated genetic map distance in centiMorgans: e.g., 0.045 cM.</td>
</tr>
<tr>
<td><a href="genetic_map_position_radiation/">genetic_map_position_radiation</a></td>
<td>Radiation hybrid map predicted chromosome location: e.g., 66.5 cR.</td>
</tr>
<tr>
<td><a href="genetic_sex/">genetic_sex</a></td>
<td>Submitted value should be a vocabulary term from the 'Genetic sexes' vocabulary</td>
</tr>
<tr>
<td><a href="genetic_sex_name/">genetic_sex_name</a></td>
<td>Name of term from the 'Genetic sexes' vocabulary</td>
</tr>
<tr>
<td><a href="genetically_interacts_with/">genetically_interacts_with</a></td>
<td>holds between two genes whose phenotypic effects are dependent on each other in some way - such that their combined phenotypic effects are the result of some interaction between the activity of their gene products. Examples include epistasis and synthetic lethality.</td>
</tr>
<tr>
<td><a href="genome_assembly/">genome_assembly</a></td>
<td>The GenomeAssembly to which a AssemblyComponent belongs.</td>
</tr>
<tr>
<td><a href="genome_assembly_identifier/">genome_assembly_identifier</a></td>
<td>The identifier for the GenomeAssembly. This can be a MOD curie, a MOD internal ID or an external (3rd party) identifier.</td>
</tr>
<tr>
<td><a href="genome_assembly_ingest_set/">genome_assembly_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="genomic_entity_identifier/">genomic_entity_identifier</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="genomic_entity_type/">genomic_entity_type</a></td>
<td>SOTerm describing genomic entity type</td>
</tr>
<tr>
<td><a href="genomic_entity_type_curie/">genomic_entity_type_curie</a></td>
<td>Curie of SOTerm describing genomic entity type</td>
</tr>
<tr>
<td><a href="genomic_information/">genomic_information</a></td>
<td>A collection of information that represents the strain or genomic background of the sample, either an id or free text</td>
</tr>
<tr>
<td><a href="genomic_information_dto/">genomic_information_dto</a></td>
<td>A collection of information that represents the strain or genomic background of the sample, either an id or free text</td>
</tr>
<tr>
<td><a href="germline_transmission_status/">germline_transmission_status</a></td>
<td>For alleles made in cell lines, have they been transmitted to the germline of an animal</td>
</tr>
<tr>
<td><a href="germline_transmission_status_name/">germline_transmission_status_name</a></td>
<td>Name of the VocabularyTerm representing the germline transmission status</td>
</tr>
<tr>
<td><a href="go_term/">go_term</a></td>
<td>the Gene Ontology (GO) term to which the gene is being annotated</td>
</tr>
<tr>
<td><a href="go_term_curie/">go_term_curie</a></td>
<td>the Gene Ontology (GO) term to which the gene is being annotated</td>
</tr>
<tr>
<td><a href="handle/">handle</a></td>
<td>A slot pointing to a free-text alias or 'handle' for a data object, such as a reference-specific alias for a data object used while curating.</td>
</tr>
<tr>
<td><a href="has_input/">has_input</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="has_output/">has_output</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="has_participant/">has_participant</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="heading_term/">heading_term</a></td>
<td>The MeSH term description or definition from PubMed. e.g. Measles in a Measles/epidemiology term.</td>
</tr>
<tr>
<td><a href="heavy_chain_isotype/">heavy_chain_isotype</a></td>
<td>The isotype of the antibody heavy chain: e.g., IgA.</td>
</tr>
<tr>
<td><a href="heavy_chain_isotype_name/">heavy_chain_isotype_name</a></td>
<td>Name of the VocabularyTerm representing the isotype of the antibody heavy chain: e.g., IgA. From the Heavy Chain Isotope CV.</td>
</tr>
<tr>
<td><a href="height/">height</a></td>
<td>The height of the image (pixels).</td>
</tr>
<tr>
<td><a href="hgvs/">hgvs</a></td>
<td>HGVS nomenclature for variant located on a genomic entity. In practice HGVS nomenclature will be derived for any or all of VariantGenomicLocationAssociation, VariantTranscriptLocationAssociation, and  VariantPolypeptideLocationAssociation.</td>
</tr>
<tr>
<td><a href="hgvs_coding_nomenclature/">hgvs_coding_nomenclature</a></td>
<td>HGVSc nomenclature for variation in coding sequence</td>
</tr>
<tr>
<td><a href="hgvs_protein_nomenclature/">hgvs_protein_nomenclature</a></td>
<td>HGVSp nomenclature for variation in protein</td>
</tr>
<tr>
<td><a href="high_throughput_expression_dataset_annotation_ingest_set/">high_throughput_expression_dataset_annotation_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="high_throughput_expression_dataset_sample_annotation_ingest_set/">high_throughput_expression_dataset_sample_annotation_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="homepage_resource_descriptor_page/">homepage_resource_descriptor_page</a></td>
<td>ResourceDescriptorPage containing URL template for organization's homepage</td>
</tr>
<tr>
<td><a href="homology_confidence/">homology_confidence</a></td>
<td>Confidence of homology match Permissible values: moderate / high / low</td>
</tr>
<tr>
<td><a href="html_file_path/">html_file_path</a></td>
<td>File path of the HTML file generated from the BIRT report</td>
</tr>
<tr>
<td><a href="htp_expression_dataset/">htp_expression_dataset</a></td>
<td>A identifier object for high throughput (HTP) expression dataset, as defined in htpId.json</td>
</tr>
<tr>
<td><a href="htp_expression_dataset_dto/">htp_expression_dataset_dto</a></td>
<td>A identifier object for high throughput (HTP) expression dataset, as defined in htpId.json</td>
</tr>
<tr>
<td><a href="htp_expression_sample/">htp_expression_sample</a></td>
<td>The IDs for this sample object, MOD or external database ID (ex: GEO); primaryId and secondaryId</td>
</tr>
<tr>
<td><a href="htp_expression_sample_age/">htp_expression_sample_age</a></td>
<td>a collection of terms that when used together represent the age and stage of the sample</td>
</tr>
<tr>
<td><a href="htp_expression_sample_age_dto/">htp_expression_sample_age_dto</a></td>
<td>a collection of terms that when used together represent the age and stage of the sample</td>
</tr>
<tr>
<td><a href="htp_expression_sample_dto/">htp_expression_sample_dto</a></td>
<td>The IDs for this sample object, MOD or external database ID (ex: GEO); primaryId and secondaryId</td>
</tr>
<tr>
<td><a href="htp_expression_sample_location_dtos/">htp_expression_sample_location_dtos</a></td>
<td>A collection of terms that when used together represents the location within an organism from which the sample was taken</td>
</tr>
<tr>
<td><a href="htp_expression_sample_locations/">htp_expression_sample_locations</a></td>
<td>A collection of terms that when used together represents the location within an organism from which the sample was taken</td>
</tr>
<tr>
<td><a href="htp_expression_sample_title/">htp_expression_sample_title</a></td>
<td>The free text title of the sample</td>
</tr>
<tr>
<td><a href="htp_expression_sample_type/">htp_expression_sample_type</a></td>
<td>Ontology ID of the biosample type. EX: total mRNA, OBI:XXXXXXX. See enumeration below.</td>
</tr>
<tr>
<td><a href="htp_expression_sample_type_curie/">htp_expression_sample_type_curie</a></td>
<td>Ontology ID of the biosample type. EX: total mRNA, OBI:XXXXXXX. See enumeration below.</td>
</tr>
<tr>
<td><a href="id/">id</a></td>
<td>Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise.</td>
</tr>
<tr>
<td><a href="id_example/">id_example</a></td>
<td>Example ID for the resource that fits the pattern described by the id_pattern slot</td>
</tr>
<tr>
<td><a href="id_pattern/">id_pattern</a></td>
<td>Regex for the expected format of the ID</td>
</tr>
<tr>
<td><a href="identity/">identity</a></td>
<td>Identity score between the two genes</td>
</tr>
<tr>
<td><a href="image_curie/">image_curie</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="image_file/">image_file</a></td>
<td>The file representing the full-sized version of this image.</td>
</tr>
<tr>
<td><a href="image_medium_file/">image_medium_file</a></td>
<td>The file representing a medium-sized version of this image.</td>
</tr>
<tr>
<td><a href="image_thumbnail_file/">image_thumbnail_file</a></td>
<td>The file representing the thumbnail of this image.</td>
</tr>
<tr>
<td><a href="image_x_origin/">image_x_origin</a></td>
<td>The x coordinate start point when specifying a spatial section within a larger image.</td>
</tr>
<tr>
<td><a href="image_y_origin/">image_y_origin</a></td>
<td>The y coordinate start point when specifying a spatial section within a larger image.</td>
</tr>
<tr>
<td><a href="images/">images</a></td>
<td>Any associated image</td>
</tr>
<tr>
<td><a href="in_collection/">in_collection</a></td>
<td>Set of high-throughput alleles made by large projects</td>
</tr>
<tr>
<td><a href="in_collection_name/">in_collection_name</a></td>
<td>Name of VocabularyTerm describing the collection from the 'Allele collection vocabulary' Vocabulary</td>
</tr>
<tr>
<td><a href="inchi/">inchi</a></td>
<td>InChi style description of the molecule</td>
</tr>
<tr>
<td><a href="inchi_key/">inchi_key</a></td>
<td>InChi key description of the molecule</td>
</tr>
<tr>
<td><a href="inferred_allele/">inferred_allele</a></td>
<td>The allele to which something is inferred to be associated.</td>
</tr>
<tr>
<td><a href="inferred_allele_identifier/">inferred_allele_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of allele to which something is inferred to be associated via an automated pipeline</td>
</tr>
<tr>
<td><a href="inferred_gene/">inferred_gene</a></td>
<td>The gene to which something is inferred to be associated.</td>
</tr>
<tr>
<td><a href="inferred_gene_identifier/">inferred_gene_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of gene to which something is inferred to be associated via an automated pipeline</td>
</tr>
<tr>
<td><a href="inheritance_mode/">inheritance_mode</a></td>
<td>Mode of inheritance, e.g. dominant / semi-dominant / recessive / unknown / codominant</td>
</tr>
<tr>
<td><a href="inheritance_mode_name/">inheritance_mode_name</a></td>
<td>Name of VocabularyTerm describing the inheritance mode from the 'Allele inheritance mode vocabulary', e.g. dominant / semi-dominant / recessive / unknown / codominant</td>
</tr>
<tr>
<td><a href="inserted_sequence/">inserted_sequence</a></td>
<td>The sequence of DNA base pairs inserted into the reference sequence by the variant.</td>
</tr>
<tr>
<td><a href="integration_method/">integration_method</a></td>
<td>WormBase captures the method by which an extrachromosomal transgene was integrated into the genome.</td>
</tr>
<tr>
<td><a href="integration_method_name/">integration_method_name</a></td>
<td>WormBase captures the method by which an extrachromosomal transgene was integrated into the genome.</td>
</tr>
<tr>
<td><a href="interaction_id/">interaction_id</a></td>
<td>Identifier for interaction provided by MOD or other external source</td>
</tr>
<tr>
<td><a href="interaction_source/">interaction_source</a></td>
<td>The interaction database that curated the interaction. e.g. BioGRID An MITerm from the AGR interaction source subset of AGR terms.</td>
</tr>
<tr>
<td><a href="interaction_source_curie/">interaction_source_curie</a></td>
<td>The interaction database that curated the interaction. e.g. BioGRID. Value is the curie of an MITerm from the AGR interaction source subset of AGR terms.</td>
</tr>
<tr>
<td><a href="interaction_type/">interaction_type</a></td>
<td>The type of interaction between the two genes or gene products. e.g. physical association. An MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interaction_type_curie/">interaction_type_curie</a></td>
<td>The type of interaction between the two genes or gene products. e.g. physical association. Value is the curie of an MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_A_genetic_perturbation/">interactor_A_genetic_perturbation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="interactor_A_genetic_perturbation_curie/">interactor_A_genetic_perturbation_curie</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="interactor_A_role/">interactor_A_role</a></td>
<td>An MITerm from the AGR interactor A role subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_A_role_curie/">interactor_A_role_curie</a></td>
<td>Curie of an MITerm from the AGR interactor A role subset of MITerm</td>
</tr>
<tr>
<td><a href="interactor_A_type/">interactor_A_type</a></td>
<td>An MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_A_type_curie/">interactor_A_type_curie</a></td>
<td>Curie of an MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_B_genetic_perturbation/">interactor_B_genetic_perturbation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="interactor_B_genetic_perturbation_curie/">interactor_B_genetic_perturbation_curie</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="interactor_B_role/">interactor_B_role</a></td>
<td>An MITerm from the AGR interactor B role subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_B_role_curie/">interactor_B_role_curie</a></td>
<td>Curie of an MITerm from the AGR interactor B role subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_B_type/">interactor_B_type</a></td>
<td>An MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interactor_B_type_curie/">interactor_B_type_curie</a></td>
<td>Curie of an MITerm from the AGR interaction type subset of MITerms</td>
</tr>
<tr>
<td><a href="interacts_with/">interacts_with</a></td>
<td>holds between any two entities that directly or indirectly interact with each other</td>
</tr>
<tr>
<td><a href="internal/">internal</a></td>
<td>Classifies the entity as private (for internal use) or not (for public use).</td>
</tr>
<tr>
<td><a href="intron_number/">intron_number</a></td>
<td>The intron number of the variant location.  If the variant is in an exon, this slot is not populated. Primarily used for zebrafish.</td>
</tr>
<tr>
<td><a href="irradiation_mutagen/">irradiation_mutagen</a></td>
<td>The irradiation used to generate the mutation through mutagenesis</td>
</tr>
<tr>
<td><a href="irradiation_mutagen_name/">irradiation_mutagen_name</a></td>
<td>The name of the irradiation used to generate the mutation through mutagenesis</td>
</tr>
<tr>
<td><a href="is_best_reverse_score/">is_best_reverse_score</a></td>
<td>DIOPT schema highlights the best pair Permissible values: Yes / No</td>
</tr>
<tr>
<td><a href="is_best_score/">is_best_score</a></td>
<td>DIOPT schema highlights the best pair Permissible values: Yes / No / Yes_Adjusted</td>
</tr>
<tr>
<td><a href="is_extinct/">is_extinct</a></td>
<td>Does the allele still exist in a population somewhere?</td>
</tr>
<tr>
<td><a href="is_extrachromosomal/">is_extrachromosomal</a></td>
<td>Used by WormBase to indicate whether a transgenic allele is known to be extrachromosomal.</td>
</tr>
<tr>
<td><a href="is_integrated/">is_integrated</a></td>
<td>Used by WormBase to indicate whether a transgenic allele is known to be integrated into the genome.</td>
</tr>
<tr>
<td><a href="iso_abbreviation/">iso_abbreviation</a></td>
<td></td>
</tr>
<tr>
<td><a href="issue_name/">issue_name</a></td>
<td>The number of the journal issue in which the article was published. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="iupac/">iupac</a></td>
<td>IUPAC name of the molecule</td>
</tr>
<tr>
<td><a href="keywords/">keywords</a></td>
<td>Keywords tagging a publication.  Aggregation of PubMed and ZFIN, editable at Alliance.</td>
</tr>
<tr>
<td><a href="label/">label</a></td>
<td>A short display name for the figure. For example: Figure 2, Figure 3B</td>
</tr>
<tr>
<td><a href="laboratory_of_origin/">laboratory_of_origin</a></td>
<td>The laboratory of origin for the entity.</td>
</tr>
<tr>
<td><a href="laboratory_of_origin_curie/">laboratory_of_origin_curie</a></td>
<td>The curie of the laboratory of origin for the entity.</td>
</tr>
<tr>
<td><a href="language/">language</a></td>
<td>Language of the reference.  Aggregation of PubMed and FlyBase, editable at Alliance.</td>
</tr>
<tr>
<td><a href="last/">last</a></td>
<td>last identifier in a range</td>
</tr>
<tr>
<td><a href="last_name/">last_name</a></td>
<td>last (family) name of a person</td>
</tr>
<tr>
<td><a href="left_boundary_marker/">left_boundary_marker</a></td>
<td>The left boundary of a feature location relative to the landmark's inherent orientation.</td>
</tr>
<tr>
<td><a href="length/">length</a></td>
<td>The length of the aligned regions between two genes</td>
</tr>
<tr>
<td><a href="light_chain_isotype/">light_chain_isotype</a></td>
<td>The isotype of the antibody light chain: e.g., i4.</td>
</tr>
<tr>
<td><a href="light_chain_isotype_name/">light_chain_isotype_name</a></td>
<td>Name of the VocabularyTerm representing the isotype of the antibody light chain: e.g., i4. From the Light Chain Isotope CV.</td>
</tr>
<tr>
<td><a href="linkml_schema_version/">linkml_schema_version</a></td>
<td>Release of LinkML for which the file was generated and submitted</td>
</tr>
<tr>
<td><a href="linkml_version/">linkml_version</a></td>
<td>Version of LinkML schema used for submitted file in the format n.n.n (e.g. 1.2.4 or 2.0.0)</td>
</tr>
<tr>
<td><a href="load_exceptions/">load_exceptions</a></td>
<td>A list of execeptions the load encountered</td>
</tr>
<tr>
<td><a href="load_files/">load_files</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="load_finished/">load_finished</a></td>
<td>The date and time that the load finished.</td>
</tr>
<tr>
<td><a href="load_started/">load_started</a></td>
<td>The date and time that the load started.</td>
</tr>
<tr>
<td><a href="loads/">loads</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="local_file_path/">local_file_path</a></td>
<td>Local file path to where the file has been saved for processing</td>
</tr>
<tr>
<td><a href="manufactured_by/">manufactured_by</a></td>
<td>Holds between a material entity and an Agent that has manufactured it: e.g., Molecular Probes.</td>
</tr>
<tr>
<td><a href="manufactured_by_identifier/">manufactured_by_identifier</a></td>
<td>Identifier for Reagent manufactured_by field.</td>
</tr>
<tr>
<td><a href="maps_to_chromosome/">maps_to_chromosome</a></td>
<td>The chromosome(s) to which a chromosome assembly or unlocalized scaffold or contig maps.</td>
</tr>
<tr>
<td><a href="md5_sum/">md5_sum</a></td>
<td>Generated md5Sum of the file that has been uploaded</td>
</tr>
<tr>
<td><a href="medline_abbreviation/">medline_abbreviation</a></td>
<td></td>
</tr>
<tr>
<td><a href="member_terms/">member_terms</a></td>
<td>Set of VocabularyTerm objects in a Vocabulary object set</td>
</tr>
<tr>
<td><a href="merged_into/">merged_into</a></td>
<td>For an entity deprecated by a merge, the entity into which it was merged.</td>
</tr>
<tr>
<td><a href="merged_into_id/">merged_into_id</a></td>
<td>ID that used to refer to this reference</td>
</tr>
<tr>
<td><a href="mesh_detail_id/">mesh_detail_id</a></td>
<td>The primary key for a MeshDetail object.</td>
</tr>
<tr>
<td><a href="microarray_sample_details/">microarray_sample_details</a></td>
<td>collection of information that is specific for microarray samples</td>
</tr>
<tr>
<td><a href="microarray_sample_details_dto/">microarray_sample_details_dto</a></td>
<td>collection of information that is specific for microarray samples</td>
</tr>
<tr>
<td><a href="middle_name/">middle_name</a></td>
<td>middle names of a person</td>
</tr>
<tr>
<td><a href="mod_corpus_association_id/">mod_corpus_association_id</a></td>
<td>An integer referring to a ModCorpusAssociation object. A primary key in the mod_corpus_association table, a foreign key if used in other tables.</td>
</tr>
<tr>
<td><a href="mod_corpus_sort_source/">mod_corpus_sort_source</a></td>
<td>origin of corpus determination for that publication and mod</td>
</tr>
<tr>
<td><a href="mod_internal_id/">mod_internal_id</a></td>
<td>The model organism database (MOD) internal identifier for the object</td>
</tr>
<tr>
<td><a href="moderate_filter/">moderate_filter</a></td>
<td>Indicates whether this homology match is displayed when the moderate filter is selected</td>
</tr>
<tr>
<td><a href="mutagenesis_method_names/">mutagenesis_method_names</a></td>
<td>Name of the VocabularyTerm describing the mutagenesis method, e.g. spontaneous / naturally occurring / radiation-induced / recombinant / ENU / CRISPR / TALEN / gamma rays / not specified / spontaneous / DNA / DNA AND CRISPR / DNA and TALEN / zinc finger nuclease / EMS</td>
</tr>
<tr>
<td><a href="mutagenesis_methods/">mutagenesis_methods</a></td>
<td>Technique used to create the allele, e.g. spontaneous / naturally occurring / radiation-induced / recombinant / ENU / CRISPR / TALEN / gamma rays / not specified / spontaneous / DNA / DNA AND CRISPR / DNA and TALEN / zinc finger nuclease / EMS</td>
</tr>
<tr>
<td><a href="mutagenesis_target/">mutagenesis_target</a></td>
<td>The target of the mutation, e.g. strain / adult females / adult males / embryos / sperm / not specified</td>
</tr>
<tr>
<td><a href="mutation_target_strain/">mutation_target_strain</a></td>
<td>The particular strain (solely for and from MGI) that is targeted by the generation method for a particular allele.</td>
</tr>
<tr>
<td><a href="mutation_target_strain_identifier/">mutation_target_strain_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the particular strain that is targeted by the generation method for a particular allele (MGI only)</td>
</tr>
<tr>
<td><a href="mutation_type_curies/">mutation_type_curies</a></td>
<td>Curies of SOTerms representing mutation type</td>
</tr>
<tr>
<td><a href="mutation_types/">mutation_types</a></td>
<td>SO term for type of mutation</td>
</tr>
<tr>
<td><a href="name/">name</a></td>
<td>a human-readable name for an entity</td>
</tr>
<tr>
<td><a href="name_type/">name_type</a></td>
<td>The type of name: e.g., symbol, full_name, systematic_name, etc.</td>
</tr>
<tr>
<td><a href="name_type_name/">name_type_name</a></td>
<td>Name of the VocabularyTerm representing the name type of the synonym - proposed values are nomenclature_symbol / full_name / systematic_name / ncbi_protein_name / uniform / non_uniform / retired_name / unspecified</td>
</tr>
<tr>
<td><a href="namespace/">namespace</a></td>
<td>the namespace of the ontology.</td>
</tr>
<tr>
<td><a href="negated/">negated</a></td>
<td>if set to true, then the association is negated i.e. is not true</td>
</tr>
<tr>
<td><a href="nomenclature_event/">nomenclature_event</a></td>
<td>any of the kinds of changes made to an object's name or symbol</td>
</tr>
<tr>
<td><a href="nomenclature_event_name/">nomenclature_event_name</a></td>
<td>Name of the VocabularyTerm describing the nomenclature event</td>
</tr>
<tr>
<td><a href="note_dto/">note_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="note_dtos/">note_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="note_type/">note_type</a></td>
<td>The type of note: e.g., cytology, comment, summary.</td>
</tr>
<tr>
<td><a href="note_type_name/">note_type_name</a></td>
<td>Name of VocabularyTerm representing note type selected from the appropriate Vocabulary</td>
</tr>
<tr>
<td><a href="number_additional_dna_base_pairs/">number_additional_dna_base_pairs</a></td>
<td>The number of additional DNA base pairs inserted into the reference sequence by the variant.</td>
</tr>
<tr>
<td><a href="number_amino_acids_inserted/">number_amino_acids_inserted</a></td>
<td>The number of amino acids added from the polypeptide as a result of the variant.</td>
</tr>
<tr>
<td><a href="number_amino_acids_removed/">number_amino_acids_removed</a></td>
<td>The number of amino acids removed from the polypeptide as a result of the variant.</td>
</tr>
<tr>
<td><a href="number_of_channels/">number_of_channels</a></td>
<td>For microarrays, the number of channels used; This can be one of two values: 1 or 2</td>
</tr>
<tr>
<td><a href="number_removed_dna_base_pairs/">number_removed_dna_base_pairs</a></td>
<td>The number of DNA base pairs removed from the reference sequence by the variant.   sequence.</td>
</tr>
<tr>
<td><a href="object_allele_identifier/">object_allele_identifier</a></td>
<td>The identifier (curie/primary_external_id/mod_internal_id) of the allele that is acting as the object of an AlleleAlleleAssociation</td>
</tr>
<tr>
<td><a href="object_gene/">object_gene</a></td>
<td>The second gene in the homology match pair</td>
</tr>
<tr>
<td><a href="object_gene_identifier/">object_gene_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the object gene in a gene-to-gene association</td>
</tr>
<tr>
<td><a href="object_set/">object_set</a></td>
<td>Applies to a property that links a ingest object to a set of objects. This is necessary in a json document to provide context for a list, and to allow for a single json object that combines multiple object types</td>
</tr>
<tr>
<td><a href="obsolete/">obsolete</a></td>
<td>Entity is no longer current.</td>
</tr>
<tr>
<td><a href="okta_email/">okta_email</a></td>
<td>The email address of the LoggedInPerson registered with Okta for user authentication</td>
</tr>
<tr>
<td><a href="okta_id/">okta_id</a></td>
<td>The Okta identifier for the person registered in Okta for user authentication</td>
</tr>
<tr>
<td><a href="old_emails/">old_emails</a></td>
<td>list of old (outdated) emails for a person</td>
</tr>
<tr>
<td><a href="old_members/">old_members</a></td>
<td>Gene which were formerly members of this GeneClass</td>
</tr>
<tr>
<td><a href="online_issn/">online_issn</a></td>
<td></td>
</tr>
<tr>
<td><a href="ontology_closure_ingest_set/">ontology_closure_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ontology_term_closure_object/">ontology_term_closure_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ontology_term_closure_subject/">ontology_term_closure_subject</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ontology_type/">ontology_type</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="open_access/">open_access</a></td>
<td>Indicates if the reference is freely available for use by anyone, usually with fewer copyright and licensing barriers.</td>
</tr>
<tr>
<td><a href="orcid/">orcid</a></td>
<td>Open Researcher and Contributor ID</td>
</tr>
<tr>
<td><a href="original_reference/">original_reference</a></td>
<td>Holds between an entity and the first reference to describe that entity.</td>
</tr>
<tr>
<td><a href="original_reference_curie/">original_reference_curie</a></td>
<td>Curie of the first reference to describe an entity.</td>
</tr>
<tr>
<td><a href="padded_base/">padded_base</a></td>
<td>A base preceded an insertion or deletion event that is reported when either the reference or variant sequence would otherwise be empty.</td>
</tr>
<tr>
<td><a href="page_area/">page_area</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="page_description/">page_description</a></td>
<td>Description of page</td>
</tr>
<tr>
<td><a href="page_range/">page_range</a></td>
<td>Page numbers of source referenced for statement or publication. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="paralogous_genes/">paralogous_genes</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="parental_populations/">parental_populations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="pdf_file_path/">pdf_file_path</a></td>
<td>File path of the PDF file generated from the BIRT report</td>
</tr>
<tr>
<td><a href="phase/">phase</a></td>
<td>Number of nucleotides from the start of a feature to the start of a complete codon</td>
</tr>
<tr>
<td><a href="phenotype_agm_ingest_set/">phenotype_agm_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="phenotype_allele_ingest_set/">phenotype_allele_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="phenotype_annotation_object/">phenotype_annotation_object</a></td>
<td>phenotype statement: The label of an individual phenotype term from a phenotype ontology or the post-composed statement of the phenotype from a post-composed terminology</td>
</tr>
<tr>
<td><a href="phenotype_annotation_subject/">phenotype_annotation_subject</a></td>
<td>The biological entity (e.g. gene, allele, genotype) to which the phenotype is associated, by direct curation.</td>
</tr>
<tr>
<td><a href="phenotype_gene_ingest_set/">phenotype_gene_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="phenotype_statement/">phenotype_statement</a></td>
<td>For MODs that curate to post-composed phenotype expressions composed of ontology terms and other necessary information (e.g. vocabularies, free text). This field captures the free text phenotype statement.</td>
</tr>
<tr>
<td><a href="phenotype_term/">phenotype_term</a></td>
<td>The phenotype ontology term used to describe the phenotype of an organism or a set of organisms.</td>
</tr>
<tr>
<td><a href="phenotype_term_curie/">phenotype_term_curie</a></td>
<td>The string representation of the phenotype ontology term (PhenotypeTerm) curie</td>
</tr>
<tr>
<td><a href="phenotype_term_curies/">phenotype_term_curies</a></td>
<td>The string representation of the phenotype ontology term (PhenotypeTerm) curie</td>
</tr>
<tr>
<td><a href="phenotype_terms/">phenotype_terms</a></td>
<td>The phenotype ontology term used to describe the phenotype of an organism or a set of organisms.</td>
</tr>
<tr>
<td><a href="phenotypes_or_traits/">phenotypes_or_traits</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="phylogenetic_order/">phylogenetic_order</a></td>
<td>The integer value assigned to a species to determine its relative ordering with respect to other Alliance species in various user interface displays and logical operations.</td>
</tr>
<tr>
<td><a href="physically_interacts_with/">physically_interacts_with</a></td>
<td>holds between two entities that make physical contact as part of some interaction</td>
</tr>
<tr>
<td><a href="placeholder/">placeholder</a></td>
<td>A boolean property to indicate whether a data record/object/entity is intended to be a placeholder data object rather than a publicly displayed object with a proper ID, name and/or symbol, for example. For example, the Cassette class has been introduced to support FlyBase Cassette objects with IDs and names, but not every Alliance member has data tables/classes for these objects and any IDs or names/symbols would likely be derived and not helpful to end users. Data records/entities where the placeholder value is set to true would have various visible properties suppressed, so as not to confuse end users with private, internal or otherwise contrived representations.</td>
</tr>
<tr>
<td><a href="plain_language_abstract/">plain_language_abstract</a></td>
<td>Lay person, readable version of the abstract. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="polyphen_prediction/">polyphen_prediction</a></td>
<td>PolyPhen-2 prediction</td>
</tr>
<tr>
<td><a href="polyphen_score/">polyphen_score</a></td>
<td>PolyPhen-2 score between 0 and 1</td>
</tr>
<tr>
<td><a href="predicted_variant_consequences/">predicted_variant_consequences</a></td>
<td>VEP-calculated variant consequences</td>
</tr>
<tr>
<td><a href="prediction_methods_matched/">prediction_methods_matched</a></td>
<td>Source of the homology match</td>
</tr>
<tr>
<td><a href="prediction_methods_not_called/">prediction_methods_not_called</a></td>
<td>Prediction methods that were not called</td>
</tr>
<tr>
<td><a href="prediction_methods_not_matched/">prediction_methods_not_matched</a></td>
<td>Prediction methods not matched</td>
</tr>
<tr>
<td><a href="preferred_cross_reference/">preferred_cross_reference</a></td>
<td>cross reference to the preferred page to link to in search</td>
</tr>
<tr>
<td><a href="preferred_cross_reference_dto/">preferred_cross_reference_dto</a></td>
<td>cross reference to the preferred page to link to in search</td>
</tr>
<tr>
<td><a href="prefix/">prefix</a></td>
<td>Denormalization to help with classifying types of crossReferences, distinguishing DOIs from PMC ids, etc.</td>
</tr>
<tr>
<td><a href="prefix_order/">prefix_order</a></td>
<td>The relative order of the resource when listed with other crossReferences.</td>
</tr>
<tr>
<td><a href="prefix_page/">prefix_page</a></td>
<td>The category of pages the resource in the context of the URL associated with the crossReference provides.  Equivalent to the 'page' attribute in the Alliance resourceDescriptor file.</td>
</tr>
<tr>
<td><a href="primary_external_id/">primary_external_id</a></td>
<td>The primary external (non-Alliance) database identifier/curie for the object. Note that this may be an external (non-Alliance member) identifier for an object, like a UniProt ID for a protein, and may act as the MOD's/Alliance member's primary key for the entity.</td>
</tr>
<tr>
<td><a href="primary_image/">primary_image</a></td>
<td>The primary image for an allele that is used to represent the allele on a page.</td>
</tr>
<tr>
<td><a href="print_issn/">print_issn</a></td>
<td></td>
</tr>
<tr>
<td><a href="product_binds_matrix/">product_binds_matrix</a></td>
<td>WB specific. ID of position matrix object</td>
</tr>
<tr>
<td><a href="protein_identifier/">protein_identifier</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="protein_ingest_set/">protein_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="proteins/">proteins</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="publisher/">publisher</a></td>
<td>Publisher associated with a reference or resource. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="pubmed_abstract_languages/">pubmed_abstract_languages</a></td>
<td>Languages for the abstract. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="pubmed_publication_status/">pubmed_publication_status</a></td>
<td>Status of the publication at PubMed. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="pubmed_type/">pubmed_type</a></td>
<td>Type of Reference as described by PubMed. Only relevant for PubMed references.</td>
</tr>
<tr>
<td><a href="qualifier_term/">qualifier_term</a></td>
<td>The MeSH term subheading from PubMed, to narrow down the topic. e.g. epidemiology in a Measles/epidemiology term.</td>
</tr>
<tr>
<td><a href="qualifiers/">qualifiers</a></td>
<td>This is the MeSH qualifier term that is optionally added to the descriptor term.</td>
</tr>
<tr>
<td><a href="rank/">rank</a></td>
<td>The rank (order) of the paralog result</td>
</tr>
<tr>
<td><a href="record_count/">record_count</a></td>
<td>The number of records found in the file</td>
</tr>
<tr>
<td><a href="reference_curie/">reference_curie</a></td>
<td>External reference curie used for ingest</td>
</tr>
<tr>
<td><a href="reference_curies/">reference_curies</a></td>
<td>External reference curies used for ingest</td>
</tr>
<tr>
<td><a href="reference_id/">reference_id</a></td>
<td>An integer referring to a Reference object in the references table. It's a primary key in the references table, a foreign key when in other tables.</td>
</tr>
<tr>
<td><a href="reference_sequence/">reference_sequence</a></td>
<td>Reference sequence of genome or genomic entity at position of aligned variant.</td>
</tr>
<tr>
<td><a href="referenced_curie/">referenced_curie</a></td>
<td>Curie of the thing being referenced in a CrossReference</td>
</tr>
<tr>
<td><a href="references/">references</a></td>
<td>holds between an object and a list of references</td>
</tr>
<tr>
<td><a href="related_figure/">related_figure</a></td>
<td>Holds between an entity and a single Figure. Singular version of related_figures.</td>
</tr>
<tr>
<td><a href="related_figures/">related_figures</a></td>
<td>Holds between an entity and a set of Figures.</td>
</tr>
<tr>
<td><a href="related_note/">related_note</a></td>
<td>Holds between an object and a Note object.</td>
</tr>
<tr>
<td><a href="related_notes/">related_notes</a></td>
<td>Holds between an object and a list of related Note objects.</td>
</tr>
<tr>
<td><a href="related_to/">related_to</a></td>
<td>A relationship that is asserted between two named things.</td>
</tr>
<tr>
<td><a href="relation/">relation</a></td>
<td>A high-level grouping for the relationship type. This is analogous to category for nodes. In RDF, this corresponds to rdf:predicate and in Neo4j this corresponds to the relationship type.</td>
</tr>
<tr>
<td><a href="relation_name/">relation_name</a></td>
<td>Name of VocabularyTerm representing relation of an Association</td>
</tr>
<tr>
<td><a href="relationship_type/">relationship_type</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="release/">release</a></td>
<td>MOD release ID</td>
</tr>
<tr>
<td><a href="resource_descriptor_page/">resource_descriptor_page</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="resource_id/">resource_id</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="resource_pages/">resource_pages</a></td>
<td>Pages for a particular resource</td>
</tr>
<tr>
<td><a href="right_boundary_marker/">right_boundary_marker</a></td>
<td>The right boundary of a feature location relative to the landmark's inherent orientation.</td>
</tr>
<tr>
<td><a href="s3_path/">s3_path</a></td>
<td>The relative path to the file in the S3 bucket</td>
</tr>
<tr>
<td><a href="s3_url/">s3_url</a></td>
<td>The full URL to the file from S3</td>
</tr>
<tr>
<td><a href="schedule_active/">schedule_active</a></td>
<td>This determines if the the schedule is active or not</td>
</tr>
<tr>
<td><a href="scheduling_error_message/">scheduling_error_message</a></td>
<td>Any errors in syntax on the cron_schedule</td>
</tr>
<tr>
<td><a href="secondary_data_provider/">secondary_data_provider</a></td>
<td>Object representing organization (e.g. MOD) that provided the data directly to the Alliance, but not the original source</td>
</tr>
<tr>
<td><a href="secondary_data_provider_cross_reference/">secondary_data_provider_cross_reference</a></td>
<td>CrossReference to the organization that provided the data to the Alliance</td>
</tr>
<tr>
<td><a href="secondary_data_provider_dto/">secondary_data_provider_dto</a></td>
<td>Ingest object representing organization (e.g. MOD) that provided the data directly to the Alliance, but not the original source, and a CrossReference to that organisation's site</td>
</tr>
<tr>
<td><a href="secondary_id/">secondary_id</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="secondary_identifiers/">secondary_identifiers</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="sequence_targeting_reagent_identifier/">sequence_targeting_reagent_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the STR</td>
</tr>
<tr>
<td><a href="sequencing_format/">sequencing_format</a></td>
<td>single or paired end sequencing; might be better in experiment info</td>
</tr>
<tr>
<td><a href="sgd_strain_background/">sgd_strain_background</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="sgd_strain_background_identifier/">sgd_strain_background_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of SGD strain background AGM</td>
</tr>
<tr>
<td><a href="short_name/">short_name</a></td>
<td>The short name of the entity</td>
</tr>
<tr>
<td><a href="sift_prediction/">sift_prediction</a></td>
<td>SIFT prediction</td>
</tr>
<tr>
<td><a href="sift_score/">sift_score</a></td>
<td>SIFT score between 0 and 1</td>
</tr>
<tr>
<td><a href="similarity/">similarity</a></td>
<td>Similarity score between the two genes</td>
</tr>
<tr>
<td><a href="single_agm/">single_agm</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_allele/">single_allele</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_cassette/">single_cassette</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_construct/">single_construct</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_gene/">single_gene</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_reference/">single_reference</a></td>
<td>holds between an object and a single reference</td>
</tr>
<tr>
<td><a href="single_transgenic_tool/">single_transgenic_tool</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="single_variant/">single_variant</a></td>
<td>The variant that is the subject of a SlotAnnotation.</td>
</tr>
<tr>
<td><a href="smiles/">smiles</a></td>
<td>Molecular structure in SMILES format</td>
</tr>
<tr>
<td><a href="source_general_consequence_curie/">source_general_consequence_curie</a></td>
<td>Curie of the SOTerm (child of SO:0001576 - transcript_variant) that describes the consequence of the variant, as stated in the source reference when no transcript ID is provided. Since a curator would determine variant location and consequences relative to at least one specific genome assembly, transcript and/or polypeptide, no slot for curated general consequence is provided.</td>
</tr>
<tr>
<td><a href="source_general_consequence_dtos/">source_general_consequence_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_general_consequences/">source_general_consequences</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_organization_abbreviation/">source_organization_abbreviation</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_variant_genomic_location_association_ingest_set/">source_variant_genomic_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_variant_genomic_locations/">source_variant_genomic_locations</a></td>
<td>Location of the variant in a genomic context as described by the source reference. Multivalued as there may be multiple source references.</td>
</tr>
<tr>
<td><a href="source_variant_polypeptide_locations/">source_variant_polypeptide_locations</a></td>
<td>Location of the variant within polypeptide entities as described by the source reference.</td>
</tr>
<tr>
<td><a href="source_variant_protein_location_association_ingest_set/">source_variant_protein_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_variant_transcript_location_association_ingest_set/">source_variant_transcript_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="source_variant_transcript_locations/">source_variant_transcript_locations</a></td>
<td>Location of the variant within transcript entities as described by the source reference.</td>
</tr>
<tr>
<td><a href="specimen_alleles/">specimen_alleles</a></td>
<td>The Allele(s) of the specimen assayed.</td>
</tr>
<tr>
<td><a href="specimen_genomic_model/">specimen_genomic_model</a></td>
<td>The AffectedGenomicModel of the specimen.</td>
</tr>
<tr>
<td><a href="specimen_genomic_model_identifier/">specimen_genomic_model_identifier</a></td>
<td>The identifier of the AffectedGenomicModel of the specimen. This can be a MOD curie, a MOD internal ID or an external (3rd party) identifier.</td>
</tr>
<tr>
<td><a href="split_from/">split_from</a></td>
<td>For an entity generated by a split, the entity from which it was split.</td>
</tr>
<tr>
<td><a href="split_into/">split_into</a></td>
<td>For an entity deprecated by a split, the list of entities generated from the split.</td>
</tr>
<tr>
<td><a href="stage/">stage</a></td>
<td>The life stage of an organism or biosample as expressed by a TemporalContext entity</td>
</tr>
<tr>
<td><a href="stage_dto/">stage_dto</a></td>
<td>The life stage of an organism or biosample as expressed by a TemporalContext entity</td>
</tr>
<tr>
<td><a href="stage_uberon_slim_term_curies/">stage_uberon_slim_term_curies</a></td>
<td>The high-level UBERON slim terms that apply to the developmental range of the TemporalContext component of an ExpressionPattern. The following UBERON terms are allowed: UBERON:0000068, UBERON:0000113. A non-UBERON term is also allowed: post embryonic, pre-adult.</td>
</tr>
<tr>
<td><a href="stage_uberon_slim_terms/">stage_uberon_slim_terms</a></td>
<td>The high-level UBERON slim terms that apply to the developmental range of the TemporalContext component of an ExpressionPattern. The following UBERON terms are allowed: UBERON:0000068, UBERON:0000113. A non-UBERON term is also allowed: post embryonic, pre-adult.</td>
</tr>
<tr>
<td><a href="start/">start</a></td>
<td>The start of the feature in positive 1-based integer coordinates relative to the reference landmark.</td>
</tr>
<tr>
<td><a href="str_association_subject/">str_association_subject</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="str_gene_association_ingest_set/">str_gene_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="str_gene_association_object/">str_gene_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="str_gene_associations/">str_gene_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="str_ingest_set/">str_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="strand/">strand</a></td>
<td>The strand of a DNA reference object to which an entity maps.</td>
</tr>
<tr>
<td><a href="strict_filter/">strict_filter</a></td>
<td>Indicates whether this homology match is displayed when the strict filter is selected</td>
</tr>
<tr>
<td><a href="sub_series/">sub_series</a></td>
<td>If the dataset is a SuperSeries, this is a list of dataset ids that are associated with this dataset</td>
</tr>
<tr>
<td><a href="subdomain_code/">subdomain_code</a></td>
<td>a three letter string, representing a subdomain (e.g 101 represents disease_annotation)</td>
</tr>
<tr>
<td><a href="subdomain_name/">subdomain_name</a></td>
<td>subdomain name (e.g disease_annotation)</td>
</tr>
<tr>
<td><a href="subject_gene/">subject_gene</a></td>
<td>The first gene in the homology match pair</td>
</tr>
<tr>
<td><a href="subject_gene_identifier/">subject_gene_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the subject gene in a gene-to-gene association</td>
</tr>
<tr>
<td><a href="subsets/">subsets</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="subtype/">subtype</a></td>
<td>Subtype of affected genomic model - permissible values: strain / genotype / fish</td>
</tr>
<tr>
<td><a href="subtype_name/">subtype_name</a></td>
<td>Name of VocabularyTerm describing subtype - permissible values: strain / genotype / fish</td>
</tr>
<tr>
<td><a href="summary/">summary</a></td>
<td></td>
</tr>
<tr>
<td><a href="symbol_renamed_from/">symbol_renamed_from</a></td>
<td>The previous symbol for a renamed entity.</td>
</tr>
<tr>
<td><a href="symbol_renamed_to/">symbol_renamed_to</a></td>
<td>The new symbol for a renamed entity.</td>
</tr>
<tr>
<td><a href="synonym_scope/">synonym_scope</a></td>
<td>the scope of the synonym - permissible values are narrow / broad / related / exact</td>
</tr>
<tr>
<td><a href="synonym_scope_name/">synonym_scope_name</a></td>
<td>Name of the VocabularyTerm representing the scope of the synonym - permissible values are narrow / broad / related / exact</td>
</tr>
<tr>
<td><a href="synonym_url/">synonym_url</a></td>
<td>URL for a synonym: e.g., NCBI URL for the NCBI synonym of an object.</td>
</tr>
<tr>
<td><a href="synonyms/">synonyms</a></td>
<td>A generic free-text field for objects that only have string representation of one or more synonyms and do not require meta data or attribution; if meta data or attribution are required to be captured for individual synonyms, consider the NameSlotAnnotation class</td>
</tr>
<tr>
<td><a href="table_key/">table_key</a></td>
<td>The primary key for a specific table entry, unique for that table.</td>
</tr>
<tr>
<td><a href="taxon/">taxon</a></td>
<td>The taxon from which the biological entity derives.</td>
</tr>
<tr>
<td><a href="taxon_curie/">taxon_curie</a></td>
<td>Curie of the NCBITaxonTerm representing the taxon from which the biological entity derives</td>
</tr>
<tr>
<td><a href="taxon_text/">taxon_text</a></td>
<td>The free text reference to a species or general taxon (e.g. "Yeast") for a a biological entity for which the exact taxon curie is unknown or unavailable.</td>
</tr>
<tr>
<td><a href="temporal_qualifier_names/">temporal_qualifier_names</a></td>
<td>Qualifiers of the stage or age in an annotation. The range is a string representing vocabulary term</td>
</tr>
<tr>
<td><a href="temporal_qualifiers/">temporal_qualifiers</a></td>
<td>Qualifiers of the stage or age in an annotation.</td>
</tr>
<tr>
<td><a href="title/">title</a></td>
<td>A human readable title for a reference. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="total_records/">total_records</a></td>
<td>The total number of records</td>
</tr>
<tr>
<td><a href="trans_splice_leaders/">trans_splice_leaders</a></td>
<td>Trans-splicing splice leaders observed to operate on this gene, in species which have spliced leader trans-splicing</td>
</tr>
<tr>
<td><a href="transcript_association_subject/">transcript_association_subject</a></td>
<td>Subject slot to be used for transcript associations</td>
</tr>
<tr>
<td><a href="transcript_cds_associations/">transcript_cds_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_coding_sequence_association_object/">transcript_coding_sequence_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_exon_association_object/">transcript_exon_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_exon_associations/">transcript_exon_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_gene_association/">transcript_gene_association</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_gene_association_object/">transcript_gene_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_genomic_location_association_object/">transcript_genomic_location_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_genomic_location_associations/">transcript_genomic_location_associations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_id/">transcript_id</a></td>
<td>The transcript identifier used by the Ensembl VEP to identify the transcript and generate HGVSc identifiers.  This corresponds to the transcript_id attribute in column 9 of transcript entries in the GFF files.</td>
</tr>
<tr>
<td><a href="transcript_identifier/">transcript_identifier</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_ingest_set/">transcript_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transcript_type/">transcript_type</a></td>
<td>SOTerm describing transcript type</td>
</tr>
<tr>
<td><a href="transcript_type_curie/">transcript_type_curie</a></td>
<td>Curie of SOTerm describing transcript type</td>
</tr>
<tr>
<td><a href="transgene_chromosome_location/">transgene_chromosome_location</a></td>
<td>The chromosome to which a transgene has been mapped. Used for WormBase transgenes that have been integrated into the genome and mapped to a chromosome.</td>
</tr>
<tr>
<td><a href="transgene_chromosome_location_name/">transgene_chromosome_location_name</a></td>
<td>The name of the chromosome to which a transgene has been mapped. Used for WormBase transgenes that have been integrated into the genome and mapped to a chromosome.</td>
</tr>
<tr>
<td><a href="transgenic_tool_association_subject/">transgenic_tool_association_subject</a></td>
<td>Subject slot to be used for TransgenicTool associations</td>
</tr>
<tr>
<td><a href="transgenic_tool_full_name/">transgenic_tool_full_name</a></td>
<td>Holds between a TransgenicTool and a full name.</td>
</tr>
<tr>
<td><a href="transgenic_tool_full_name_dto/">transgenic_tool_full_name_dto</a></td>
<td>Holds between a TransgenicTool and a full name.</td>
</tr>
<tr>
<td><a href="transgenic_tool_identifier/">transgenic_tool_identifier</a></td>
<td>Either the MOD curie or internal ID that was used in the submission of the associated transgenic tool</td>
</tr>
<tr>
<td><a href="transgenic_tool_ingest_set/">transgenic_tool_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transgenic_tool_object_identifier/">transgenic_tool_object_identifier</a></td>
<td>Curie or internal ID of TransgenicTool that is the object of an association</td>
</tr>
<tr>
<td><a href="transgenic_tool_subject_identifier/">transgenic_tool_subject_identifier</a></td>
<td>Curie or internal ID of TransgenicTool that is the subject of an association</td>
</tr>
<tr>
<td><a href="transgenic_tool_symbol/">transgenic_tool_symbol</a></td>
<td>Holds between a TransgenicTool and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="transgenic_tool_symbol_dto/">transgenic_tool_symbol_dto</a></td>
<td>Holds between a TransgenicTool and the one current accepted symbol</td>
</tr>
<tr>
<td><a href="transgenic_tool_synonym_dtos/">transgenic_tool_synonym_dtos</a></td>
<td>Holds between a TransgenicTool and a list of synonyms.</td>
</tr>
<tr>
<td><a href="transgenic_tool_synonyms/">transgenic_tool_synonyms</a></td>
<td>Holds between a TransgenicTool and a list of synonyms.</td>
</tr>
<tr>
<td><a href="transgenic_tool_transgenic_tool_association_ingest_set/">transgenic_tool_transgenic_tool_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transgenic_tool_transgenic_tool_association_object/">transgenic_tool_transgenic_tool_association_object</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="transposon_insertion/">transposon_insertion</a></td>
<td>Associated transposon insertion that causes the mutation</td>
</tr>
<tr>
<td><a href="transposon_origin/">transposon_origin</a></td>
<td>If this gene contains or is originating from a transposon</td>
</tr>
<tr>
<td><a href="type/">type</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="uncertain/">uncertain</a></td>
<td>If set to true, then the related entity is uncertain.</td>
</tr>
<tr>
<td><a href="unique_id/">unique_id</a></td>
<td>A non-curie unique identifier for a thing.</td>
</tr>
<tr>
<td><a href="updated_by/">updated_by</a></td>
<td>The individual that last modified the entity.</td>
</tr>
<tr>
<td><a href="updated_by_curie/">updated_by_curie</a></td>
<td>Curie of the Person object representing the individual that updated the entity</td>
</tr>
<tr>
<td><a href="url_prefix/">url_prefix</a></td>
<td>The prefix of the url before the accession number.</td>
</tr>
<tr>
<td><a href="url_suffix/">url_suffix</a></td>
<td>The suffix of the url after the accession number.</td>
</tr>
<tr>
<td><a href="url_template/">url_template</a></td>
<td>URL template for constructing link to resource using provided ID, eg. "https://www.omim.org/phenotypicSeries/[%s]"</td>
</tr>
<tr>
<td><a href="use_curies/">use_curies</a></td>
<td>Holds between a TransgenicTool and a list of curies of FBcv terms with namespace 'experimental_tool_descriptor'</td>
</tr>
<tr>
<td><a href="user_settings/">user_settings</a></td>
<td>JSON Blob containing details of UI settings for LoggedInPerson</td>
</tr>
<tr>
<td><a href="uses/">uses</a></td>
<td>Holds between a TransgenicTool and a list of FBcv terms with namespace 'experimental_tool_descriptor'</td>
</tr>
<tr>
<td><a href="variant_association_subject/">variant_association_subject</a></td>
<td>Subject slot to be used for variant associations</td>
</tr>
<tr>
<td><a href="variant_genomic_location/">variant_genomic_location</a></td>
<td>VariantGenomicLocationAssociation for which consequences are calculated</td>
</tr>
<tr>
<td><a href="variant_genomic_location_association_ingest_set/">variant_genomic_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="variant_genomic_location_association_object/">variant_genomic_location_association_object</a></td>
<td>The location reference object should be a chromosome assembly curie.</td>
</tr>
<tr>
<td><a href="variant_identifier/">variant_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the variant</td>
</tr>
<tr>
<td><a href="variant_ingest_set/">variant_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="variant_locations/">variant_locations</a></td>
<td>Location of the variant within genomic entities. Variant_locations can include any or all of: one VariantGenomicLocationAssociation stanza, one or more VariantTranscriptLocationAssociation stanzas and/or one or more VariantPolypeptideLocationAssociation stanzas.</td>
</tr>
<tr>
<td><a href="variant_polypeptide_location_association_object/">variant_polypeptide_location_association_object</a></td>
<td>Polypeptide associated with variant and for which a specific location and consequence of that variant is provided, as specified by curator. Multivalued=false for this slot because although a variant can have multiple VariantPolypeptideLocationAssociation stanzas, each stanza will have one and only one curated polypeptide ID.</td>
</tr>
<tr>
<td><a href="variant_protein_location_association_ingest_set/">variant_protein_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="variant_sequence/">variant_sequence</a></td>
<td>Sequence that differs from the reference sequence of genome or genomic entity at position of variant, as specified by curator.</td>
</tr>
<tr>
<td><a href="variant_source_general_consequence/">variant_source_general_consequence</a></td>
<td>The general consequence of the variant on the transcript.</td>
</tr>
<tr>
<td><a href="variant_status/">variant_status</a></td>
<td>VocabularyTerm describing the status of the variant.  From the Variant Status CV</td>
</tr>
<tr>
<td><a href="variant_status_name/">variant_status_name</a></td>
<td>Name of the VocabularyTerm describing the status of the variant.  From the Variant Status CV</td>
</tr>
<tr>
<td><a href="variant_transcript/">variant_transcript</a></td>
<td>Transcript to which variant effect prediction applies</td>
</tr>
<tr>
<td><a href="variant_transcript_location_association_ingest_set/">variant_transcript_location_association_ingest_set</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="variant_transcript_location_object/">variant_transcript_location_object</a></td>
<td>Transcript associated with variant and for which a specific location and consequence of that variant is provided, as specified at source.  Multivalued=false for this slot because although a variant can have multiple VariantTranscriptLocationAssociation stanzas, each stanza will have one and only one source transcript ID.</td>
</tr>
<tr>
<td><a href="variant_type/">variant_type</a></td>
<td>SOTerm describing the type of variant. In practice, variant type will be limited to a subset of the SO specified in an Alliance controlled vocabulary in order to maintain consistency.</td>
</tr>
<tr>
<td><a href="variant_type_curie/">variant_type_curie</a></td>
<td>Curie of the SOTerm describing the type of variant. In practice, variant type will be limited to a subset of the SO specified in an Alliance controlled vocabulary in order to maintain consistency.</td>
</tr>
<tr>
<td><a href="variation_strand/">variation_strand</a></td>
<td>The strand on which the variant is located.  This is the strand of the reference sequence, not the variant sequence.</td>
</tr>
<tr>
<td><a href="vep_consequences/">vep_consequences</a></td>
<td>VEP consequence</td>
</tr>
<tr>
<td><a href="vep_impact/">vep_impact</a></td>
<td>VEP predicted impact of variation on molecule</td>
</tr>
<tr>
<td><a href="video_still/">video_still</a></td>
<td>An image represents a video still.</td>
</tr>
<tr>
<td><a href="vocabulary_description/">vocabulary_description</a></td>
<td>The free text description of a Vocabulary including its intended use.</td>
</tr>
<tr>
<td><a href="vocabulary_label/">vocabulary_label</a></td>
<td>Immutable label used for identification of vocabulary or vocabulary term set.  This is distinct from the "name" slot in that it cannot be changed once created.</td>
</tr>
<tr>
<td><a href="vocabulary_term_set_description/">vocabulary_term_set_description</a></td>
<td>The free text description of a VocabularyTermSet including its intended use.</td>
</tr>
<tr>
<td><a href="vocabulary_term_set_vocabulary/">vocabulary_term_set_vocabulary</a></td>
<td>The Vocabulary from which the terms contained in theVocabularyTermSet belong</td>
</tr>
<tr>
<td><a href="volume/">volume</a></td>
<td>Volume associated with a reference. From PubMed otherwise from Mod or manual reference creation.</td>
</tr>
<tr>
<td><a href="wbprocess/">wbprocess</a></td>
<td>WB specific. Eg WBbiop00000015 Corpse engulfment</td>
</tr>
<tr>
<td><a href="when_expressed/">when_expressed</a></td>
<td>When some entity is observed to be present in a specimen. If no temporal information is available for an annotation, use a root stage term for the developmental_stage_start.</td>
</tr>
<tr>
<td><a href="when_expressed_stage_name/">when_expressed_stage_name</a></td>
<td>A human-readable stage name. To be used when an ontology term for the relevant stage is not available.</td>
</tr>
<tr>
<td><a href="where_expressed/">where_expressed</a></td>
<td>Where some entity is observed to be present in a specimen. If no spatial information is available for an annotation, use a root anatomical term for the anatomical_structure.</td>
</tr>
<tr>
<td><a href="where_expressed_statement/">where_expressed_statement</a></td>
<td>A human-readable free-text description of the anatomical location of the expression pattern.</td>
</tr>
<tr>
<td><a href="width/">width</a></td>
<td>The width of the image (pixels).</td>
</tr>
<tr>
<td><a href="with_gene_identifiers/">with_gene_identifiers</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of genes http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#with-or-from-column-8</td>
</tr>
<tr>
<td><a href="with_or_from/">with_or_from</a></td>
<td>http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#with-or-from-column-8</td>
</tr>
<tr>
<td><a href="xls_file_path/">xls_file_path</a></td>
<td>File path of the Excel file generated from the BIRT report</td>
</tr>
<tr>
<td><a href="zygosity/">zygosity</a></td>
<td>GENO ontology term for allele zygosity</td>
</tr>
<tr>
<td><a href="zygosity_curie/">zygosity_curie</a></td>
<td>Curie of GENO ontology ID for allele zygosity - permissible_values: GENO:0000602 / GENO:0000603 / GENO:0000604 / GENO:0000605 / GENO:0000606 / GENO:0000135 / GENO:0000136 / GENO:0000137 / GENO:0000134</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="BackendBulkDataTypeEnum/">BackendBulkDataTypeEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="BackendBulkLoadTypeEnum/">BackendBulkLoadTypeEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="BulkLoadStatusEnum/">BulkLoadStatusEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="HighThroughputExpressionAssayValues/">HighThroughputExpressionAssayValues</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ModCorpusSortSourceEnum/">ModCorpusSortSourceEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="OntologyBulkLoadTypeEnum/">OntologyBulkLoadTypeEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="PubmedPublicationStatusEnum/">PubmedPublicationStatusEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="PubmedTypeEnum/">PubmedTypeEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="ReferenceCategoryEnum/">ReferenceCategoryEnum</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SampleTypeValues/">SampleTypeValues</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="SequencingFormatValues/">SequencingFormatValues</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="StrandEnum/">StrandEnum</a></td>
<td>Specify DNA strand.</td>
</tr>
<tr>
<td><a href="SubtypeValues/">SubtypeValues</a></td>
<td>None</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="AllianceSubset/">AllianceSubset</a></td>
<td>Subset consisting of just the alliance activities</td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": ".", "features": [], "search": "assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>