# AllianceModel

None

URI: https://github.com/alliance-genome/agr_curation_schema/alliance_schema

## Classes

| Class | Description |
| --- | --- |
| [AffectedGenomicModel](AffectedGenomicModel.md) | Includes inbred strains, stocks, disease models and mutant genotypes || --- | --- |
| [AffectedGenomicModelComponent](AffectedGenomicModelComponent.md) | Allele that affects the model and its zygosity || --- | --- |
| [AffectedGenomicModelComponentDTO](AffectedGenomicModelComponentDTO.md) | None || --- | --- |
| [AffectedGenomicModelDTO](AffectedGenomicModelDTO.md) | Ingest class for AGMs || --- | --- |
| [Agent](Agent.md) | An individual, group, organization or project that provides information and/or materials. || --- | --- |
| [AGMDiseaseAnnotation](AGMDiseaseAnnotation.md) | An annotation asserting an association between an AGM and a disease supported by evidence. || --- | --- |
| [AGMDiseaseAnnotationDTO](AGMDiseaseAnnotationDTO.md) | Ingest class for an association between an AGM and a disease || --- | --- |
| [AGMPhenotypeAnnotation](AGMPhenotypeAnnotation.md) | An annotation asserting an association between an AGM and a phenotype supported by evidence. || --- | --- |
| [Allele](Allele.md) | One of multiple possible forms of a functional genomic element (most often described as a locus or gene), differing from the reference DNA sequence.  The genomic element can be endogenous or constructed. || --- | --- |
| [AlleleAlleleAssociation](AlleleAlleleAssociation.md) | Association between an allele and another allele || --- | --- |
| [AlleleAlleleAssociationDTO](AlleleAlleleAssociationDTO.md) | Association between an allele and another allele || --- | --- |
| [AlleleCellLineAssociation](AlleleCellLineAssociation.md) | The relationship between an allele and a cell line.  Includes mutant/ embryonic stem cell lines known to carry the allele, and parental cell line of alleles made in embryonic stem cells. || --- | --- |
| [AlleleCellLineAssociationDTO](AlleleCellLineAssociationDTO.md) | The relationship between an allele and a cell line.  Includes mutant/ embryonic stem cell lines known to carry the allele, and parental cell line of alleles made in embryonic stem cells. || --- | --- |
| [AlleleConstructAssociation](AlleleConstructAssociation.md) | The relationship between an allele and constructs contained in that allele. || --- | --- |
| [AlleleConstructAssociationDTO](AlleleConstructAssociationDTO.md) | The relationship between an allele and constructs contained in that allele. || --- | --- |
| [AlleleDatabaseStatusSlotAnnotation](AlleleDatabaseStatusSlotAnnotation.md) | None || --- | --- |
| [AlleleDatabaseStatusSlotAnnotationDTO](AlleleDatabaseStatusSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleDiseaseAnnotation](AlleleDiseaseAnnotation.md) | An annotation asserting an association between an allele and a disease supported by evidence. || --- | --- |
| [AlleleDiseaseAnnotationDTO](AlleleDiseaseAnnotationDTO.md) | Ingest class for an association between an allele and a disease || --- | --- |
| [AlleleDTO](AlleleDTO.md) | Ingest class for an Allele object || --- | --- |
| [AlleleFullNameSlotAnnotation](AlleleFullNameSlotAnnotation.md) | The one current full name for the allele. || --- | --- |
| [AlleleFunctionalImpactSlotAnnotation](AlleleFunctionalImpactSlotAnnotation.md) | None || --- | --- |
| [AlleleFunctionalImpactSlotAnnotationDTO](AlleleFunctionalImpactSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleGeneAssociation](AlleleGeneAssociation.md) | Association between an allele and a gene || --- | --- |
| [AlleleGeneAssociationDTO](AlleleGeneAssociationDTO.md) | Association between an allele and a gene || --- | --- |
| [AlleleGenerationMethodAssociation](AlleleGenerationMethodAssociation.md) | None || --- | --- |
| [AlleleGenerationMethodAssociationDTO](AlleleGenerationMethodAssociationDTO.md) | None || --- | --- |
| [AlleleGenomicEntityAssociation](AlleleGenomicEntityAssociation.md) | Association between an allele and a genomic entity || --- | --- |
| [AlleleGenomicEntityAssociationDTO](AlleleGenomicEntityAssociationDTO.md) | None || --- | --- |
| [AlleleGermlineTransmissionStatusSlotAnnotation](AlleleGermlineTransmissionStatusSlotAnnotation.md) | None || --- | --- |
| [AlleleGermlineTransmissionStatusSlotAnnotationDTO](AlleleGermlineTransmissionStatusSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleImageAssociation](AlleleImageAssociation.md) | The relationship between an allele and an image. || --- | --- |
| [AlleleImageAssociationDTO](AlleleImageAssociationDTO.md) | The relationship between an allele and an image. || --- | --- |
| [AlleleInheritanceModeSlotAnnotation](AlleleInheritanceModeSlotAnnotation.md) | None || --- | --- |
| [AlleleInheritanceModeSlotAnnotationDTO](AlleleInheritanceModeSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleMolecularMutationSlotAnnotation](AlleleMolecularMutationSlotAnnotation.md) | None || --- | --- |
| [AlleleMolecularMutationSlotAnnotationDTO](AlleleMolecularMutationSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleMutationTypeSlotAnnotation](AlleleMutationTypeSlotAnnotation.md) | None || --- | --- |
| [AlleleMutationTypeSlotAnnotationDTO](AlleleMutationTypeSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleNomenclatureEventSlotAnnotation](AlleleNomenclatureEventSlotAnnotation.md) | None || --- | --- |
| [AlleleNomenclatureEventSlotAnnotationDTO](AlleleNomenclatureEventSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleNoteSlotAnnotation](AlleleNoteSlotAnnotation.md) | None || --- | --- |
| [AlleleNoteSlotAnnotationDTO](AlleleNoteSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleOriginAssociation](AlleleOriginAssociation.md) | The relationship between an allele and the origin of the allele. || --- | --- |
| [AlleleOriginAssociationDTO](AlleleOriginAssociationDTO.md) | The relationship between an allele and the AGM origin of the allele. || --- | --- |
| [AllelePhenotypeAnnotation](AllelePhenotypeAnnotation.md) | An annotation asserting an association between an allele and a phenotype supported by evidence. || --- | --- |
| [AlleleProteinAssociation](AlleleProteinAssociation.md) | Association between an allele and a protein || --- | --- |
| [AlleleProteinAssociationDTO](AlleleProteinAssociationDTO.md) | Association between an allele and a protein || --- | --- |
| [AlleleSecondaryIdSlotAnnotation](AlleleSecondaryIdSlotAnnotation.md) | None || --- | --- |
| [AlleleSecondaryIdSlotAnnotationDTO](AlleleSecondaryIdSlotAnnotationDTO.md) | None || --- | --- |
| [AlleleSymbolSlotAnnotation](AlleleSymbolSlotAnnotation.md) | The one current symbol for the allele. || --- | --- |
| [AlleleSynonymSlotAnnotation](AlleleSynonymSlotAnnotation.md) | All aliases (non-preferred names) for the allele. Any type of synonym is acceptable. || --- | --- |
| [AlleleTranscriptAssociation](AlleleTranscriptAssociation.md) | Association between an allele and a transcript || --- | --- |
| [AlleleTranscriptAssociationDTO](AlleleTranscriptAssociationDTO.md) | Association between an allele and a transcript || --- | --- |
| [AlleleVariantAssociation](AlleleVariantAssociation.md) | The relationship between an allele and a variant is many to many. An Allele may have many variants and a variant can be present in many alleles. || --- | --- |
| [AlleleVariantAssociationDTO](AlleleVariantAssociationDTO.md) | The relationship between an allele and a variant is many to many. An Allele may have many variants and a variant can be present in many alleles. || --- | --- |
| [AllianceMember](AllianceMember.md) | An organization that is a member of the Alliance of Genome Resources. || --- | --- |
| [AnatomicalSite](AnatomicalSite.md) | The developmental stage and/or age of the specimen in an annotation. || --- | --- |
| [AnatomicalTerm](AnatomicalTerm.md) | None || --- | --- |
| [Antibody](Antibody.md) | Immunoglobulin proteins that bind specific molecule(s). Can be used experimentally for the purposes of detection or purification. || --- | --- |
| [Assembly](Assembly.md) | None || --- | --- |
| [Association](Association.md) | A typed association between two entities, supported by evidence.  Associations have three base slots: subject, object, and predicate, but they can have any number of additional attributes that help qualify the relationship between the subject and the object.  The subject is the curie (or identifier) of the class that is the subject of the association, and likewise the object is the curie (or identifier of the class that is the object. The relationship between subject and object is defined by the predicate slot (which can also be constrained using the range of the predicate). || --- | --- |
| [ATPTerm](ATPTerm.md) | An ontology term from the Alliance Tags for Papers ontology (ATP) || --- | --- |
| [AuditedObject](AuditedObject.md) | Base class for all other LinkML classes. Some entity for which changes are tracked, including date/time of change, the agent responsible for the change, and whether the entity is internal (private). || --- | --- |
| [AuditedObjectDTO](AuditedObjectDTO.md) | Base class for all other LinkML DTO classes. || --- | --- |
| [AuthorReference](AuthorReference.md) | None || --- | --- |
| [BiologicalEntity](BiologicalEntity.md) | An entity of biological origin that can be unambiguously attributed to a single species. || --- | --- |
| [BiologicalEntityDTO](BiologicalEntityDTO.md) | None || --- | --- |
| [BulkFMSLoad](BulkFMSLoad.md) | This bulk load automatically pulls files from the BulkFMSLoad || --- | --- |
| [BulkLoad](BulkLoad.md) | Base class for all loads || --- | --- |
| [BulkLoadFile](BulkLoadFile.md) | This class is used to hold version of the files being loaded || --- | --- |
| [BulkLoadFileHistory](BulkLoadFileHistory.md) | Object used to describe the indiviual run of this BulkLoadFile || --- | --- |
| [BulkLoadGroup](BulkLoadGroup.md) | This class is use to group together bulk load_files || --- | --- |
| [BulkManualLoad](BulkManualLoad.md) | This bulk load is used by DQM's to submit their files to the curation system || --- | --- |
| [BulkScheduledLoad](BulkScheduledLoad.md) | This a bulk load that includes a cron scheuld || --- | --- |
| [BulkURLLoad](BulkURLLoad.md) | This bulk load automatically pulls files from a defined BulkURLLoad || --- | --- |
| [CellLine](CellLine.md) | Dummy cell line class || --- | --- |
| [CellLineDTO](CellLineDTO.md) | Dummy cell line DTO class || --- | --- |
| [CHEBITerm](CHEBITerm.md) | None || --- | --- |
| [ChemicalTerm](ChemicalTerm.md) | An ontology term representing a chemical or molecule || --- | --- |
| [Chromosome](Chromosome.md) | The ID of the landmark used to establish the coordinate system for the current feature. || --- | --- |
| [CLTerm](CLTerm.md) | None || --- | --- |
| [Company](Company.md) | None || --- | --- |
| [ConditionRelation](ConditionRelation.md) | A pairing of an experimental condition relation (i.e. has_condition) with a list of 1 or more ExperimentalCondition objects. Annotation objects can connect directly to a set of 0 or more of these ConditionRelation objects via a 'condition_relations' slot to express the experimental conditions relevant to the annotation. || --- | --- |
| [ConditionRelationDTO](ConditionRelationDTO.md) | Ingest class for the pairing of an experimental condition relation with a list of one or more conditions || --- | --- |
| [Construct](Construct.md) |  || --- | --- |
| [ConstructComponent](ConstructComponent.md) | None || --- | --- |
| [ConstructComponentAssociation](ConstructComponentAssociation.md) | The predicate should be a VocabularyTerm with one of the following values - expresses (RO:0002292) / is_regulated_by (RO:0002334) / targets (RO:0002436) || --- | --- |
| [ConstructDTO](ConstructDTO.md) | None || --- | --- |
| [CrossReference](CrossReference.md) | None || --- | --- |
| [CrossReferenceDTO](CrossReferenceDTO.md) | None || --- | --- |
| [CurationReport](CurationReport.md) | Base class for all curation reports || --- | --- |
| [CurationReportGroup](CurationReportGroup.md) | This class is use to group together curation reports || --- | --- |
| [CurationReportHistory](CurationReportHistory.md) | Object used to describe the indiviual run of this curation report || --- | --- |
| [DAOTerm](DAOTerm.md) | None || --- | --- |
| [DataProvider](DataProvider.md) | None || --- | --- |
| [DataProviderDTO](DataProviderDTO.md) | None || --- | --- |
| [DiseaseAnnotation](DiseaseAnnotation.md) | An annotation asserting an association between a biological entity and a disease supported by evidence. || --- | --- |
| [DiseaseAnnotationDTO](DiseaseAnnotationDTO.md) | Ingest class for association between a biological entity and a disease || --- | --- |
| [DNAClone](DNAClone.md) | None || --- | --- |
| [DOTerm](DOTerm.md) | None || --- | --- |
| [ECOTerm](ECOTerm.md) | None || --- | --- |
| [EMAPATerm](EMAPATerm.md) | None || --- | --- |
| [ExperimentalCondition](ExperimentalCondition.md) | The environmental context in which an experiment is carried out. This may (e.g. drug treatment) or may not (e.g. standard conditions) directly influence the outcome of the experiment. || --- | --- |
| [ExperimentalConditionDTO](ExperimentalConditionDTO.md) | Ingest class for describing the environmental context in which an experiment is carried out || --- | --- |
| [ExperimentalConditionOntologyTerm](ExperimentalConditionOntologyTerm.md) | None || --- | --- |
| [ExpressionAnnotation](ExpressionAnnotation.md) | A description of when and where gene products are observed to be present, including experimental details, supporting evidence, and curator notes. || --- | --- |
| [ExpressionAnnotationImagePane](ExpressionAnnotationImagePane.md) | None || --- | --- |
| [ExpressionExperiment](ExpressionExperiment.md) | Defined by the gene of interest, the specimen, the assay, the reagents (Antibody, Probe), and the reference. It groups ExpressionAnnotations. || --- | --- |
| [ExternalDatabaseLink](ExternalDatabaseLink.md) | Base relation that holds the identifier prefix, base url and url suffix to help in generating URLs in crossReferences. || --- | --- |
| [FBCVTerm](FBCVTerm.md) | None || --- | --- |
| [FBDVTerm](FBDVTerm.md) | None || --- | --- |
| [Figure](Figure.md) | An entity representing a figure or table in a publication. || --- | --- |
| [File](File.md) | A dummy object. || --- | --- |
| [FullNameSlotAnnotationDTO](FullNameSlotAnnotationDTO.md) | None || --- | --- |
| [FunctionalGeneSet](FunctionalGeneSet.md) | None || --- | --- |
| [Gene](Gene.md) | A DNA genomic entity from which one or more functional* RNA transcript molecules are transcribed, along with cis-regulatory elements responsible for regulating expression (transcription) of the gene. * A functional RNA molecule here can mean one that is directly responsible for the gene's function (e.g. catalysis, structure, etc.) or one that is translated to produce a functional polypeptide/protein. A pseudogene may be considered a gene under this definition, albeit no longer functional. || --- | --- |
| [GeneCluster](GeneCluster.md) | A gene cluster is a set of genes which have a biological significance, and which are probably clustered together on the genome, like histones and miRNAs || --- | --- |
| [GeneCollection](GeneCollection.md) | A gene collection is a set of genes which have been grouped based on experimental evidence, for example a set of interacting genes, genes in expression cluster, or a set of ChIP binding peaks || --- | --- |
| [GeneDiseaseAnnotation](GeneDiseaseAnnotation.md) | An annotation asserting an association between a gene and a disease supported by evidence. || --- | --- |
| [GeneDiseaseAnnotationDTO](GeneDiseaseAnnotationDTO.md) | Ingest class for an association between a gene and a disease || --- | --- |
| [GeneDTO](GeneDTO.md) | Ingest class for genes || --- | --- |
| [GeneFullNameSlotAnnotation](GeneFullNameSlotAnnotation.md) | The one current full name for the gene. || --- | --- |
| [GeneGeneticInteraction](GeneGeneticInteraction.md) | A genetic interaction between genes (e.g. phenotypic suppression) || --- | --- |
| [GeneHistory](GeneHistory.md) | The history of a gene || --- | --- |
| [GeneInteraction](GeneInteraction.md) | An interaction between two genes or gene products; this may be a physical molecular interaction between gene products (e.g. protein-protein interactions), or may be a genetic interaction between genes (e.g. phenotypic suppression) || --- | --- |
| [GeneMolecularInteraction](GeneMolecularInteraction.md) | A physical molecular interaction between gene products (e.g. protein-protein interactions or protein-RNA interactions) or between genes and DNA-binding factors (e.g. protein-DNA interactions) || --- | --- |
| [GeneNomenclatureSet](GeneNomenclatureSet.md) | WB specific. A gene class is a set of genes which share nomenclature, belonging to the same gene class. || --- | --- |
| [GenePhenotypeAnnotation](GenePhenotypeAnnotation.md) | An annotation asserting an association between a gene and a phenotype supported by evidence. || --- | --- |
| [GenerationMethod](GenerationMethod.md) | None || --- | --- |
| [GenerationMethodDTO](GenerationMethodDTO.md) | None || --- | --- |
| [GeneSymbolSlotAnnotation](GeneSymbolSlotAnnotation.md) | The one current symbol for the gene. || --- | --- |
| [GeneSynonymSlotAnnotation](GeneSynonymSlotAnnotation.md) | All aliases (non-preferred names) for the gene. Any type of synonym is acceptable. || --- | --- |
| [GeneSystematicNameSlotAnnotation](GeneSystematicNameSlotAnnotation.md) | The one current systematic name for the gene. || --- | --- |
| [GeneticMapPosition](GeneticMapPosition.md) | A genetic map position. || --- | --- |
| [GeneToGeneAssociation](GeneToGeneAssociation.md) | abstract parent class for different kinds of gene-gene or gene product to gene product relationships. Includes homology and interaction. || --- | --- |
| [GeneToGeneOrthology](GeneToGeneOrthology.md) |  || --- | --- |
| [GeneToGeneOrthologyCurated](GeneToGeneOrthologyCurated.md) | Class that holds the properties necessary to store a curated orthology record. || --- | --- |
| [GeneToGeneOrthologyGenerated](GeneToGeneOrthologyGenerated.md) | Class that holds the properties necessary to record an orthology record from DIOPT || --- | --- |
| [GeneToPathwayAssociation](GeneToPathwayAssociation.md) | None || --- | --- |
| [GenomicEntity](GenomicEntity.md) | An entity that is part of a genome (i.e. segment of the DNA molecule), is derived directly from the genome (i.e. RNA transcript molecule), or is derived indirectly from the genome (i.e. polypeptide or protein via RNA transcript translation). || --- | --- |
| [GenomicEntityDTO](GenomicEntityDTO.md) | None || --- | --- |
| [GenomicLocationAssociation](GenomicLocationAssociation.md) | None || --- | --- |
| [GenomicLocationAssociationDTO](GenomicLocationAssociationDTO.md) | None || --- | --- |
| [GOTerm](GOTerm.md) | None || --- | --- |
| [Identifier](Identifier.md) | None || --- | --- |
| [IdentifiersRange](IdentifiersRange.md) | None || --- | --- |
| [Image](Image.md) | The set of files and metadata that constitute an image. || --- | --- |
| [ImagePane](ImagePane.md) | Part of an Image that is relevant to some annotation. An annotation may reference many panes of an Image. || --- | --- |
| [InformationContentEntity](InformationContentEntity.md) | a piece of information that typically is used as support for an assertion or annotation. || --- | --- |
| [Ingest](Ingest.md) | None || --- | --- |
| [Laboratory](Laboratory.md) | None || --- | --- |
| [LoggedInPerson](LoggedInPerson.md) | None || --- | --- |
| [MATerm](MATerm.md) | None || --- | --- |
| [MeshDetail](MeshDetail.md) | Medical Subject Headings information coming from PubMed. || --- | --- |
| [MITerm](MITerm.md) | None || --- | --- |
| [MMOTerm](MMOTerm.md) | None || --- | --- |
| [MMUSDVTerm](MMUSDVTerm.md) | None || --- | --- |
| [ModCorpusAssociation](ModCorpusAssociation.md) | For a given reference and Mod, whether it is inside corpus, outside corpus, or needs review, as well as where this sorting came from. || --- | --- |
| [Molecule](Molecule.md) | Molecules as described by WormBase || --- | --- |
| [MPTerm](MPTerm.md) | None || --- | --- |
| [NameSlotAnnotation](NameSlotAnnotation.md) | Some symbol or name for an object, including current names as well as aliases, with accompanying metadata. The entity to which the symbol/name applies is specified in objects that inherit from this object. || --- | --- |
| [NameSlotAnnotationDTO](NameSlotAnnotationDTO.md) | None || --- | --- |
| [NCBITaxonTerm](NCBITaxonTerm.md) | None || --- | --- |
| [Note](Note.md) | Note object for capturing free-text describing some attribute of an entity, coupled with a 'note type', internal boolean, and an optional list of references. Permissible values for note_type can be viewed and managed in the A-Team curation UI Controlled Vocabulary Terms Table. || --- | --- |
| [NoteDTO](NoteDTO.md) | None || --- | --- |
| [OntologyTerm](OntologyTerm.md) | A concept or class in an ontology, vocabulary or thesaurus. Note, ontology terms can be found in multiple ontologies (ie: mireoting). defining_slots helps define an alternative key for ontology terms. || --- | --- |
| [OntologyTermClosure](OntologyTermClosure.md) | None || --- | --- |
| [Operon](Operon.md) | The DNA region of a group of adjacent genes whose transcription is coordinated on one or several mutually overlapping transcription units transcribed in the same direction and sharing at least one gene. || --- | --- |
| [Organization](Organization.md) | An organization that provides information and/or materials to the Alliance. This includes Alliance member organizations (see AllianceMember subclass). || --- | --- |
| [Pathway](Pathway.md) | None || --- | --- |
| [Person](Person.md) | None || --- | --- |
| [PersonalCommunication](PersonalCommunication.md) | a piece of information that is used to support an assertion or annotation, where the information comes from a person other than the author of the assertion or annotation, or the author of the reference. || --- | --- |
| [PhenotypeAnnotation](PhenotypeAnnotation.md) | An annotation asserting an association between a biological entity and a phenotype supported by evidence. || --- | --- |
| [PhenotypeTerm](PhenotypeTerm.md) | An ontology term representing a characteristic of an organism. This may or may not be expressed as a difference in comparison to a reference organism. || --- | --- |
| [Protein](Protein.md) | None || --- | --- |
| [ProteinComplex](ProteinComplex.md) | None || --- | --- |
| [Reagent](Reagent.md) | A material entity used in experiments. || --- | --- |
| [Reference](Reference.md) | None || --- | --- |
| [Resource](Resource.md) | None || --- | --- |
| [ResourceDescriptor](ResourceDescriptor.md) | None || --- | --- |
| [ResourceDescriptorPage](ResourceDescriptorPage.md) | None || --- | --- |
| [RNAClone](RNAClone.md) | None || --- | --- |
| [ROTerm](ROTerm.md) | None || --- | --- |
| [SequenceTargetingReagent](SequenceTargetingReagent.md) |  || --- | --- |
| [SequenceTargetingReagentDTO](SequenceTargetingReagentDTO.md) | None || --- | --- |
| [SequenceTargetingReagentToGeneAssociation](SequenceTargetingReagentToGeneAssociation.md) | the relationship between a Sequence Targeting Reagent and its targeted genes. The predicate should be a VocabularyTerm with one of the following values - targets || --- | --- |
| [SlotAnnotation](SlotAnnotation.md) | SlotAnnotation classes should be used when we need to attach metadata (in particular evidence and provenance) to a slot in the context of its referencing class, that can not be fully captured using an Association between the full class itself, and an InformationContentEntity. Evidence and provenance can exist here in the form of an evidence code, a publication, a personal communication or any other kind of InformationContentEntity. SlotAnnotation classes are used where the slot is not referencing a class in and of itself, and often has a scalar range. || --- | --- |
| [SlotAnnotationDTO](SlotAnnotationDTO.md) | None || --- | --- |
| [SOTerm](SOTerm.md) | None || --- | --- |
| [SourceVariantLocation](SourceVariantLocation.md) | Links a paper to the variant locations described in that paper || --- | --- |
| [StageTerm](StageTerm.md) | None || --- | --- |
| [SymbolSlotAnnotationDTO](SymbolSlotAnnotationDTO.md) | None || --- | --- |
| [SystematicNameSlotAnnotationDTO](SystematicNameSlotAnnotationDTO.md) | None || --- | --- |
| [TemporalContext](TemporalContext.md) | The developmental stage and/or age of the specimen in an annotation. Developmental_stage_stop is optional. Add an uncertainty flag here? || --- | --- |
| [Transcript](Transcript.md) | Placeholder. || --- | --- |
| [UBERONTerm](UBERONTerm.md) | None || --- | --- |
| [UniGeneSet](UniGeneSet.md) | XenBase-specific. A set of three genes from X. tropicalis and X. laevis (S and L forms) representing a single unigene. || --- | --- |
| [Variant](Variant.md) | A DNA, RNA or protein/polypeptide sequence that differs relative to a designated reference sequence.  The sequence occurs at a single position or in a range of contiguous nucleotides or amino acids. || --- | --- |
| [VariantConsequence](VariantConsequence.md) | Parent class for gene- and transcript-level consequences || --- | --- |
| [VariantGeneConsequence](VariantGeneConsequence.md) | Class for gene-level VEP results || --- | --- |
| [VariantGenomeLocation](VariantGenomeLocation.md) | Links a variant to a genomic position and the resulting consequence to the sequence and/or function. In practice, functional consequences for variants which overlap genes are not generally provided at the genome level but rather are calculated and annotated relative to a specific transcript or protein isoform. || --- | --- |
| [VariantLocation](VariantLocation.md) | Base class linking a variant to a position on a genomic entity and the resulting consequence to the sequence and/or function of that genomic entity. Slots are provided for data taken from a source publication or data load and for data resulting from manual curation. Where the values are the same, the curator has confirmed the information from the source.  In other cases, the curator's analysis has resulted in different values, for instance, if the assembly is different, the source did not specify the transcript or protein isoform, the definition of the transcript or protein isoform used by the source has changed, or if there was an error in the source data. || --- | --- |
| [VariantPolypeptideLocation](VariantPolypeptideLocation.md) | Links a variant to a position on a specified polypeptide and the resulting consequence to the sequence and/or function of that polypeptide. || --- | --- |
| [VariantTranscriptConsequence](VariantTranscriptConsequence.md) | Class for transcript-level VEP results || --- | --- |
| [VariantTranscriptLocation](VariantTranscriptLocation.md) | Links a variant to a position on a specified transcript and the resulting consequence to the sequence and/or function of that transcript. || --- | --- |
| [Vocabulary](Vocabulary.md) | A set of VocabularyTerm objects. || --- | --- |
| [VocabularyTerm](VocabularyTerm.md) | A concept or class in a simple vocabulary. || --- | --- |
| [VocabularyTermSet](VocabularyTermSet.md) | A subset of terms from a Vocabulary that are valid for particular applications || --- | --- |
| [WBBTTerm](WBBTTerm.md) | None || --- | --- |
| [WBLSTerm](WBLSTerm.md) | None || --- | --- |
| [XBATerm](XBATerm.md) | The Xenbase anatomy ontology XAO is home to Xenopus anatomy terms as well as Xenopus developmental life stage terms, differentiated by namespace. The anatomy term class in LinkML will be named 'XBATerm' for Xenbase Anatomy Term and life stage terms will be named 'XBSTerm' for Xenbase Stage Term. || --- | --- |
| [XBEDTerm](XBEDTerm.md) | None || --- | --- |
| [XBSTerm](XBSTerm.md) | The Xenbase anatomy ontology XAO is home to Xenopus anatomy terms as well as Xenopus developmental life stage terms, differentiated by namespace. The anatomy term class in LinkML will be named 'XBATerm' for Xenbase Anatomy Term and life stage terms will be named 'XBSTerm' for Xenbase Stage Term. || --- | --- |
| [XCOTerm](XCOTerm.md) | None || --- | --- |
| [XPOTerm](XPOTerm.md) | None || --- | --- |
| [XSMOTerm](XSMOTerm.md) | None || --- | --- |
| [ZECOTerm](ZECOTerm.md) | None || --- | --- |
| [ZFATerm](ZFATerm.md) | None || --- | --- |
| [ZFSTerm](ZFSTerm.md) | None |

## Slots

| Property | Description |
| --- | --- |
| [abbreviation](abbreviation.md) | None |
| [aberration](aberration.md) | Associated deficiency (etc.) whose breakpoint causes the mutation |
| [abstract](abstract.md) | The author summary of the publication. From PubMed otherwise from Mod or manual reference creation. |
| [acquires_merge](acquires_merge.md) | Genes which have been merged into this gene |
| [affiliated_alliance_member](affiliated_alliance_member.md) | The Alliance Member the person is affiliated with |
| [age](age.md) | The age at which an annotated event was observed. |
| [aggregation_database](aggregation_database.md) | None |
| [agm_curie](agm_curie.md) | None |
| [agm_ingest_set](agm_ingest_set.md) |  |
| [allele_allele_associations](allele_allele_associations.md) | None |
| [allele_cell_line_association_ingest_set](allele_cell_line_association_ingest_set.md) |  |
| [allele_cell_line_associations](allele_cell_line_associations.md) | None |
| [allele_construct_association_ingest_set](allele_construct_association_ingest_set.md) |  |
| [allele_construct_associations](allele_construct_associations.md) | None |
| [allele_curie](allele_curie.md) | None |
| [allele_database_status](allele_database_status.md) | Database status of a given allele |
| [allele_database_status_dto](allele_database_status_dto.md) | None |
| [allele_full_name](allele_full_name.md) | The one current full name for an allele: e.g., wg<sup>1</sup>. |
| [allele_full_name_dto](allele_full_name_dto.md) | The one current full name for an allele: e.g., wg<sup>1</sup>. |
| [allele_functional_impact_dtos](allele_functional_impact_dtos.md) | None |
| [allele_functional_impacts](allele_functional_impacts.md) | Functional impacts of a given allele |
| [allele_gene_association_ingest_set](allele_gene_association_ingest_set.md) |  |
| [allele_gene_associations](allele_gene_associations.md) | None |
| [allele_generation_method_association_ingest_set](allele_generation_method_association_ingest_set.md) |  |
| [allele_generation_method_associations](allele_generation_method_associations.md) | None |
| [allele_genomic_location_association_ingest_set](allele_genomic_location_association_ingest_set.md) | An ingest set of genomic locations DTOs that are associated with an allele |
| [allele_germline_transmission_status](allele_germline_transmission_status.md) | Germline transmission status for a given allele |
| [allele_germline_transmission_status_dto](allele_germline_transmission_status_dto.md) | None |
| [allele_image_association_ingest_set](allele_image_association_ingest_set.md) |  |
| [allele_image_associations](allele_image_associations.md) | None |
| [allele_ingest_set](allele_ingest_set.md) |  |
| [allele_inheritance_mode_dtos](allele_inheritance_mode_dtos.md) | One or more allele inheritance mode DTO objects to be submitted |
| [allele_inheritance_modes](allele_inheritance_modes.md) | Inheritance modes for an allele |
| [allele_molecular_mutation_dtos](allele_molecular_mutation_dtos.md) | None |
| [allele_molecular_mutations](allele_molecular_mutations.md) | Molecular mutations of a given allele |
| [allele_mutation_type_dtos](allele_mutation_type_dtos.md) | None |
| [allele_mutation_types](allele_mutation_types.md) | Mutation types for a given allele |
| [allele_nomenclature_event_dtos](allele_nomenclature_event_dtos.md) | None |
| [allele_nomenclature_events](allele_nomenclature_events.md) | Nomenclature events of a given allele |
| [allele_note_dtos](allele_note_dtos.md) | None |
| [allele_notes](allele_notes.md) | Notes for a given allele |
| [allele_origin_association_ingest_set](allele_origin_association_ingest_set.md) |  |
| [allele_origin_associations](allele_origin_associations.md) | None |
| [allele_protein_association_ingest_set](allele_protein_association_ingest_set.md) |  |
| [allele_protein_associations](allele_protein_associations.md) | None |
| [allele_secondary_id_dtos](allele_secondary_id_dtos.md) | None |
| [allele_secondary_ids](allele_secondary_ids.md) | Secondary IDs of a given allele |
| [allele_symbol](allele_symbol.md) | The one current accepted symbol for the allele: e.g., wg<sup>1</sup>. |
| [allele_symbol_dto](allele_symbol_dto.md) | The one current accepted symbol for the allele: e.g., wg<sup>1</sup>. |
| [allele_synonym_dtos](allele_synonym_dtos.md) | None |
| [allele_synonyms](allele_synonyms.md) | Holds between an Allele and a synonym. |
| [allele_transcript_association_ingest_set](allele_transcript_association_ingest_set.md) |  |
| [allele_transcript_associations](allele_transcript_associations.md) | None |
| [allele_variant_association_ingest_set](allele_variant_association_ingest_set.md) |  |
| [allele_variant_associations](allele_variant_associations.md) | None |
| [alliance_member_id](alliance_member_id.md) | An integer referring to an AllianceMember object in the AllianceMember/MOD database table. It's a primary key in the AllianceMember/MOD table, a foreign key if used in other tables. |
| [amino_acid_reference](amino_acid_reference.md) | reference genome amino acid sequence at variant position |
| [amino_acid_variant](amino_acid_variant.md) | variant amino acid sequence at variant position |
| [analyses](analyses.md) | TODO: added as a result of 'origin' class - please advise on a more descriptive slot/class definition. |
| [anatomical_structure](anatomical_structure.md) | None |
| [anatomical_substructure](anatomical_substructure.md) | None |
| [anatomy_function](anatomy_function.md) | WB specific. Allow the connection between Anatomy_term, Phenotype and gene Eg WBbtf0001 |
| [ancestors](ancestors.md) | The ancestors of this term in the ontology, including the term itself. This language works well for the majority of use cases, however for a relationship like "develops_from", ancestors are the terms from which this term develops (not a true parent/child or ancestor/descendant relationship). |
| [annotation_type](annotation_type.md) | The type of annotation classified according to curation method. Submitted value should be a vocabulary term from the 'Annotation types' vocabulary |
| [annotation_type_name](annotation_type_name.md) | Name of the VocabularyTerm describing the annotation type selected from the 'Annotation types' Vocabulary |
| [antibody_target_genes](antibody_target_genes.md) | The genes whose gene products are recognized by the antibody. |
| [antigen_taxon](antigen_taxon.md) | Holds between an Antibody and the species from which the antigen originates (as represented by taxon). |
| [api_token](api_token.md) | The API token of the LoggedInPerson for the curation system |
| [assay_notes](assay_notes.md) | None |
| [assay_used](assay_used.md) | The assay used to experimentally determine gene expression. |
| [assembly](assembly.md) | Assembly to which variant is aligned. |
| [assembly_curie](assembly_curie.md) | None |
| [asserted_allele](asserted_allele.md) | The allele to which something is manually asserted to be associated. |
| [asserted_allele_curie](asserted_allele_curie.md) | Curie of the allele to which something is manually asserted to be associated |
| [asserted_gene_curies](asserted_gene_curies.md) | Curies of the gene(s) to which something is manually asserted to be associated |
| [asserted_genes](asserted_genes.md) | The gene(s) to which something is manually asserted to be associated. |
| [associated_transcripts](associated_transcripts.md) | Transcript(s) associated with polypeptide to which variant is aligned. |
| [associated_with_figure](associated_with_figure.md) | Holds between an entity and a Figure. |
| [association_slot](association_slot.md) | any slot that relates an association to another entity |
| [authors](authors.md) | Ordered author entities for this publication.  An Author is associated with only one publication.  A Person can be associated with multiple publications. |
| [backend_bulk_load_type](backend_bulk_load_type.md) | None |
| [backend_load_type](backend_load_type.md) | None |
| [belongs_to_expression_experiment](belongs_to_expression_experiment.md) | Holds between an ExpressionAnnotation and an ExpressionExperiment. |
| [biological_entity_assayed](biological_entity_assayed.md) | Holds between a BiologicalEntity and an ExpressionExperiment that reports on its expression. |
| [birt_report_file_path](birt_report_file_path.md) | File path to where the BIRT file has been saved |
| [bulk_load](bulk_load.md) | A link back to the parent bulk load |
| [bulkload_group](bulkload_group.md) | Bulk load group designed to group together bulk loads |
| [bulkload_status](bulkload_status.md) | Status used to capture the progress of the load |
| [bulkload_url](bulkload_url.md) | The URL that this bulk load will download for ingest |
| [caption](caption.md) | Text describing the contents of a figure or table in a publication. |
| [category](category.md) | The alliance category type.  Only relevant at Alliance. |
| [cdna_end](cdna_end.md) | end position of variation in cDNA coordinates |
| [cdna_start](cdna_start.md) | start position of variation in cDNA coordinates |
| [cds_end](cds_end.md) | end position of variation in CDS coordinates |
| [cds_start](cds_start.md) | start position of variation in CDS coordinates |
| [cell_line_curie](cell_line_curie.md) | None |
| [cellular_component](cellular_component.md) | None |
| [chemical_mutagen](chemical_mutagen.md) | The chemical used to generate the mutation through mutagenesis |
| [chemical_mutagen_name](chemical_mutagen_name.md) | The name of the chemical used to generate the mutation through mutagenesis |
| [chromosome](chromosome.md) | Chromosome to which variant is aligned. |
| [chromosome_curie](chromosome_curie.md) | None |
| [clonality](clonality.md) | The clonality of the antibody: e.g., monoclonal. |
| [codon_reference](codon_reference.md) | reference sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cTa) |
| [codon_variant](codon_variant.md) | variant sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cAa) |
| [completed_records](completed_records.md) | The number of completed records |
| [component_dtos](component_dtos.md) | None |
| [components](components.md) | Collection of genomic components that make up a model, i.e. 'allele', each with a zygosity |
| [condition_anatomy](condition_anatomy.md) | None |
| [condition_anatomy_curie](condition_anatomy_curie.md) | Curie of AnatomicalTerm associated with condition |
| [condition_chemical](condition_chemical.md) | None |
| [condition_chemical_curie](condition_chemical_curie.md) | Curie of ChemicalTerm associated with condition |
| [condition_class](condition_class.md) | None |
| [condition_class_curie](condition_class_curie.md) | Curie of ZECOTerm describing condition class |
| [condition_dtos](condition_dtos.md) | None |
| [condition_free_text](condition_free_text.md) | Free-text description of the experimental condition |
| [condition_gene_ontology](condition_gene_ontology.md) | None |
| [condition_gene_ontology_curie](condition_gene_ontology_curie.md) | Curie of GOTerm associated with condition |
| [condition_id](condition_id.md) | None |
| [condition_id_curie](condition_id_curie.md) | Curie of ExperimentalConditionOntologyTerm describing condition |
| [condition_quantity](condition_quantity.md) | None |
| [condition_relation_dtos](condition_relation_dtos.md) | None |
| [condition_relation_type](condition_relation_type.md) | Submitted value should be a vocabulary term from the 'Condition relation types' vocabulary |
| [condition_relation_type_name](condition_relation_type_name.md) | Name of VocabularyTerm from 'Condition relation types' vocabulary |
| [condition_relations](condition_relations.md) | None |
| [condition_summary](condition_summary.md) | A human-readable summary of the experimental condition accounting for all ontology terms and free-text components of the ExperimentalCondition object. Not generated or submitted by DQMs, but generated at the Alliance after ingest only (or generated/updated in the curation interface). |
| [condition_taxon](condition_taxon.md) | None |
| [condition_taxon_curie](condition_taxon_curie.md) | Curie of NCBITaxonTerm associated with condition |
| [conditions](conditions.md) | None |
| [confidence](confidence.md) | None |
| [consequence](consequence.md) | SOTerm (child of SO:0001576 - transcript_variant) that describes the consequence of the variant, as stated in the source reference. In practice source consequence will be associated with locations at any or all of VariantGenomeLocation, VariantTranscriptLocation, and VariantPolypeptideLocation. |
| [construct_component_dtos](construct_component_dtos.md) | None |
| [construct_components](construct_components.md) | None |
| [construct_curie](construct_curie.md) | None |
| [contig](contig.md) | Contig or clone this gene is located to |
| [copyright_date](copyright_date.md) |  |
| [corpus](corpus.md) | in-out-review corpus status of the publication at the mod. null means needs_review, false means outside_corpus,  true means inside_corpus. |
| [corresponding_author](corresponding_author.md) | Indicates if the author is a corresponding author. |
| [counter](counter.md) | a number to identify an alliance resource in a subdomain |
| [created_by](created_by.md) | The individual that created the entity. |
| [created_by_curie](created_by_curie.md) | Curie of the Person object representing the individual that created the entity |
| [cron_schedule](cron_schedule.md) | A string describing the cron syntax for the schedule |
| [cropped_from](cropped_from.md) | Another larger image from which this image was cropped. |
| [cross_reference](cross_reference.md) | None |
| [cross_reference_dto](cross_reference_dto.md) | None |
| [cross_reference_dtos](cross_reference_dtos.md) | None |
| [cross_references](cross_references.md) | Holds between an object and its CrossReferences. |
| [curated_consequence](curated_consequence.md) | SOTerm that describes the consequence of the variant, as determined by the curator. In practice curated consequence will be associated with locations at any or all of VariantGenomeLocation, VariantTranscriptLocation, and VariantPolypeptideLocation. |
| [curation_report](curation_report.md) | Links a report history to its associated report |
| [curation_report_group](curation_report_group.md) | Links a curation report to its report group |
| [curation_report_status](curation_report_status.md) | Describes the status of the curation report |
| [curation_report_timestamp](curation_report_timestamp.md) | Timestamp for creation of BIRT report file |
| [curator_comment](curator_comment.md) | A publicly visible explanatory note from curators about some entity. |
| [curie](curie.md) | A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI |
| [current](current.md) | None |
| [current_status](current_status.md) | Current status of this object |
| [current_version](current_version.md) | Current version of this object with each version eg 1,2,3 describing major updates |
| [data_provider](data_provider.md) | Object representing the organization (e.g. MOD) from which the data was sourced and a CrossReference to that organisation's site |
| [data_provider_dto](data_provider_dto.md) | Ingest object representing the organization (e.g. MOD) from which the data was sourced and a CrossReference to that organisation's site |
| [database_status](database_status.md) | Database status of the allele |
| [database_status_name](database_status_name.md) | Name of the VocabularyTerm describing the database status |
| [date_arrived_in_pubmed](date_arrived_in_pubmed.md) | Day in which a reference or resource was created in PubMed. Only relevant for PubMed references. |
| [date_created](date_created.md) | The date on which an entity was created. This can be applied to nodes or edges. |
| [date_last_loaded](date_last_loaded.md) | Indicates when the file was last loaded |
| [date_last_modified_in_pubmed](date_last_modified_in_pubmed.md) | Date on which entity was last updated at PubMed. Only relevant for PubMed references. |
| [date_published](date_published.md) | Date on which an entity was published.  From PubMed otherwise from Mod or manual reference creation. |
| [date_updated](date_updated.md) | Date on which an entity was last modified. |
| [db_date_created](db_date_created.md) | The date on which an entity was created in the Alliance database.  This is disinct from date_created, which represents the date when the entity was originally created (i.e. at the MOD for imported data). |
| [db_date_updated](db_date_updated.md) | Date on which an entity was last modified in the Alliance database.  This is disinct from date_updated, which represents the date when the entity was last modified and may predate import into the Alliance database. |
| [dbkey](dbkey.md) | Typically the primary key on the table.  Should be a global sequence in the database to insure uniqueness over the entire suite of tables.  Alternatively, could be a serial8 identifier. Tables with a dbkey should have an alternate key to establish uniqueness based on the data in the table. |
| [default_url_template](default_url_template.md) | Default URL template for the resource |
| [definition](definition.md) | The explanation of the meaning of a term. |
| [definition_urls](definition_urls.md) | None |
| [descendants](descendants.md) | The descendants of this term in the ontology, including the term itself. This language works well for the majority of use cases, however for a relationship like "develops_from", ancestors are the terms to which this term develops into (not a true parent/child or ancestor/descendant relationship). |
| [description](description.md) | a human-readable description of an entity |
| [designating_laboratories](designating_laboratories.md) | A laboratory, rarely laboratories, which designated this gene |
| [designating_laboratory](designating_laboratory.md) | A laboratory which designated this gene class |
| [designating_persons](designating_persons.md) | A person/persons who designated this gene |
| [detection_method](detection_method.md) | None |
| [developmental_stage_start](developmental_stage_start.md) | The beginning developmental stage at which an annotated event was observed. |
| [developmental_stage_stop](developmental_stage_stop.md) | The end developmental stage at which an annotated event was observed. |
| [disease_agm_ingest_set](disease_agm_ingest_set.md) |  |
| [disease_allele_ingest_set](disease_allele_ingest_set.md) |  |
| [disease_gene_ingest_set](disease_gene_ingest_set.md) |  |
| [disease_genetic_modifier](disease_genetic_modifier.md) | Specifies a genetic object that modifies the disease model. May be a gene, allele, AGM. |
| [disease_genetic_modifier_curie](disease_genetic_modifier_curie.md) | Curie of BiologcalEntity that modifies the disease model |
| [disease_genetic_modifier_relation](disease_genetic_modifier_relation.md) | A relation describing how the genetic modifier modifies the disease model. Submitted value should be a vocabulary term from the 'Disease genetic modifiers' vocabulary |
| [disease_genetic_modifier_relation_name](disease_genetic_modifier_relation_name.md) | Name of the VocabularyTerm that describes how the genetic modifier modifies the disease model, selected from the 'Disease genetic modifier relations' Vocabulary. |
| [disease_qualifier_names](disease_qualifier_names.md) | Names of terms from the 'Disease qualifiers' vocabulary |
| [disease_qualifiers](disease_qualifiers.md) | Submitted values should be vocabulary terms from the 'Disease qualifiers' Vocabulary |
| [disease_relation_name](disease_relation_name.md) | Name of term from 'Disease Relation Vocabulary' vocabulary |
| [display_name](display_name.md) | None |
| [display_synonym](display_synonym.md) | Placeholder. |
| [display_text](display_text.md) | A version of a synonym string for display. Any UTF8 character is permitted. |
| [distance_between](distance_between.md) | None |
| [do_term_curie](do_term_curie.md) | Curie of DOTerm describing the disease |
| [editors](editors.md) | holds between a resource and a editor_resource |
| [emails](emails.md) | list of emails for a person |
| [enables](enables.md) | None |
| [end](end.md) | The end of the feature in positive 1-based integer coordinates |
| [end_position](end_position.md) | End position of variant on genomic entity. |
| [error_message](error_message.md) | Error message string if an error occurs |
| [evidence](evidence.md) |  |
| [evidence_code](evidence_code.md) | None |
| [evidence_code_curie](evidence_code_curie.md) | Curie of ECOTerm |
| [evidence_code_curies](evidence_code_curies.md) | List of ECOTerm curies |
| [evidence_codes](evidence_codes.md) | ECO term IDs |
| [evidence_curies](evidence_curies.md) | Curies of InformationContentEntity objects given as evidence |
| [experiment_type](experiment_type.md) | Type of experiment by which these genes were collated eg chip-seq, interaction, expression |
| [expression_qualifiers](expression_qualifiers.md) | Qualifiers that describe additional characteristics of gene expression. For example: aint, intense, restricted. |
| [failed_records](failed_records.md) | The number of failed records |
| [file_size](file_size.md) | The size of the file |
| [first](first.md) | first identifier in a range |
| [first_author](first_author.md) | Indicates if the author is a first author. |
| [first_name](first_name.md) | first name of a person |
| [fms_data_sub_type](fms_data_sub_type.md) | The dataSubType paramater in the FMS |
| [fms_data_type](fms_data_type.md) | The dataType paramater in the FMS |
| [format_text](format_text.md) | A version of a synonym string using only ASCII characters, which is easier to type (for searches), print and parse. For example, Greek characters are transliterated. |
| [formula](formula.md) | Formula of the molecule |
| [free_text](free_text.md) | A free text string that describes some aspect of an entity. |
| [from_image](from_image.md) | Holds between an ImagePane and an Image. |
| [full_name](full_name.md) | The full name of the Organization. e.g. Mouse Genome Database, FlyBase, Online Mendelian Inheritance of Man |
| [functional_impact_names](functional_impact_names.md) | Name of the VocabularyTerm describing the experimentally assessed functional impact of the allele, e.g. knockout / amorphic |
| [functional_impacts](functional_impacts.md) | Experimentally assessed functional impact of the allele, e.g. knockout / amorphic |
| [gene_curie](gene_curie.md) | None |
| [gene_full_name](gene_full_name.md) | The one current full name for a gene: e.g., wingless, paired box 2a. |
| [gene_full_name_dto](gene_full_name_dto.md) | The one current full name for a gene: e.g., wingless, paired box 2a. |
| [gene_ingest_set](gene_ingest_set.md) |  |
| [gene_object](gene_object.md) | None |
| [gene_subject](gene_subject.md) | None |
| [gene_symbol](gene_symbol.md) | The one current accepted symbol for a gene: e.g., wg, pax2a. |
| [gene_symbol_dto](gene_symbol_dto.md) | The one current accepted symbol for a gene: e.g., wg, pax2a. |
| [gene_synonym_dtos](gene_synonym_dtos.md) | None |
| [gene_synonyms](gene_synonyms.md) | Holds between a Gene and a synonym. |
| [gene_systematic_name](gene_systematic_name.md) | The one current systematic name for a gene: e.g., YHR084W, R09F10.2. |
| [gene_systematic_name_dto](gene_systematic_name_dto.md) | The one current systematic name for a gene: e.g., YHR084W, R09F10.2. |
| [gene_type](gene_type.md) | SOTerm describing gene type |
| [gene_type_curie](gene_type_curie.md) | Curie of SOTerm describing gene type |
| [gene_types_secondary](gene_types_secondary.md) | SOTerm describing alternate gene types |
| [generated_by](generated_by.md) | Holds between a material entity and an Agent that generated it: e.g., Thomas Blumenthal, Kornberg Laboratory. |
| [generation_method_dto](generation_method_dto.md) | None |
| [genes](genes.md) | None |
| [genetic_map_band](genetic_map_band.md) | Genetic map predicted chromosome location eg 10q12 |
| [genetic_map_chromosome](genetic_map_chromosome.md) | Chromosome or contig to which the gene has been genetically mapped |
| [genetic_map_position_centimorgan](genetic_map_position_centimorgan.md) | Genetic map predicted chromosome location in centimorgans eg 22.3366 cM |
| [genetic_map_position_centimorgan_error](genetic_map_position_centimorgan_error.md) | Genetic map calculated error in the predicted chromosome location in centimorgans eg 0.045 cM |
| [genetic_map_position_radiation](genetic_map_position_radiation.md) | Radiation hybrid map predicted chromosome location eg 66.5 cR |
| [genetic_sex](genetic_sex.md) | Submitted value should be a vocabulary term from the 'Genetic sexes' vocabulary |
| [genetic_sex_name](genetic_sex_name.md) | Name of term from the 'Genetic sexes' vocabulary |
| [genetically_interacts_with](genetically_interacts_with.md) | holds between two genes whose phenotypic effects are dependent on each other in some way - such that their combined phenotypic effects are the result of some interaction between the activity of their gene products. Examples include epistasis and synthetic lethality. |
| [genomic_entity_curie](genomic_entity_curie.md) | None |
| [genomic_location_association_dtos](genomic_location_association_dtos.md) | None |
| [genomic_location_associations](genomic_location_associations.md) | None |
| [germline_transmission_status](germline_transmission_status.md) | For alleles made in cell lines, have they been transmitted to the germline of an animal |
| [germline_transmission_status_name](germline_transmission_status_name.md) | Name of the VocabularyTerm representing the germline transmission status |
| [handle](handle.md) | A slot pointing to a free-text alias or 'handle' for a data object, such as a reference-specific alias for a data object used while curating. |
| [has_allele](has_allele.md) | None |
| [has_assembly](has_assembly.md) | None |
| [has_input](has_input.md) | None |
| [has_output](has_output.md) | None |
| [has_participant](has_participant.md) | None |
| [heading_term](heading_term.md) | The MeSH term description or definition from PubMed. e.g. Measles in a Measles/epidemiology term. |
| [heavy_chain_isotype](heavy_chain_isotype.md) | The isotype of the antibody heavy chain: e.g., IgA. |
| [height](height.md) | The height of the image (pixels). |
| [hgvs](hgvs.md) | HGVS nomenclature for variant located on a genomic entity. In practice HGVS nomenclature will be derived for any or all of VariantGenomeLocation, VariantTranscriptLocation, and VariantPolypeptideLocation. |
| [hgvs_coding_nomenclature](hgvs_coding_nomenclature.md) | HGVSc nomenclature for variation in coding sequence |
| [hgvs_protein_nomenclature](hgvs_protein_nomenclature.md) | HGVSp nomenclature for variation in protein |
| [homepage_resource_descriptor_page](homepage_resource_descriptor_page.md) | ResourceDescriptorPage containing URL template for organization's homepage |
| [html_file_path](html_file_path.md) | File path of the HTML file generated from the BIRT report |
| [id](id.md) | Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise. |
| [id_example](id_example.md) | Example ID for the resource that fits the pattern described by the id_pattern slot |
| [id_pattern](id_pattern.md) | Regex for the expected format of the ID |
| [image](image.md) | The image associated with the object. |
| [image_curie](image_curie.md) | None |
| [image_file](image_file.md) | The file representing the full-sized version of this image. |
| [image_medium_file](image_medium_file.md) | The file representing a medium-sized version of this image. |
| [image_thumbnail_file](image_thumbnail_file.md) | The file representing the thumbnail of this image. |
| [image_x_origin](image_x_origin.md) | The x coordinate start point when specifying a spatial section within a larger image. |
| [image_y_origin](image_y_origin.md) | The y coordinate start point when specifying a spatial section within a larger image. |
| [images](images.md) | Any associated image |
| [in_collection](in_collection.md) | Set of high-throughput alleles made by large projects |
| [in_collection_name](in_collection_name.md) | Name of VocabularyTerm describing the collection from the 'Allele collection vocabulary' Vocabulary |
| [inchi](inchi.md) | InChi style description of the molecule |
| [inchi_key](inchi_key.md) | InChi key description of the molecule |
| [inferred_allele](inferred_allele.md) | The allele to which something is inferred to be associated. |
| [inferred_allele_curie](inferred_allele_curie.md) | Curie of allele to which something is inferred to be associated via an automated pipeline |
| [inferred_gene](inferred_gene.md) | The gene to which something is inferred to be associated. |
| [inferred_gene_curie](inferred_gene_curie.md) | Curie of gene to which something is inferred to be associated via an automated pipeline |
| [inheritance_mode](inheritance_mode.md) | Mode of inheritance, e.g. dominant / semi-dominant / recessive / unknown / codominant |
| [inheritance_mode_name](inheritance_mode_name.md) | Name of VocabularyTerm describing the inheritance mode from the 'Allele inheritance mode vocabulary', e.g. dominant / semi-dominant / recessive / unknown / codominant |
| [integration_method](integration_method.md) | WormBase captures the method by which an extrachromosomal transgene was integrated into the genome. |
| [integration_method_name](integration_method_name.md) | WormBase captures the method by which an extrachromosomal transgene was integrated into the genome. |
| [interaction_data_provider](interaction_data_provider.md) | The interaction database that curated the interaction. e.g. BioGRID |
| [interaction_type](interaction_type.md) | The type of interaction between the two genes or gene products. e.g. physical association |
| [interactor_A_genetic_perturbation](interactor_A_genetic_perturbation.md) | None |
| [interactor_A_role](interactor_A_role.md) | None |
| [interactor_A_type](interactor_A_type.md) | None |
| [interactor_B_genetic_perturbation](interactor_B_genetic_perturbation.md) | None |
| [interactor_B_role](interactor_B_role.md) | None |
| [interactor_B_type](interactor_B_type.md) | None |
| [interacts_with](interacts_with.md) | holds between any two entities that directly or indirectly interact with each other |
| [internal](internal.md) | Classifies the entity as private (for internal use) or not (for public use). |
| [irradiation_mutagen](irradiation_mutagen.md) | The irradiation used to generate the mutation through mutagenesis |
| [irradiation_mutagen_name](irradiation_mutagen_name.md) | The name of the irradiation used to generate the mutation through mutagenesis |
| [is_best_reverse_score](is_best_reverse_score.md) | None |
| [is_best_score](is_best_score.md) | None |
| [is_extinct](is_extinct.md) | Does the allele still exist in a population somewhere? |
| [is_extrachromosomal](is_extrachromosomal.md) | Used by WormBase to indicate whether a transgenic allele is known to be extrachromosomal. |
| [is_integrated](is_integrated.md) | Used by WormBase to indicate whether a transgenic allele is known to be integrated into the genome. |
| [iso_abbreviation](iso_abbreviation.md) |  |
| [issue_name](issue_name.md) | The number of the journal issue in which the article was published. From PubMed otherwise from Mod or manual reference creation. |
| [iupac](iupac.md) | IUPAC name of the molecule |
| [keywords](keywords.md) | Keywords tagging a publication.  Aggregation of PubMed and ZFIN, editable at Alliance. |
| [label](label.md) | A short display name for the figure. For example: Figure 2, Figure 3B |
| [laboratory_of_origin](laboratory_of_origin.md) | The laboratory of origin for the entity. |
| [laboratory_of_origin_curie](laboratory_of_origin_curie.md) | The curie of the laboratory of origin for the entity. |
| [language](language.md) | Language of the reference.  Aggregation of PubMed and FlyBase, editable at Alliance. |
| [last](last.md) | last identifier in a range |
| [last_name](last_name.md) | last (family) name of a person |
| [light_chain_isotype](light_chain_isotype.md) | The isotype of the antibody light chain: e.g., i4. |
| [linkml_version](linkml_version.md) | Version of LinkML schema used for submitted file in the format n.n.n (e.g. 1.2.4 or 2.0.0) |
| [load_exceptions](load_exceptions.md) | A list of execeptions the load encountered |
| [load_files](load_files.md) | None |
| [load_finished](load_finished.md) | The date and time that the load finished. |
| [load_started](load_started.md) | The date and time that the load started. |
| [loads](loads.md) | None |
| [local_file_path](local_file_path.md) | Local file path to where the file has been saved for processing |
| [manufactured_by](manufactured_by.md) | Holds between a material entity and an Agent that has manufactured it: e.g., Molecular Probes. |
| [md5_sum](md5_sum.md) | Generated md5Sum of the file that has been uploaded |
| [medline_abbreviation](medline_abbreviation.md) |  |
| [member_terms](member_terms.md) | Set of VocabularyTerm objects in a Vocabulary object set |
| [merged_into](merged_into.md) | This gene has been merged into |
| [merged_into_id](merged_into_id.md) | ID that used to refer to this reference |
| [mesh_detail_id](mesh_detail_id.md) | The primary key for a MeshDetail object. |
| [middle_name](middle_name.md) | middle names of a person |
| [mod_corpus_association_id](mod_corpus_association_id.md) | An integer referring to a ModCorpusAssociation object. A primary key in the mod_corpus_association table, a foreign key if used in other tables. |
| [mod_corpus_sort_source](mod_corpus_sort_source.md) | origin of corpus determination for that publication and mod |
| [mod_entity_id](mod_entity_id.md) | The model organism database (MOD) identifier/curie for the object |
| [moderate_filter](moderate_filter.md) | None |
| [molecular_mutation_names](molecular_mutation_names.md) | Name of the VocabularyTerm describing the molecular mutation |
| [molecular_mutations](molecular_mutations.md) | Description of the DNA changes if precise genomic location unknown |
| [mutagenesis_method_names](mutagenesis_method_names.md) | Name of the VocabularyTerm describing the mutagenesis method, e.g. spontaneous / naturally occurring / radiation-induced / recombinant / ENU / CRISPR / TALEN / gamma rays / not specified / spontaneous / DNA / DNA AND CRISPR / DNA and TALEN / zinc finger nuclease / EMS |
| [mutagenesis_methods](mutagenesis_methods.md) | Technique used to create the allele, e.g. spontaneous / naturally occurring / radiation-induced / recombinant / ENU / CRISPR / TALEN / gamma rays / not specified / spontaneous / DNA / DNA AND CRISPR / DNA and TALEN / zinc finger nuclease / EMS |
| [mutagenesis_target](mutagenesis_target.md) | The target of the mutation, e.g. strain / adult females / adult males / embryos / sperm / not specified |
| [mutation_target_strain](mutation_target_strain.md) | The particular strain (solely for and from MGI) that is targeted by the generation method for a particular allele. |
| [mutation_target_strain_curie](mutation_target_strain_curie.md) | Curie of the particular strain that is targeted by the generation method for a particular allele (MGI only) |
| [mutation_type_curies](mutation_type_curies.md) | Curies of SOTerms representing mutation type |
| [mutation_types](mutation_types.md) | SO term for type of mutation |
| [name](name.md) | a human-readable name for an entity |
| [name_type](name_type.md) | The type of name: e.g., symbol, full_name, systematic_name, etc. |
| [name_type_name](name_type_name.md) | Name of the VocabularyTerm representing the name type of the synonym - proposed values are nomenclature_symbol / full_name / systematic_name / ncbi_protein_name / uniform / non_uniform / retired_name / unspecified |
| [namespace](namespace.md) | the namespace of the ontology. |
| [negated](negated.md) | if set to true, then the association is negated i.e. is not true |
| [nomenclature_event](nomenclature_event.md) | any of the kinds of changes made to an object's name or symbol |
| [nomenclature_event_name](nomenclature_event_name.md) | Name of the VocabularyTerm describing the nomenclature event |
| [note_dto](note_dto.md) | None |
| [note_dtos](note_dtos.md) | None |
| [note_type](note_type.md) | The type of note: e.g., cytology, comment, summary. Permissible values for 'note_type' currently = disease_summary, disease_note |
| [note_type_name](note_type_name.md) | Name of VocabularyTerm representing note type selected from the appropriate Vocabulary |
| [object](object.md) | connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object. |
| [object_allele_curie](object_allele_curie.md) | The curie of the allele that is acting as the object of an AlleleAlleleAssociation |
| [object_set](object_set.md) | Applies to a property that links a ingest object to a set of objects. This is necessary in a json document to provide context for a list, and to allow for a single json object that combines multiple object types |
| [obsolete](obsolete.md) | Entity is no longer current. |
| [okta_email](okta_email.md) | The email address of the LoggedInPerson registered with Okta for user authentication |
| [okta_id](okta_id.md) | The Okta identifier for the person registered in Okta for user authentication |
| [old_emails](old_emails.md) | list of old (outdated) emails for a person |
| [old_members](old_members.md) | Gene which were formerly members of this GeneClass |
| [online_issn](online_issn.md) |  |
| [ontology_closure_ingest_set](ontology_closure_ingest_set.md) |  |
| [ontology_type](ontology_type.md) | None |
| [open_access](open_access.md) | Indicates if the reference is freely available for use by anyone, usually with fewer copyright and licensing barriers. |
| [orcid](orcid.md) | Open Researcher and Contributor ID |
| [original_reference](original_reference.md) | Holds between an entity and the first reference to describe that entity. |
| [orthologous_to](orthologous_to.md) | a homology relationship between entities (typically genes) that diverged after a speciation event. |
| [page_area](page_area.md) | None |
| [page_description](page_description.md) | Description of page |
| [page_range](page_range.md) | Page numbers of source referenced for statement or publication. From PubMed otherwise from Mod or manual reference creation. |
| [paralogous_genes](paralogous_genes.md) | None |
| [parental_populations](parental_populations.md) | None |
| [pdf_file_path](pdf_file_path.md) | File path of the PDF file generated from the BIRT report |
| [phenotype_or_trait](phenotype_or_trait.md) | None |
| [phenotype_statement](phenotype_statement.md) | For MODs that curate to post-composed phenotype expressions composed of ontology terms and other necessary information (e.g. vocabularies, free text). This field captures the free text phenotype statement. |
| [phenotype_term](phenotype_term.md) | The phenotype ontology term used to describe the phenotype of an organism or a set of organisms. |
| [phenotype_term_curie](phenotype_term_curie.md) | The string representation of the phenotype ontology term (PhenotypeTerm) curie |
| [physically_interacts_with](physically_interacts_with.md) | holds between two entities that make physical contact as part of some interaction |
| [plain_language_abstract](plain_language_abstract.md) | Lay person, readable version of the abstract. Only relevant for PubMed references. |
| [polypeptide](polypeptide.md) | Transcript associated with variant and for which a specific location and consequence of that variant is provided, as specified by curator. Multivalued=false for this slot because although a variant can have multiple VariantTranscriptLocation stanzas, each stanza will have one and only one curated transcript ID. |
| [polyphen_prediction](polyphen_prediction.md) | PolyPhen-2 prediction |
| [polyphen_score](polyphen_score.md) | PolyPhen-2 score between 0 and 1 |
| [predicate](predicate.md) | A high-level grouping for the relationship type. This is analogous to category for nodes. In RDF, this corresponds to rdf:predicate and in Neo4j this corresponds to the relationship type. |
| [predicate_name](predicate_name.md) | Name of VocabularyTerm representing predicate of an Association |
| [prefix](prefix.md) | Denormalization to help with classifying types of crossReferences, distinguishing DOIs from PMC ids, etc. |
| [prefix_order](prefix_order.md) | The relative order of the resource when listed with other crossReferences. |
| [prefix_page](prefix_page.md) | The cateogry of pages the resource in the context of the URL associated with the crossReference provides.  Equivalent to the 'page' attribute in the Alliance resourceDescriptor file. |
| [primary_image](primary_image.md) | The primary image for an allele that is used to represent the allele on a page. |
| [print_issn](print_issn.md) |  |
| [product_binds_matrix](product_binds_matrix.md) | WB specific. ID of position matrix object |
| [protein_curie](protein_curie.md) | None |
| [protein_end](protein_end.md) | end position of variation in protein amino acid coordinates |
| [protein_start](protein_start.md) | start position of variation in protein amino acid coordinates |
| [proteins](proteins.md) | None |
| [publication](publication.md) | None |
| [publisher](publisher.md) | Publisher associated with a reference or resource. From PubMed otherwise from Mod or manual reference creation. |
| [pubmed_abstract_languages](pubmed_abstract_languages.md) | Languages for the abstract. Only relevant for PubMed references. |
| [pubmed_publication_status](pubmed_publication_status.md) | Status of the publication at PubMed. Only relevant for PubMed references. |
| [pubmed_type](pubmed_type.md) | Type of Reference as described by PubMed. Only relevant for PubMed references. |
| [qualifier_term](qualifier_term.md) | The MeSH term subheading from PubMed, to narrow down the topic. e.g. epidemiology in a Measles/epidemiology term. |
| [qualifiers](qualifiers.md) | This is the MeSH qualifier term that is optionally added to the descriptor term. |
| [reagents_used](reagents_used.md) | Reagents used in the expression assay: e.g., antibodies, probes. |
| [record_count](record_count.md) | The number of records found in the file |
| [reference_curie](reference_curie.md) | External reference curie used for ingest |
| [reference_curies](reference_curies.md) | External reference curies used for ingest |
| [reference_id](reference_id.md) | An integer referring to a Reference object in the references table. It's a primary key in the references table, a foreign key when in other tables. |
| [reference_sequence](reference_sequence.md) | Reference sequence of genome or genomic entity at position of aligned variant. |
| [referenced_curie](referenced_curie.md) | Curie of the thing being referenced in a CrossReference |
| [references](references.md) | holds between an object and a list of references |
| [related_note](related_note.md) | Holds between an object and a Note object. |
| [related_notes](related_notes.md) | Holds between an object and a list of related Note objects. |
| [related_to](related_to.md) | A relationship that is asserted between two named things. |
| [relationship_type](relationship_type.md) | None |
| [release](release.md) | MOD release ID |
| [resource_descriptor_page](resource_descriptor_page.md) | None |
| [resource_id](resource_id.md) | None |
| [resource_pages](resource_pages.md) | Pages for a particular resource |
| [s3_path](s3_path.md) | The relative path to the file in the S3 bucket |
| [s3_url](s3_url.md) | The full URL to the file from S3 |
| [schedule_active](schedule_active.md) | This determines if the the schedule is active or not |
| [scheduling_error_message](scheduling_error_message.md) | Any errors in syntax on the cron_schedule |
| [secondary_data_provider](secondary_data_provider.md) | Object representing organization (e.g. MOD) that provided the data directly to the Alliance, but not the original source, and a CrossReference to that organisation's site |
| [secondary_data_provider_dto](secondary_data_provider_dto.md) | Ingest object representing organization (e.g. MOD) that provided the data directly to the Alliance, but not the original source, and a CrossReference to that organisation's site |
| [secondary_id](secondary_id.md) | None |
| [secondary_identifiers](secondary_identifiers.md) | None |
| [sequence_targeting_reagent_curies](sequence_targeting_reagent_curies.md) | None |
| [sequence_targeting_reagents](sequence_targeting_reagents.md) | None |
| [sgd_strain_background](sgd_strain_background.md) | None |
| [sgd_strain_background_curie](sgd_strain_background_curie.md) | Curie of SGD strain background AGM |
| [short_name](short_name.md) | The short name of the organization. For Alliance Members, this is the short name used in the Members list on the website. e.g. MGD, FB, OMIM |
| [sift_prediction](sift_prediction.md) | SIFT prediction |
| [sift_score](sift_score.md) | SIFT score between 0 and 1 |
| [single_allele](single_allele.md) | None |
| [single_gene](single_gene.md) | None |
| [single_reference](single_reference.md) | holds between an object and a single reference |
| [smiles](smiles.md) | Molecular structure in SMILES format |
| [source_general_consequence](source_general_consequence.md) | SOTerm (child of SO:0001576 - transcript_variant) that describes the consequence of the variant, as stated in the source reference when no transcript ID is provided. Since a curator would determine variant location and consequences relative to at least one specific genome assembly, transcript and/or polypeptide, no slot for curated general consequence is provided. |
| [source_organization](source_organization.md) | None |
| [source_organization_abbreviation](source_organization_abbreviation.md) | None |
| [source_variant_locations](source_variant_locations.md) | Location of the variant within genomic entities,as described in the source references. |
| [spatial_qualifiers](spatial_qualifiers.md) | Qualifiers that describe the spatial characteristics of an event. |
| [specimen_alleles](specimen_alleles.md) | The Allele(s) of the specimen assayed. |
| [specimen_genomic_model](specimen_genomic_model.md) | The AffectedGenomicModel of the specimen assayed. |
| [split_from](split_from.md) | This gene exists because it has been split from this gene |
| [split_into](split_into.md) | This gene has been split into these genes |
| [sqtr_ingest_set](sqtr_ingest_set.md) |  |
| [start](start.md) | The start of the feature in positive 1-based integer coordinates |
| [start_position](start_position.md) | Start position of variant on genomic entity. |
| [strict_filter](strict_filter.md) | None |
| [subdomain_code](subdomain_code.md) | a three letter string, representing a subdomain (e.g 101 represents disease_annotation) |
| [subdomain_name](subdomain_name.md) | subdomain name (e.g disease_annotation) |
| [subject](subject.md) | connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object. |
| [subsets](subsets.md) | None |
| [subtype](subtype.md) | Subtype of affected genomic model - permissible values: strain / genotype / fish |
| [subtype_name](subtype_name.md) | Name of VocabularyTerm describing subtype - permissible values: strain / genotype / fish |
| [summary](summary.md) |  |
| [synonym_scope](synonym_scope.md) | the scope of the synonym - permissible values are narrow / broad / related / exact |
| [synonym_scope_name](synonym_scope_name.md) | Name of the VocabularyTerm representing the scope of the synonym - permissible values are narrow / broad / related / exact |
| [synonym_url](synonym_url.md) | URL for a synonym: e.g., NCBI URL for the NCBI synonym of an object. |
| [synonyms](synonyms.md) | Placeholder? Some objects still use this slot. Not clear how it fits in with NameSlotAnnotation (which captures evidence). |
| [table_key](table_key.md) | The primary key for a specific table entry, unique for that table. |
| [taxon](taxon.md) | The taxon from which the biological entity derives. |
| [taxon_curie](taxon_curie.md) | Curie of the NCBITaxonTerm representing the taxon from which the biological entity derives |
| [temporal_qualifiers](temporal_qualifiers.md) | Qualifiers of the stage or age in an annotation. |
| [text_synonyms](text_synonyms.md) | Free text synonym(s) of a term, used for controlled vocabulary terms; this is distinct from the 'synonyms' slot which has a range of a Synonym class object. |
| [title](title.md) | A human readable title for a reference. From PubMed otherwise from Mod or manual reference creation. |
| [total_records](total_records.md) | The total number of records |
| [trans_splice_leaders](trans_splice_leaders.md) | Trans-splicing splice leaders observed to operate on this gene, in species which have spliced leader trans-splicing |
| [transcript](transcript.md) | Transcript associated with variant and for which a specific location and consequence of that variant is provided, as specified at source.  Multivalued=false for this slot because although a variant can have multiple VariantTranscriptLocation stanzas, each stanza will have one and only one source transcript ID. |
| [transcript_curie](transcript_curie.md) | None |
| [transgene_chromosome_location](transgene_chromosome_location.md) | The chromosome to which a transgene has been mapped. Used for WormBase transgenes that have been integrated into the genome and mapped to a chromosome. |
| [transgene_chromosome_location_curie](transgene_chromosome_location_curie.md) | The curie string of the chromosome to which a transgene has been mapped. Used for WormBase transgenes that have been integrated into the genome and mapped to a chromosome. |
| [transposon_insertion](transposon_insertion.md) | Associated transposon insertion that causes the mutation |
| [transposon_origin](transposon_origin.md) | If this gene contains or is originating from a transposon |
| [type](type.md) | None |
| [uncertain](uncertain.md) | If set to true, then the related entity is uncertain. |
| [unique_id](unique_id.md) | A non-curie unique identifier for a thing. |
| [updated_by](updated_by.md) | The individual that last modified the entity. |
| [updated_by_curie](updated_by_curie.md) | Curie of the Person object representing the individual that updated the entity |
| [url_prefix](url_prefix.md) | The prefix of the url before the accession number. |
| [url_suffix](url_suffix.md) | The suffix of the url after the accession number. |
| [url_template](url_template.md) | URL template for constructing link to resource using prodived ID, eg. "https://www.omim.org/phenotypicSeries/[%s]" |
| [user_settings](user_settings.md) | JSON Blob containing details of UI settings for LoggedInPerson |
| [variant_curie](variant_curie.md) | None |
| [variant_genome_locations](variant_genome_locations.md) | Location of the variant in a genomic context.  multiple variant_genome_locations represent multiple assemblies. |
| [variant_ingest_set](variant_ingest_set.md) |  |
| [variant_locations](variant_locations.md) | Location of the variant within genomic entities. Variant_locations can include any or all of: one VariantGenomeLocation stanza, one or more VariantTranscriptLocation stanzas and/or one or more VariantPolypeptideLocation stanzas. |
| [variant_polypeptide_locations](variant_polypeptide_locations.md) | Location of the variant within polypeptide entities. |
| [variant_sequence](variant_sequence.md) | Sequence that differs from the reference sequence of genome or genomic entity at position of variant, as specified by curator. |
| [variant_status](variant_status.md) |  |
| [variant_transcript_locations](variant_transcript_locations.md) | Location of the variant within transcript entities. |
| [variant_type](variant_type.md) | SOTerm describing the type of variant. In practice, variant type will be limited to a subset of the SO specified in an Alliance controlled vocabulary in order to maintain consistency. |
| [vep_consequence](vep_consequence.md) | VEP consequence |
| [vep_impact](vep_impact.md) | VEP predicted impact of variation on molecule |
| [video_still](video_still.md) | An image represents a video still. |
| [vocabulary_description](vocabulary_description.md) | The free text description of a Vocabulary including its intended use. |
| [vocabulary_term_set_description](vocabulary_term_set_description.md) | The free text description of a VocabularyTermSet including its intended use. |
| [vocabulary_term_set_vocabulary](vocabulary_term_set_vocabulary.md) | The Vocabulary from which the terms contained in theVocabularyTermSet belong |
| [volume](volume.md) | Volume associated with a reference. From PubMed otherwise from Mod or manual reference creation. |
| [wbprocess](wbprocess.md) | WB specific. Eg WBbiop00000015 Corpse engulfment |
| [when_expressed](when_expressed.md) | When a gene product is observed to be present. |
| [where_expressed](where_expressed.md) | Where a gene product is observed to be present. |
| [width](width.md) | The width of the image (pixels). |
| [with_gene_curies](with_gene_curies.md) | http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#with-or-from-column-8 |
| [with_or_from](with_or_from.md) | http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/#with-or-from-column-8 |
| [xls_file_path](xls_file_path.md) | File path of the Excel file generated from the BIRT report |
| [zygosity](zygosity.md) | GENO onotology ID for allele zygosity |
| [zygosity_curie](zygosity_curie.md) | Curie of GENO ontology ID for allele zygosity - permissible_values: GENO:0000602 / GENO:0000603 / GENO:0000604 / GENO:0000605 / GENO:0000606 / GENO:0000135 / GENO:0000136 / GENO:0000137 / GENO:0000134 |


## Enumerations

| Enumeration | Description |
| --- | --- |
| [AggregationDatabaseEnum](AggregationDatabaseEnum.md) | None |
| [AntibodyClonalitySet](AntibodyClonalitySet.md) | None |
| [AntibodyNoteTypeSet](AntibodyNoteTypeSet.md) | None |
| [BackendBulkDataTypeEnum](BackendBulkDataTypeEnum.md) | None |
| [BackendBulkLoadTypeEnum](BackendBulkLoadTypeEnum.md) | None |
| [BulkLoadStatusEnum](BulkLoadStatusEnum.md) | None |
| [ConstructComponentRelationEnum](ConstructComponentRelationEnum.md) | None |
| [DetectionMethodsEnum](DetectionMethodsEnum.md) | None |
| [ExpressionConditionRelationEnum](ExpressionConditionRelationEnum.md) | None |
| [ExpressionQualifierSet](ExpressionQualifierSet.md) | None |
| [ExpressionStatementTypeEnum](ExpressionStatementTypeEnum.md) | None |
| [HeavyChainIsotypeSet](HeavyChainIsotypeSet.md) | None |
| [InteractionSourceEnum](InteractionSourceEnum.md) | None |
| [InteractionTypeEnum](InteractionTypeEnum.md) | None |
| [InteractorARoleEnum](InteractorARoleEnum.md) | None |
| [InteractorBRoleEnum](InteractorBRoleEnum.md) | None |
| [InteractorTypeEnum](InteractorTypeEnum.md) | None |
| [LightChainIsotypeSet](LightChainIsotypeSet.md) | None |
| [ModCorpusSortSourceEnum](ModCorpusSortSourceEnum.md) | None |
| [NomenclatureEventEnum](NomenclatureEventEnum.md) | None |
| [OntologyBulkLoadTypeEnum](OntologyBulkLoadTypeEnum.md) | None |
| [PolyphenPredictionLevels](PolyphenPredictionLevels.md) | None |
| [PubmedPublicationStatusEnum](PubmedPublicationStatusEnum.md) | None |
| [PubmedTypeEnum](PubmedTypeEnum.md) | None |
| [ReferenceCategoryEnum](ReferenceCategoryEnum.md) | None |
| [SiftPredictionLevels](SiftPredictionLevels.md) | None |
| [SpatialQualifierSet](SpatialQualifierSet.md) | None |
| [SqtrRelationEnum](SqtrRelationEnum.md) | None |
| [SubtypeValues](SubtypeValues.md) | None |
| [TemporalQualifierSet](TemporalQualifierSet.md) | None |
| [VariantStatusEnum](VariantStatusEnum.md) | None |
| [VepConsequenceLevels](VepConsequenceLevels.md) | None |
| [ZygosityValues](ZygosityValues.md) | None |


## Subsets

| Subset | Description |
| --- | --- |
| [AllianceSubset](AllianceSubset.md) | Subset consisting of just the alliance activities |
