
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://github.com/alliance-genome/agr_curation_schema/GeneExpressionAnnotation/" rel="canonical"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.1, mkdocs-material-9.6.14" name="generator"/>
<title>GeneExpressionAnnotation - Alliance Persistent Store</title>
<link href="../assets/stylesheets/main.342714a4.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#geneexpressionannotation">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Alliance Persistent Store" class="md-header__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Alliance Persistent Store
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              GeneExpressionAnnotation
            
          </span>
</div>
</div>
</div>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Alliance Persistent Store" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Alliance Persistent Store
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
<span class="md-ellipsis">
    Browse the Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../developing-the-model/">
<span class="md-ellipsis">
    Developing the Alliance Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="https://linkml.io/linkml/intro/tutorial.html">
<span class="md-ellipsis">
    LinkML Tutorial
    
  </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#parent-classes">
<span class="md-ellipsis">
      Parent Classes
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
<span class="md-ellipsis">
      Slots
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
<span class="md-ellipsis">
      Mappings
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-specification">
<span class="md-ellipsis">
      LinkML Specification
    </span>
</a>
<nav aria-label="LinkML Specification" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
<span class="md-ellipsis">
      Direct
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
<span class="md-ellipsis">
      Induced
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="geneexpressionannotation">GeneExpressionAnnotation</h1>
<p>An annotation describing an expression pattern for a gene.</p>
<div class="mermaid"> classDiagram
      ExpressionAnnotation &lt;|-- GeneExpressionAnnotation

      GeneExpressionAnnotation : condition_relations
      GeneExpressionAnnotation : created_by
      GeneExpressionAnnotation : cross_references
      GeneExpressionAnnotation : curie
      GeneExpressionAnnotation : data_provider
      GeneExpressionAnnotation : data_provider_cross_reference
      GeneExpressionAnnotation : date_created
      GeneExpressionAnnotation : date_updated
      GeneExpressionAnnotation : db_date_created
      GeneExpressionAnnotation : db_date_updated
      GeneExpressionAnnotation : evidence_item
      GeneExpressionAnnotation : expression_annotation_image_associations
      GeneExpressionAnnotation : expression_annotation_subject
      GeneExpressionAnnotation : expression_pattern
      GeneExpressionAnnotation : internal
      GeneExpressionAnnotation : mod_internal_id
      GeneExpressionAnnotation : negated
      GeneExpressionAnnotation : obsolete
      GeneExpressionAnnotation : primary_external_id
      GeneExpressionAnnotation : related_figures
      GeneExpressionAnnotation : related_notes
      GeneExpressionAnnotation : relation
      GeneExpressionAnnotation : uncertain
      GeneExpressionAnnotation : unique_id
      GeneExpressionAnnotation : updated_by
      GeneExpressionAnnotation : when_expressed_stage_name
      GeneExpressionAnnotation : where_expressed_statement


</div>
<p>URI: <a href="http://alliancegenome.org/GeneExpressionAnnotation">alliance:GeneExpressionAnnotation</a></p>
<h2 id="parent-classes">Parent Classes</h2>
<ul>
<li><a href="../AuditedObject/">AuditedObject</a><ul>
<li><a href="../Association/">Association</a><ul>
<li><a href="../SingleReferenceAssociation/">SingleReferenceAssociation</a><ul>
<li><a href="../Annotation/">Annotation</a><ul>
<li><a href="../ExpressionAnnotation/">ExpressionAnnotation</a><ul>
<li><strong>GeneExpressionAnnotation</strong></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<!-- no inheritance hierarchy -->
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../condition_relations/">condition_relations</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../created_by/">created_by</a></td>
<td>The individual that created the entity.</td>
</tr>
<tr>
<td><a href="../cross_references/">cross_references</a></td>
<td>list of cross references for an annotation (e.g. WBExpr variables, ZFIN figures, MGI assays)</td>
</tr>
<tr>
<td><a href="../curie/">curie</a></td>
<td>The Alliance-minted ID for the ExpressionAnnotation.</td>
</tr>
<tr>
<td><a href="../data_provider/">data_provider</a></td>
<td>The organization (e.g. MOD) from which the data was sourced</td>
</tr>
<tr>
<td><a href="../data_provider_cross_reference/">data_provider_cross_reference</a></td>
<td>CrossReference to the organization from which the data was sourced</td>
</tr>
<tr>
<td><a href="../date_created/">date_created</a></td>
<td>The date on which an entity was created. This can be applied to nodes or edges.</td>
</tr>
<tr>
<td><a href="../date_updated/">date_updated</a></td>
<td>Date on which an entity was last modified.</td>
</tr>
<tr>
<td><a href="../db_date_created/">db_date_created</a></td>
<td>The date on which an entity was created in the Alliance database.  This is distinct from date_created, which represents the date when the entity was originally created (i.e. at the MOD for imported data).</td>
</tr>
<tr>
<td><a href="../db_date_updated/">db_date_updated</a></td>
<td>Date on which an entity was last modified in the Alliance database.  This is distinct from date_updated, which represents the date when the entity was last modified and may predate import into the Alliance database.</td>
</tr>
<tr>
<td><a href="../evidence_item/">evidence_item</a></td>
<td>Single item of evidence that supports some assertion.</td>
</tr>
<tr>
<td><a href="../expression_annotation_image_associations/">expression_annotation_image_associations</a></td>
<td>Images for the ExpressionAnnotation.</td>
</tr>
<tr>
<td><a href="../expression_annotation_subject/">expression_annotation_subject</a></td>
<td>The gene to which the expression annotation applies.</td>
</tr>
<tr>
<td><a href="../expression_pattern/">expression_pattern</a></td>
<td>The spatiotemporal expression pattern of the expression annotation subject.</td>
</tr>
<tr>
<td><a href="../internal/">internal</a></td>
<td>Classifies the entity as private (for internal use) or not (for public use).</td>
</tr>
<tr>
<td><a href="../mod_internal_id/">mod_internal_id</a></td>
<td>The model organism database internal identifier for the ExpressionAnnotation.</td>
</tr>
<tr>
<td><a href="../negated/">negated</a></td>
<td>if set to true, then the association is negated i.e. is not true</td>
</tr>
<tr>
<td><a href="../obsolete/">obsolete</a></td>
<td>Entity is no longer current.</td>
</tr>
<tr>
<td><a href="../primary_external_id/">primary_external_id</a></td>
<td>The primary external (non-AGRKB) database identifier/curie for the ExpressionAnnotation.</td>
</tr>
<tr>
<td><a href="../related_figures/">related_figures</a></td>
<td>The figure(s) that support(s) the expression annotation.</td>
</tr>
<tr>
<td><a href="../related_notes/">related_notes</a></td>
<td>Use different note types to add extra information about specific aspects of the ExpressionAnnotation: temporal, structure, substructure, cellular_component, annotation-level, etc. Appropriate VocabularyTerms will need to be created: expression_annotation_note, etc.</td>
</tr>
<tr>
<td><a href="../relation/">relation</a></td>
<td>For a gene, the relation should be constrained to the expresssed_in VocabularyTerm.</td>
</tr>
<tr>
<td><a href="../uncertain/">uncertain</a></td>
<td>If set to true, then the related entity is uncertain.</td>
</tr>
<tr>
<td><a href="../unique_id/">unique_id</a></td>
<td>Unique identifier for the ExpressionAnnotation. Will be generated at AGR if not submitted by the model organism database.</td>
</tr>
<tr>
<td><a href="../updated_by/">updated_by</a></td>
<td>The individual that last modified the entity.</td>
</tr>
<tr>
<td><a href="../when_expressed_stage_name/">when_expressed_stage_name</a></td>
<td>A human-readable stage name. To be used when an ontology term for the relevant stage is not available.</td>
</tr>
<tr>
<td><a href="../where_expressed_statement/">where_expressed_statement</a></td>
<td>A human-readable free-text description of the anatomical location of the expression pattern.</td>
</tr>
</tbody>
</table>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>['alliance:GeneExpressionAnnotation']</td>
</tr>
<tr>
<td>native</td>
<td>['alliance:GeneExpressionAnnotation']</td>
</tr>
</tbody>
</table>
<h2 id="linkml-specification">LinkML Specification</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: GeneExpressionAnnotation
description: An annotation describing an expression pattern for a gene.
notes:
- "A mapping of the agr_schemas (FMS) model to this LinkML model: geneId (REQUIRED):\
  \                                                  ExpressionAnnotation.expression_annotation_subject\
  \ assay (REQUIRED):                                                   ExpressionExperiment.expression_assay_used\
  \ evidence (REQUIRED):                                                ExpressionAnnotation.single_reference\
  \ (SingleReferenceAssociation -&gt; Annotation -&gt; ExpressionAnnotation) dateAssigned\
  \ (REQUIRED):                                            ExpressionAnnotation.date_created\
  \ (AuditedObject -&gt; Association -&gt; SingleReferenceAssociation -&gt; Annotation -&gt; ExpressionAnnotation)\
  \ whenExpressed.stageName (REQUIRED):                                 ExpressionAnnotation.when_expressed_stage_name\
  \ whenExpressed.stageTermId:                                          ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.TemporalContext.developmental_stage_start) whenExpressed.stageUberonSlimTerm:\
  \                                  ExpressionAnnotation.expression_pattern (ExpressionPattern.TemporalContext.stage_uberon_slim_terms)\
  \ whereExpressed.anatomicalStructureTermId: (*REQUIRED)               ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_structure) whereExpressed.anatomicalStructureQualifierTermId:\
  \                  ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_structure_qualifiers)\
  \ whereExpressed.anatomicalStructureUberonSlimTermIds: (*REQUIRED)    ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_structure_uberon_terms) whereExpressed.anatomicalSubStructureTermId:\
  \                        ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_substructure)\
  \ whereExpressed.anatomicalSubStructureQualifierTermId:               ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_substructure_qualifiers) whereExpressed.anatomicalSubStructureUberonSlimTermIds:\
  \             ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_substructure_uberon_terms)\
  \ whereExpressed.cellularComponentTermId: (*REQUIRED)                 ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.cellular_component) whereExpressed.cellularComponentQualifierTermId:\
  \                    ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.cellular_component_qualifiers)\
  \ whereExpressed.whereExpressedStatement (REQUIRED):                  ExpressionAnnotation.where_expressed_statement\
  \ cross_reference:                                                    Consolidation\
  \ of similar annotations creates list ExpressionAnnotation.cross_references\n  \
  \                                                                  Each FMS annotation\
  \ to be consolidated contributes one cross_reference to the list\n* NOTE: For the\
  \ FMS whereExpressed component, either of the following is required. 1) anatomicalStructureTermId\
  \ AND anatomicalStructureUberonSlimTermIds, or ... 2) cellularComponentTermId. In\
  \ this LinkML model, the where_expressed slot is required, with further postconditions\
  \ requirements in the AnatomicalSite object, though these postconditions are still\
  \ not as specific as the FMS model. New required slots: relation. New non-required\
  \ slots: curie, unique_id, primary_external_id, mod_internal_id, related_notes,\
  \ expression_qualifiers, negated, uncertain, related_figures, expression_annotation_image_associations,\
  \ condition_relations."
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: ExpressionAnnotation
slots:
- cross_references
slot_usage:
  expression_annotation_subject:
    name: expression_annotation_subject
    description: The gene to which the expression annotation applies.
    range: Gene
  relation:
    name: relation
    description: For a gene, the relation should be constrained to the expresssed_in
      VocabularyTerm.
  cross_references:
    name: cross_references
    description: list of cross references for an annotation (e.g. WBExpr variables,
      ZFIN figures, MGI assays)
    range: Reference
    multivalued: true

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: GeneExpressionAnnotation
description: An annotation describing an expression pattern for a gene.
notes:
- "A mapping of the agr_schemas (FMS) model to this LinkML model: geneId (REQUIRED):\
  \                                                  ExpressionAnnotation.expression_annotation_subject\
  \ assay (REQUIRED):                                                   ExpressionExperiment.expression_assay_used\
  \ evidence (REQUIRED):                                                ExpressionAnnotation.single_reference\
  \ (SingleReferenceAssociation -&gt; Annotation -&gt; ExpressionAnnotation) dateAssigned\
  \ (REQUIRED):                                            ExpressionAnnotation.date_created\
  \ (AuditedObject -&gt; Association -&gt; SingleReferenceAssociation -&gt; Annotation -&gt; ExpressionAnnotation)\
  \ whenExpressed.stageName (REQUIRED):                                 ExpressionAnnotation.when_expressed_stage_name\
  \ whenExpressed.stageTermId:                                          ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.TemporalContext.developmental_stage_start) whenExpressed.stageUberonSlimTerm:\
  \                                  ExpressionAnnotation.expression_pattern (ExpressionPattern.TemporalContext.stage_uberon_slim_terms)\
  \ whereExpressed.anatomicalStructureTermId: (*REQUIRED)               ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_structure) whereExpressed.anatomicalStructureQualifierTermId:\
  \                  ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_structure_qualifiers)\
  \ whereExpressed.anatomicalStructureUberonSlimTermIds: (*REQUIRED)    ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_structure_uberon_terms) whereExpressed.anatomicalSubStructureTermId:\
  \                        ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_substructure)\
  \ whereExpressed.anatomicalSubStructureQualifierTermId:               ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.anatomical_substructure_qualifiers) whereExpressed.anatomicalSubStructureUberonSlimTermIds:\
  \             ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.anatomical_substructure_uberon_terms)\
  \ whereExpressed.cellularComponentTermId: (*REQUIRED)                 ExpressionAnnotation.expression_pattern\
  \ (ExpressionPattern.AnatomicalSite.cellular_component) whereExpressed.cellularComponentQualifierTermId:\
  \                    ExpressionAnnotation.expression_pattern (ExpressionPattern.AnatomicalSite.cellular_component_qualifiers)\
  \ whereExpressed.whereExpressedStatement (REQUIRED):                  ExpressionAnnotation.where_expressed_statement\
  \ cross_reference:                                                    Consolidation\
  \ of similar annotations creates list ExpressionAnnotation.cross_references\n  \
  \                                                                  Each FMS annotation\
  \ to be consolidated contributes one cross_reference to the list\n* NOTE: For the\
  \ FMS whereExpressed component, either of the following is required. 1) anatomicalStructureTermId\
  \ AND anatomicalStructureUberonSlimTermIds, or ... 2) cellularComponentTermId. In\
  \ this LinkML model, the where_expressed slot is required, with further postconditions\
  \ requirements in the AnatomicalSite object, though these postconditions are still\
  \ not as specific as the FMS model. New required slots: relation. New non-required\
  \ slots: curie, unique_id, primary_external_id, mod_internal_id, related_notes,\
  \ expression_qualifiers, negated, uncertain, related_figures, expression_annotation_image_associations,\
  \ condition_relations."
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: ExpressionAnnotation
slot_usage:
  expression_annotation_subject:
    name: expression_annotation_subject
    description: The gene to which the expression annotation applies.
    range: Gene
  relation:
    name: relation
    description: For a gene, the relation should be constrained to the expresssed_in
      VocabularyTerm.
  cross_references:
    name: cross_references
    description: list of cross references for an annotation (e.g. WBExpr variables,
      ZFIN figures, MGI assays)
    range: Reference
    multivalued: true
attributes:
  cross_references:
    name: cross_references
    description: list of cross references for an annotation (e.g. WBExpr variables,
      ZFIN figures, MGI assays)
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - xrefs
    singular_name: cross_reference
    alias: cross_references
    owner: GeneExpressionAnnotation
    domain_of:
    - GenomicEntity
    - GenomeAssembly
    - ExpressionExperiment
    - GeneExpressionAnnotation
    - AuthorReference
    - GeneInteraction
    - Antibody
    - TransgenicTool
    - OntologyTerm
    - ExternalDataBaseEntity
    range: Reference
    multivalued: true
  expression_annotation_subject:
    name: expression_annotation_subject
    description: The gene to which the expression annotation applies.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: ExpressionAnnotation
    alias: expression_annotation_subject
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: Gene
    required: true
    multivalued: false
  expression_pattern:
    name: expression_pattern
    description: The spatiotemporal expression pattern of the expression annotation
      subject.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: ExpressionAnnotation
    alias: expression_pattern
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: ExpressionPattern
    required: true
    multivalued: false
  when_expressed_stage_name:
    name: when_expressed_stage_name
    description: A human-readable stage name. To be used when an ontology term for
      the relevant stage is not available.
    notes:
    - Corresponds to whenExpressed.stageName in frozen JSON schema.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: when_expressed_stage_name
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    - BioSampleAge
    - BioSampleAgeDTO
    range: string
    required: true
    multivalued: false
  where_expressed_statement:
    name: where_expressed_statement
    description: A human-readable free-text description of the anatomical location
      of the expression pattern.
    notes:
    - Corresponds to whereExpressedStatement in frozen JSON schema.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: ExpressionAnnotation
    alias: where_expressed_statement
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: string
    required: true
    multivalued: false
  negated:
    name: negated
    description: if set to true, then the association is negated i.e. is not true
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: negated
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    - PhenotypeAnnotation
    - PhenotypeAnnotationDTO
    - DiseaseAnnotation
    - DiseaseAnnotationDTO
    range: boolean
  uncertain:
    name: uncertain
    description: If set to true, then the related entity is uncertain.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: uncertain
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: boolean
  related_figures:
    name: related_figures
    description: The figure(s) that support(s) the expression annotation.
    notes:
    - This can seem redundant with ExpressionAnnotationImagePane, but sometimes we
      know the figure but don't have the image file. Optional, as existing annotations
      from some MODs have no related images or figures.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: related_figures
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: Figure
    required: false
    multivalued: true
  expression_annotation_image_associations:
    name: expression_annotation_image_associations
    description: Images for the ExpressionAnnotation.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: ExpressionAnnotation
    alias: expression_annotation_image_associations
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionAnnotation
    range: ExpressionAnnotationImagePaneAssociation
    required: false
    multivalued: true
  curie:
    name: curie
    description: The Alliance-minted ID for the ExpressionAnnotation.
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: curie
    owner: GeneExpressionAnnotation
    domain_of:
    - Laboratory
    - CurieObject
    - Identifier
    - GeneNomenclatureSet
    - OntologyTerm
    - Annotation
    - ExternalDataBaseEntityDTO
    range: uriorcurie
    multivalued: false
  unique_id:
    name: unique_id
    description: Unique identifier for the ExpressionAnnotation. Will be generated
      at AGR if not submitted by the model organism database.
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: unique_id
    owner: GeneExpressionAnnotation
    domain_of:
    - Person
    - Exon
    - CodingSequence
    - ExpressionExperiment
    - GeneInteraction
    - Reagent
    - Annotation
    - ExperimentalCondition
    - ConditionRelation
    range: string
    multivalued: false
  primary_external_id:
    name: primary_external_id
    description: The primary external (non-AGRKB) database identifier/curie for the
      ExpressionAnnotation.
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: primary_external_id
    owner: GeneExpressionAnnotation
    domain_of:
    - Person
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
  mod_internal_id:
    name: mod_internal_id
    description: The model organism database internal identifier for the ExpressionAnnotation.
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: mod_internal_id
    owner: GeneExpressionAnnotation
    domain_of:
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
  condition_relations:
    name: condition_relations
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: condition_relations
    owner: GeneExpressionAnnotation
    domain_of:
    - ExpressionExperiment
    - Annotation
    range: ConditionRelation
    multivalued: true
  related_notes:
    name: related_notes
    description: 'Use different note types to add extra information about specific
      aspects of the ExpressionAnnotation: temporal, structure, substructure, cellular_component,
      annotation-level, etc. Appropriate VocabularyTerms will need to be created:
      expression_annotation_note, etc.'
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: related_notes
    owner: GeneExpressionAnnotation
    domain_of:
    - SubmittedObject
    - Gene
    - GeneChangeEventSlotAnnotation
    - ExpressionExperiment
    - ConstructComponentSlotAnnotation
    - ConstructGenomicEntityAssociation
    - ConstructCassetteAssociation
    - CassetteComponentSlotAnnotation
    - CassetteGenomicEntityAssociation
    - CassetteTransgenicToolAssociation
    - CassetteStrAssociation
    - Annotation
    - HTPExpressionDatasetSampleAnnotation
    range: Note
    multivalued: true
  data_provider:
    name: data_provider
    description: The organization (e.g. MOD) from which the data was sourced
    notes:
    - Slot is inherited from Annotation class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: data_provider
    owner: GeneExpressionAnnotation
    domain_of:
    - SubmittedObject
    - Chromosome
    - Species
    - Annotation
    - HTPExpressionDatasetAnnotation
    range: Organization
    required: true
    multivalued: false
  data_provider_cross_reference:
    name: data_provider_cross_reference
    description: CrossReference to the organization from which the data was sourced
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: data_provider_cross_reference
    owner: GeneExpressionAnnotation
    domain_of:
    - SubmittedObject
    - Chromosome
    - Species
    - Annotation
    - HTPExpressionDatasetAnnotation
    range: CrossReference
    required: false
    multivalued: false
  evidence_item:
    name: evidence_item
    description: Single item of evidence that supports some assertion.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: evidence_item
    owner: GeneExpressionAnnotation
    domain_of:
    - SingleReferenceAssociation
    range: InformationContentEntity
  relation:
    name: relation
    description: For a gene, the relation should be constrained to the expresssed_in
      VocabularyTerm.
    notes:
    - Slot is inherited from Association class.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    exact_mappings:
    - biolink:predicate
    is_a: association_slot
    alias: relation
    owner: GeneExpressionAnnotation
    domain_of:
    - Association
    - ConstructComponentSlotAnnotation
    - CassetteComponentSlotAnnotation
    range: VocabularyTerm
    required: true
  created_by:
    name: created_by
    description: The individual that created the entity.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObject
    alias: created_by
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    range: Person
    multivalued: false
  date_created:
    name: date_created
    description: The date on which an entity was created. This can be applied to nodes
      or edges.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - creation_date
    exact_mappings:
    - dct:createdOn
    - WIKIDATA_PROPERTY:P577
    alias: date_created
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  updated_by:
    name: updated_by
    description: The individual that last modified the entity.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObject
    alias: updated_by
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    range: Person
    multivalued: false
  date_updated:
    name: date_updated
    description: Date on which an entity was last modified.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - date_last_modified
    alias: date_updated
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_created:
    name: db_date_created
    description: The date on which an entity was created in the Alliance database.  This
      is distinct from date_created, which represents the date when the entity was
      originally created (i.e. at the MOD for imported data).
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_created
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_updated:
    name: db_date_updated
    description: Date on which an entity was last modified in the Alliance database.  This
      is distinct from date_updated, which represents the date when the entity was
      last modified and may predate import into the Alliance database.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_updated
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  internal:
    name: internal
    description: Classifies the entity as private (for internal use) or not (for public
      use).
    notes:
    - Default value is true.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: internal
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean
    required: true
  obsolete:
    name: obsolete
    description: Entity is no longer current.
    notes:
    - Obsolete entities are preserved in the database for posterity but should not
      be publicly displayed.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: obsolete
    owner: GeneExpressionAnnotation
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean

</code></pre>
</details>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
<script src="../assets/javascripts/bundle.13a4f30d.min.js"></script>
<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>