<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="None">
  
  
  <link rel="shortcut icon" href="img/favicon.ico">
  <title>Model to describe the Alliance curation/persistence data store</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="css/theme.css" />
  <link rel="stylesheet" href="css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Index";
    var mkdocs_page_input_path = "index.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="js/jquery-2.1.1.min.js" defer></script>
  <script src="js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href="." class="icon icon-home"> Model to describe the Alliance curation/persistence data store</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="./search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="home/">Home</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href=".">Index</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#classes">Classes</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#mixins">Mixins</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#slots">Slots</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#enums">Enums</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#types">Types</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#built-in">Built in</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#defined">Defined</a>
    </li>
        </ul>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="credits/">Credits</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href=".">Model to describe the Alliance curation/persistence data store</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".">Docs</a> &raquo;</li>
    
      
    
    <li>Index</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="alliance-schema-prototype-schema">Alliance-Schema-Prototype schema</h1>
<p>Alliance Schema Prototype</p>
<h3 id="classes">Classes</h3>
<ul>
<li><a href="AffectedGenomicModelComponent/">AffectedGenomicModelComponent</a> - Allele that affects the model and its zygosity</li>
<li><a href="Allele.md">Allele</a> - One of multiple possible forms of a functional genomic element (most often described as a locus or gene), differing from the reference DNA sequence.  The genomic element can be endogenous or constructed.</li>
<li><a href="Assembly.md">Assembly</a></li>
<li><a href="AuthorReference/">AuthorReference</a></li>
<li><a href="CellLine/">CellLine</a> - Dummy cell line class</li>
<li><a href="Chromosome/">Chromosome</a> - The ID of the landmark used to establish the coordinate system for the current feature.</li>
<li><a href="Construct/">Construct</a> - Dummy construct class</li>
<li><a href="Disease/">Disease</a></li>
<li><a href="DiseaseAssociation/">DiseaseAssociation</a> - Dummy disease association class</li>
<li><a href="GeneGenomicLocation/">GeneGenomicLocation</a></li>
<li><a href="GenomicEntity/">GenomicEntity</a><ul>
<li><a href="AffectedGenomicModel/">AffectedGenomicModel</a> - Includes inbred strains, stocks, disease models and mutant genotypes</li>
<li><a href="Gene/">Gene</a></li>
<li><a href="Transcript/">Transcript</a></li>
<li><a href="Variant/">Variant</a> - A DNA sequence that differs relative to a designated reference sequence.  The sequence occurs at a single position or in contiguous nucleotides.</li>
</ul>
</li>
<li><a href="Image/">Image</a> - Dummy image class</li>
<li><a href="InformationContentEntity/">InformationContentEntity</a> - a piece of information that typically describes some topic of discourse or is used as support. Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise.<ul>
<li><a href="CrossReference/">CrossReference</a></li>
<li><a href="Reference.md">Reference</a></li>
<li><a href="Resource/">Resource</a></li>
</ul>
</li>
<li><a href="NoteType/">NoteType</a> - Describes the relation between a note and an object</li>
<li><a href="Person/">Person</a></li>
<li><a href="Phenotype/">Phenotype</a></li>
<li><a href="PhenotypeAssociation/">PhenotypeAssociation</a> - Dummy phenotype association class</li>
<li><a href="ReferenceType/">ReferenceType</a> - Describes the relation between a reference and an object</li>
<li><a href="SequenceTargetingReagent/">SequenceTargetingReagent</a> - Dummy sequence targeting reagent class</li>
<li><a href="Species/">Species</a></li>
<li><a href="VariantGeneConsequence/">VariantGeneConsequence</a></li>
<li><a href="VariantGenomicLocation/">VariantGenomicLocation</a></li>
<li><a href="VariantTranscriptConsequence/">VariantTranscriptConsequence</a> - Class for transcript-level VEP results</li>
</ul>
<h3 id="mixins">Mixins</h3>
<h3 id="slots">Slots</h3>
<ul>
<li><a href="PubMed_type/">PubMed_type</a> - Type of InformationContentEntity as described by pub med.</li>
<li><a href="aberration/">aberration</a> - Associated deficiency (etc.) whose breakpoint causes the mutation</li>
<li><a href="abstract/">abstract</a></li>
<li><a href="address/">address</a> - the particulars of the place where someone or an organization is situated.  For now, this slot is a simple text "blob" containing all relevant details of the given location for fitness of purpose. For the moment, this "address" can include other contact details such as email and phone number(?).</li>
<li><a href="affected_genes/">affected_genes</a> - Gene(s) altered in allele</li>
<li><a href="allele/">allele</a></li>
<li><a href="alliance_category/">alliance_category</a></li>
<li><a href="amino_acid_reference/">amino_acid_reference</a> - reference genome amino acid sequence at variant position<ul>
<li><a href="VariantTranscriptConsequence_amino_acid_reference/">VariantTranscriptConsequence➞amino_acid_reference</a> - Amino acid sequence encoded by codon(s) in reference genome sequence altered by the variant</li>
</ul>
</li>
<li><a href="amino_acid_variant/">amino_acid_variant</a> - variant amino acid sequence at variant position<ul>
<li><a href="VariantTranscriptConsequence_amino_acid_variant/">VariantTranscriptConsequence➞amino_acid_variant</a> - Amino acid sequence encoded by codon(s) in variant sequence</li>
</ul>
</li>
<li><a href="assembly/">assembly</a></li>
<li><a href="associated_notes/">associated_notes</a> - Notes associated with the entity in some way</li>
<li><a href="associated_references/">associated_references</a> - References associated with the entity in some way</li>
<li><a href="association_slot/">association_slot</a> - any slot that relates an association to another entity<ul>
<li><a href="subject/">subject</a> - connects an association to the subject of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.<ul>
<li><a href="GeneGenomicLocation_subject/">GeneGenomicLocation➞subject</a> - subject should be the gene identifier</li>
<li><a href="VariantGeneConsequence_subject/">VariantGeneConsequence➞subject</a> - subject should be the variant identifier</li>
<li><a href="VariantGenomicLocation_subject/">VariantGenomicLocation➞subject</a> - subject should be the variant identifier</li>
<li><a href="VariantTranscriptConsequence_subject/">VariantTranscriptConsequence➞subject</a> - subject should be the variant identifier</li>
</ul>
</li>
</ul>
</li>
<li><a href="authors/">authors</a></li>
<li><a href="carried_by/">carried_by</a> - Affected genomic models that carry the allele</li>
<li><a href="cdna_end/">cdna_end</a> - end position of variation in cDNA coordinates</li>
<li><a href="cdna_start/">cdna_start</a> - start position of variation in cDNA coordinates<ul>
<li><a href="VariantTranscriptConsequence_cdna_start/">VariantTranscriptConsequence➞cdna_start</a> - Start position of variant in cDNA coordinates</li>
</ul>
</li>
<li><a href="cds_end/">cds_end</a> - end position of variation in CDS coordinates<ul>
<li><a href="VariantTranscriptConsequence_cds_end/">VariantTranscriptConsequence➞cds_end</a> - End position of variant in CDS coordinates</li>
</ul>
</li>
<li><a href="cds_start/">cds_start</a> - start position of variation in CDS coordinates<ul>
<li><a href="VariantTranscriptConsequence_cds_start/">VariantTranscriptConsequence➞cds_start</a> - Start position of variant in CDS coordinates</li>
</ul>
</li>
<li><a href="citation/">citation</a></li>
<li><a href="cnda_end/">cnda_end</a> - End position of variant in cDNA coordinates<ul>
<li><a href="VariantTranscriptConsequence_cnda_end/">VariantTranscriptConsequence➞cnda_end</a></li>
</ul>
</li>
<li><a href="codon_reference/">codon_reference</a> - reference sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cTa)<ul>
<li><a href="VariantTranscriptConsequence_codon_reference/">VariantTranscriptConsequence➞codon_reference</a> - Reference genome sequence of codon(s) altered by the variant.  Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
</ul>
</li>
<li><a href="codon_variant/">codon_variant</a> - variant sequence of codon(s) affected by variation - bases outside of the variant region are in lower case, those within are in upper case (e.g. cAa)<ul>
<li><a href="VariantTranscriptConsequence_codon_variant/">VariantTranscriptConsequence➞codon_variant</a> - Sequence of codon(s) in variant sequence altered by the variant. Bases affected by the variant are given in upper case, bases flanking the variation are given in lower case</li>
</ul>
</li>
<li><a href="components/">components</a> - Collection of genomic components that make up a model, i.e. 'allele', each with a zygosity</li>
<li><a href="computed_gene/">computed_gene</a></li>
<li><a href="contains_construct/">contains_construct</a></li>
<li><a href="copyright_date/">copyright_date</a></li>
<li><a href="corresponding_author/">corresponding_author</a></li>
<li><a href="creation_date/">creation_date</a> - date on which an entity was created. This can be applied to nodes or edges</li>
<li><a href="cross_references/">cross_references</a></li>
<li><a href="curie/">curie</a></li>
<li><a href="data_provider/">data_provider</a> - MOD that provided data</li>
<li><a href="database_status/">database_status</a> - Database status of the allele</li>
<li><a href="date_arrived_in_PubMed/">date_arrived_in_PubMed</a> - Day in which a reference or resource was created in PUBMED. Only relevant for PUBMED references.</li>
<li><a href="date_last_modified/">date_last_modified</a> - Date on which an entity was last modified by the MOD.</li>
<li><a href="date_last_modified_in_PubMed/">date_last_modified_in_PubMed</a> - Date on which an entity was created.</li>
<li><a href="date_produced/">date_produced</a> - date AGR data generated by MOD</li>
<li><a href="date_published/">date_published</a> - Date on which an entity was created.</li>
<li><a href="day_published/">day_published</a> - Day in which an entity was created.</li>
<li><a href="description/">description</a> - a human-readable description of an entity</li>
<li><a href="disease_associations/">disease_associations</a></li>
<li><a href="display_name/">display_name</a></li>
<li><a href="editors/">editors</a> - holds between a resource and a editor_resource</li>
<li><a href="embryonic_stem_cell_lines/">embryonic_stem_cell_lines</a> - Embryonic stem cell lines known to carry the allele</li>
<li><a href="end/">end</a> - The end of the feature in positive 1-based integer coordinates</li>
<li><a href="feature_type/">feature_type</a> - Allele classification</li>
<li><a href="first_name/">first_name</a></li>
<li><a href="from_resource/">from_resource</a></li>
<li><a href="from_species/">from_species</a> - holds between a genomic entity and a species</li>
<li><a href="functional_impact/">functional_impact</a> - Experimentally assessed functional impact of the allele, e.g. knockout / amorphic</li>
<li><a href="gene_synopsis/">gene_synopsis</a></li>
<li><a href="gene_synopsis_URL/">gene_synopsis_URL</a></li>
<li><a href="generation_method/">generation_method</a> - Technique used to create the allele, e.g. spontaneous / naturally occuring / radiation</li>
<li><a href="genomic_locations/">genomic_locations</a></li>
<li><a href="genomic_reference_sequence/">genomic_reference_sequence</a> - reference genome sequence at variation position</li>
<li><a href="genomic_variant_sequence/">genomic_variant_sequence</a> - variant genome sequence at variation position</li>
<li><a href="germline_transmission_status/">germline_transmission_status</a> - For alleles made in cell lines, have they been transmitted to the germline of an animal</li>
<li><a href="hgvs_coding_nomenclature/">hgvs_coding_nomenclature</a> - HGVSc nomenclature for variation in coding sequence<ul>
<li><a href="VariantTranscriptConsequence_hgvs_coding_nomenclature/">VariantTranscriptConsequence➞hgvs_coding_nomenclature</a> - HGVS coding sequence (HGVSc) name</li>
</ul>
</li>
<li><a href="hgvs_nomenclature/">hgvs_nomenclature</a> - HGVSg nomenclature for variant</li>
<li><a href="hgvs_protein_nomenclature/">hgvs_protein_nomenclature</a> - HGVSp nomenclature for variation in protein<ul>
<li><a href="VariantTranscriptConsequence_hgvs_protein_nomenclature/">VariantTranscriptConsequence➞hgvs_protein_nomenclature</a> - HGVS protein sequence (HGVSp) name</li>
</ul>
</li>
<li><a href="id/">id</a> - A unique identifier for a thing. Must be either a CURIE shorthand for a URI or a complete URI<ul>
<li><a href="Reference_id.md">Reference➞id</a></li>
<li><a href="Resource_id/">Resource➞id</a> - Precedence of identifiers for references is as follows: PMID if available; DOI if not; actual alternate CURIE otherwise.</li>
</ul>
</li>
<li><a href="images/">images</a> - Any associated image</li>
<li><a href="in_collection/">in_collection</a> - Set of high-throughput alleles made by large projects</li>
<li><a href="inheritence_mode/">inheritence_mode</a> - Mode of inheritence, e.g. dominant / semi-dominant / recessive / unknown</li>
<li><a href="initials/">initials</a></li>
<li><a href="is_extinct/">is_extinct</a> - Does the allele still exist in a population somewhere?</li>
<li><a href="is_variant_of_allele/">is_variant_of_allele</a></li>
<li><a href="is_variant_of_transcript/">is_variant_of_transcript</a></li>
<li><a href="iso_abbreviation/">iso_abbreviation</a></li>
<li><a href="issue_date/">issue_date</a> - Date on which an entity was issued.</li>
<li><a href="issue_name/">issue_name</a></li>
<li><a href="keywords/">keywords</a> - keywords tagging a publication</li>
<li><a href="last_modified/">last_modified</a> - Date on which an entity was last modified in PubMed or in the MOD.</li>
<li><a href="last_name/">last_name</a></li>
<li><a href="located_on/">located on</a></li>
<li><a href="medline_abbreviation/">medline_abbreviation</a></li>
<li><a href="middle_names/">middle_names</a></li>
<li><a href="mod_reference_types/">mod_reference_types</a> - List of types as assigned at a MOD.</li>
<li><a href="molecular_mutation/">molecular_mutation</a> - Description of the DNA changes if precise genomic location unknown</li>
<li><a href="month_published/">month_published</a> - Month in which an entity was created.</li>
<li><a href="mutant_cell_lines/">mutant_cell_lines</a> - Mutant cell lines known to carry the allele</li>
<li><a href="name/">name</a> - a human-readable name for an entity</li>
<li><a href="negated/">negated</a> - if set to true, then the association is negated i.e. is not true</li>
<li><a href="note_association/">note_association</a> - Description of how a note is associated with an entity, e.g. curators comments / private / origin</li>
<li><a href="notes/">notes</a></li>
<li><a href="object/">object</a> - connects an association to the object of the association. For example, in a gene-to-phenotype association, the gene is subject and phenotype is object.<ul>
<li><a href="GeneGenomicLocation_object/">GeneGenomicLocation➞object</a> - object should be the chromosome identifier</li>
<li><a href="VariantGeneConsequence_object/">VariantGeneConsequence➞object</a> - object should be the gene identifier</li>
<li><a href="VariantGenomicLocation_object/">VariantGenomicLocation➞object</a> - object should be the chromosome identifier</li>
<li><a href="VariantTranscriptConsequence_object/">VariantTranscriptConsequence➞object</a> - object should be the transcript identifier</li>
</ul>
</li>
<li><a href="online_issn/">online_issn</a></li>
<li><a href="orcid/">orcid</a></li>
<li><a href="origins/">origins</a> - Affected genomic models that the allele was introduced in</li>
<li><a href="padding_left/">padding_left</a> - flanking sequence upstream of the variation position on the sense strand of the reference genome</li>
<li><a href="padding_right/">padding_right</a> - flanking sequence downstream of the variation position on the sense strand of the reference genome</li>
<li><a href="page_areas/">page_areas</a></li>
<li><a href="pages/">pages</a> - page number of source referenced for statement or publication</li>
<li><a href="parent_cell_line/">parent_cell_line</a> - Parental cell line of alleles made in embryonic stem cells</li>
<li><a href="parental_populations/">parental_populations</a></li>
<li><a href="person_id/">person id</a><ul>
<li><a href="Person_person_id/">Person➞person id</a></li>
</ul>
</li>
<li><a href="phenotype_associations/">phenotype_associations</a></li>
<li><a href="polyphen_prediction/">polyphen_prediction</a> - PolyPhen-2 prediction</li>
<li><a href="polyphen_score/">polyphen_score</a> - PolyPhen-2 score between 0 and 1</li>
<li><a href="predicate/">predicate</a> - A high-level grouping for the relationship type. This is analogous to category for nodes. In RDF, this corresponds to rdf:predicate and in Neo4j this corresponds to the relationship type.</li>
<li><a href="prefix/">prefix</a> - Denormalization to help with classifying types of crossReferences, distinguishing DOIs from PMC ids, etc.</li>
<li><a href="print_issn/">print_issn</a></li>
<li><a href="protein_end/">protein_end</a> - end position of variation in protein amino acid coordinates<ul>
<li><a href="VariantTranscriptConsequence_protein_end/">VariantTranscriptConsequence➞protein_end</a> - End position of variant in amino acid sequence</li>
</ul>
</li>
<li><a href="protein_sequence/">protein_sequence</a></li>
<li><a href="protein_start/">protein_start</a> - start position of variation in protein amino acid coordinates<ul>
<li><a href="VariantTranscriptConsequence_protein_start/">VariantTranscriptConsequence➞protein_start</a> - Start position of variant in amino acid sequence</li>
</ul>
</li>
<li><a href="publisher/">publisher</a></li>
<li><a href="qualifiers/">qualifiers</a> - This is the MeSH qualifier term that is optionally added to the descriptor term.</li>
<li><a href="reference/">reference</a> - Associated reference</li>
<li><a href="reference_association/">reference_association</a> - Description of how a reference is associated with an entity, e.g. molecular / origin / other</li>
<li><a href="reference_id/">reference_id</a></li>
<li><a href="references/">references</a> - holds between an object and a reference</li>
<li><a href="release/">release</a> - MOD release ID</li>
<li><a href="secondary_identifiers/">secondary_identifiers</a></li>
<li><a href="sequence_targeting_reagents/">sequence_targeting_reagents</a></li>
<li><a href="sift_prediction/">sift_prediction</a> - SIFT prediction</li>
<li><a href="sift_score/">sift_score</a> - SIFT score between 0 and 1</li>
<li><a href="start/">start</a> - The start of the feature in positive 1-based integer coordinates</li>
<li><a href="subtype/">subtype</a> - Subtype of affected genomic model</li>
<li><a href="summary/">summary</a></li>
<li><a href="symbol/">symbol</a> - Symbol for a particular thing</li>
<li><a href="synonyms/">synonyms</a> - holds between a named thing and a synonym</li>
<li><a href="tags/">tags</a> - Optional ist of controlled vocabulary tags that give categories to each pub (ie: "can show images").</li>
<li><a href="taxon/">taxon</a></li>
<li><a href="taxon_id/">taxon_id</a> - NCBI taxid</li>
<li><a href="title/">title</a> - A human readable title for a reference.<ul>
<li><a href="Resource_title/">Resource➞title</a> - the title of the publication</li>
</ul>
</li>
<li><a href="topics/">topics</a> - Connects an object to a vocabulary term or ontology that describes some aspect of the entity.</li>
<li><a href="transposon_insertion/">transposon_insertion</a> - Associated transposon insertion that causes the mutation</li>
<li><a href="type/">type</a></li>
<li><a href="variant_of_allele/">variant_of_allele</a></li>
<li><a href="variant_of_transcript/">variant_of_transcript</a></li>
<li><a href="vep_consequence/">vep_consequence</a> - VEP consequence</li>
<li><a href="vep_impact/">vep_impact</a> - VEP predicted impact of variation on molecule</li>
<li><a href="volume/">volume</a></li>
<li><a href="volumes/">volumes</a></li>
<li><a href="year_published/">year_published</a> - Year in which an entity was created.</li>
<li><a href="zygosity/">zygosity</a> - GENO onotology ID for allele zygosity</li>
</ul>
<h3 id="enums">Enums</h3>
<ul>
<li><a href="database_statuses.md">database_statuses</a></li>
<li><a href="modes_of_inheritence.md">modes_of_inheritence</a></li>
<li><a href="note_association_types.md">note_association_types</a></li>
<li><a href="polyphen_prediction_levels.md">polyphen_prediction_levels</a></li>
<li><a href="reference_association_types.md">reference_association_types</a></li>
<li><a href="sift_prediction_levels.md">sift_prediction_levels</a></li>
<li><a href="subtype_values.md">subtype_values</a></li>
<li><a href="tag_set.md">tag_set</a></li>
<li><a href="vep_consequence_levels.md">vep_consequence_levels</a></li>
<li><a href="zygosity_values.md">zygosity_values</a></li>
</ul>
<h3 id="types">Types</h3>
<h4 id="built-in">Built in</h4>
<ul>
<li><strong>Bool</strong></li>
<li><strong>Decimal</strong></li>
<li><strong>ElementIdentifier</strong></li>
<li><strong>NCName</strong></li>
<li><strong>NodeIdentifier</strong></li>
<li><strong>URI</strong></li>
<li><strong>URIorCURIE</strong></li>
<li><strong>XSDDate</strong></li>
<li><strong>XSDDateTime</strong></li>
<li><strong>XSDTime</strong></li>
<li><strong>float</strong></li>
<li><strong>int</strong></li>
<li><strong>str</strong></li>
</ul>
<h4 id="defined">Defined</h4>
<ul>
<li><a href="BiologicalSequence/">BiologicalSequence</a>  (<a href="String/">String</a>) </li>
<li><a href="Boolean/">Boolean</a>  (<strong>Bool</strong>)  - A binary (true or false) value</li>
<li><a href="Date/">Date</a>  (<strong>XSDDate</strong>)  - a date (year, month and day) in an idealized calendar</li>
<li><a href="Datetime/">Datetime</a>  (<strong>XSDDateTime</strong>)  - The combination of a date and time</li>
<li><a href="Decimal/">Decimal</a>  (<strong>Decimal</strong>)  - A real number with arbitrary precision that conforms to the xsd:decimal specification</li>
<li><a href="Double/">Double</a>  (<strong>float</strong>)  - A real number that conforms to the xsd:double specification</li>
<li><a href="Float/">Float</a>  (<strong>float</strong>)  - A real number that conforms to the xsd:float specification</li>
<li><a href="Integer/">Integer</a>  (<strong>int</strong>)  - An integer</li>
<li><a href="Ncname/">Ncname</a>  (<strong>NCName</strong>)  - Prefix part of CURIE</li>
<li><a href="Nodeidentifier/">Nodeidentifier</a>  (<strong>NodeIdentifier</strong>)  - A URI, CURIE or BNODE that represents a node in a model.</li>
<li><a href="Objectidentifier/">Objectidentifier</a>  (<strong>ElementIdentifier</strong>)  - A URI or CURIE that represents an object in the model.</li>
<li><a href="String/">String</a>  (<strong>str</strong>)  - A character string</li>
<li><a href="Time/">Time</a>  (<strong>XSDTime</strong>)  - A time object represents a (local) time of day, independent of any particular day</li>
<li><a href="Uri/">Uri</a>  (<strong>URI</strong>)  - a complete URI</li>
<li><a href="Uriorcurie/">Uriorcurie</a>  (<strong>URIorCURIE</strong>)  - a URI or a CURIE</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="credits/" class="btn btn-neutral float-right" title="Credits">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="home/" class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="home/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="credits/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '.';</script>
    <script src="js/theme.js" defer></script>
      <script src="search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>

<!--
MkDocs version : 1.1.2
Build Date UTC : 2021-05-25 20:13:58.320462+00:00
-->
