
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="https://github.com/alliance-genome/agr_curation_schema/AGMPhenotypeAnnotationDTO/" rel="canonical"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.1, mkdocs-material-9.6.14" name="generator"/>
<title>AGMPhenotypeAnnotationDTO - Alliance Persistent Store</title>
<link href="../assets/stylesheets/main.342714a4.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#agmphenotypeannotationdto">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Alliance Persistent Store" class="md-header__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Alliance Persistent Store
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              AGMPhenotypeAnnotationDTO
            
          </span>
</div>
</div>
</div>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Alliance Persistent Store" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Alliance Persistent Store">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Alliance Persistent Store
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/alliance-genome/agr_curation_schema" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
<span class="md-ellipsis">
    Browse the Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../developing-the-model/">
<span class="md-ellipsis">
    Developing the Alliance Model
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="https://linkml.io/linkml/intro/tutorial.html">
<span class="md-ellipsis">
    LinkML Tutorial
    
  </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#parent-classes">
<span class="md-ellipsis">
      Parent Classes
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
<span class="md-ellipsis">
      Slots
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
<span class="md-ellipsis">
      Mappings
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-specification">
<span class="md-ellipsis">
      LinkML Specification
    </span>
</a>
<nav aria-label="LinkML Specification" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
<span class="md-ellipsis">
      Direct
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
<span class="md-ellipsis">
      Induced
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="agmphenotypeannotationdto">AGMPhenotypeAnnotationDTO</h1>
<p>Ingest class for an association between an AGM and a phenotype</p>
<div class="mermaid"> classDiagram
      PhenotypeAnnotationDTO &lt;|-- AGMPhenotypeAnnotationDTO

      AGMPhenotypeAnnotationDTO : agm_identifier
      AGMPhenotypeAnnotationDTO : asserted_allele_identifiers
      AGMPhenotypeAnnotationDTO : asserted_gene_identifiers
      AGMPhenotypeAnnotationDTO : condition_relation_dtos
      AGMPhenotypeAnnotationDTO : created_by_curie
      AGMPhenotypeAnnotationDTO : cross_reference_dto
      AGMPhenotypeAnnotationDTO : data_provider_dto
      AGMPhenotypeAnnotationDTO : date_created
      AGMPhenotypeAnnotationDTO : date_updated
      AGMPhenotypeAnnotationDTO : db_date_created
      AGMPhenotypeAnnotationDTO : db_date_updated
      AGMPhenotypeAnnotationDTO : evidence_curie
      AGMPhenotypeAnnotationDTO : inferred_allele_identifier
      AGMPhenotypeAnnotationDTO : inferred_gene_identifier
      AGMPhenotypeAnnotationDTO : internal
      AGMPhenotypeAnnotationDTO : mod_internal_id
      AGMPhenotypeAnnotationDTO : negated
      AGMPhenotypeAnnotationDTO : note_dtos
      AGMPhenotypeAnnotationDTO : obsolete
      AGMPhenotypeAnnotationDTO : phenotype_statement
      AGMPhenotypeAnnotationDTO : phenotype_term_curies
      AGMPhenotypeAnnotationDTO : primary_external_id
      AGMPhenotypeAnnotationDTO : updated_by_curie


</div>
<p>URI: <a href="http://alliancegenome.org/AGMPhenotypeAnnotationDTO">alliance:AGMPhenotypeAnnotationDTO</a></p>
<h2 id="parent-classes">Parent Classes</h2>
<ul>
<li><a href="../AuditedObjectDTO/">AuditedObjectDTO</a><ul>
<li><a href="../SingleReferenceAssociationDTO/">SingleReferenceAssociationDTO</a><ul>
<li><a href="../AnnotationDTO/">AnnotationDTO</a><ul>
<li><a href="../PhenotypeAnnotationDTO/">PhenotypeAnnotationDTO</a><ul>
<li><strong>AGMPhenotypeAnnotationDTO</strong></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<!-- no inheritance hierarchy -->
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../agm_identifier/">agm_identifier</a></td>
<td>Curie or internal ID of AGM</td>
</tr>
<tr>
<td><a href="../asserted_allele_identifiers/">asserted_allele_identifiers</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of the allele to which something is manually asserted to be associated</td>
</tr>
<tr>
<td><a href="../asserted_gene_identifiers/">asserted_gene_identifiers</a></td>
<td>Identifiers (curie/primary_external_id/mod_internal_id) of the gene(s) to which something is manually asserted to be associated</td>
</tr>
<tr>
<td><a href="../condition_relation_dtos/">condition_relation_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../created_by_curie/">created_by_curie</a></td>
<td>Curie of the Person object representing the individual that created the entity</td>
</tr>
<tr>
<td><a href="../cross_reference_dto/">cross_reference_dto</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../data_provider_dto/">data_provider_dto</a></td>
<td>Ingest object representing the organization (e.g. MOD) from which the data was sourced and a CrossReference to that organisation's site</td>
</tr>
<tr>
<td><a href="../date_created/">date_created</a></td>
<td>The date on which an entity was created. This can be applied to nodes or edges.</td>
</tr>
<tr>
<td><a href="../date_updated/">date_updated</a></td>
<td>Date on which an entity was last modified.</td>
</tr>
<tr>
<td><a href="../db_date_created/">db_date_created</a></td>
<td>The date on which an entity was created in the Alliance database.  This is distinct from date_created, which represents the date when the entity was originally created (i.e. at the MOD for imported data).</td>
</tr>
<tr>
<td><a href="../db_date_updated/">db_date_updated</a></td>
<td>Date on which an entity was last modified in the Alliance database.  This is distinct from date_updated, which represents the date when the entity was last modified and may predate import into the Alliance database.</td>
</tr>
<tr>
<td><a href="../evidence_curie/">evidence_curie</a></td>
<td>Curie of InformationContentEntity object given as evidence</td>
</tr>
<tr>
<td><a href="../inferred_allele_identifier/">inferred_allele_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of allele to which something is inferred to be associated via an automated pipeline</td>
</tr>
<tr>
<td><a href="../inferred_gene_identifier/">inferred_gene_identifier</a></td>
<td>Identifier (curie/primary_external_id/mod_internal_id) of gene to which something is inferred to be associated via an automated pipeline</td>
</tr>
<tr>
<td><a href="../internal/">internal</a></td>
<td>Classifies the entity as private (for internal use) or not (for public use).</td>
</tr>
<tr>
<td><a href="../mod_internal_id/">mod_internal_id</a></td>
<td>The model organism database (MOD) internal identifier for the object</td>
</tr>
<tr>
<td><a href="../negated/">negated</a></td>
<td>if set to true, then the association is negated i.e. is not true</td>
</tr>
<tr>
<td><a href="../note_dtos/">note_dtos</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../obsolete/">obsolete</a></td>
<td>Entity is no longer current.</td>
</tr>
<tr>
<td><a href="../phenotype_statement/">phenotype_statement</a></td>
<td>For MODs that curate to post-composed phenotype expressions composed of ontology terms and other necessary information (e.g. vocabularies, free text). This field captures the free text phenotype statement.</td>
</tr>
<tr>
<td><a href="../phenotype_term_curies/">phenotype_term_curies</a></td>
<td>The string representation of the phenotype ontology term (PhenotypeTerm) curie</td>
</tr>
<tr>
<td><a href="../primary_external_id/">primary_external_id</a></td>
<td>The primary external (non-Alliance) database identifier/curie for the object. Note that this may be an external (non-Alliance member) identifier for an object, like a UniProt ID for a protein, and may act as the MOD's/Alliance member's primary key for the entity.</td>
</tr>
<tr>
<td><a href="../updated_by_curie/">updated_by_curie</a></td>
<td>Curie of the Person object representing the individual that updated the entity</td>
</tr>
</tbody>
</table>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>['alliance:AGMPhenotypeAnnotationDTO']</td>
</tr>
<tr>
<td>native</td>
<td>['alliance:AGMPhenotypeAnnotationDTO']</td>
</tr>
</tbody>
</table>
<h2 id="linkml-specification">LinkML Specification</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: AGMPhenotypeAnnotationDTO
description: Ingest class for an association between an AGM and a phenotype
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: PhenotypeAnnotationDTO
slots:
- agm_identifier
- inferred_gene_identifier
- asserted_gene_identifiers
- inferred_allele_identifier
- asserted_allele_identifiers

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: AGMPhenotypeAnnotationDTO
description: Ingest class for an association between an AGM and a phenotype
from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
is_a: PhenotypeAnnotationDTO
attributes:
  agm_identifier:
    name: agm_identifier
    description: Curie or internal ID of AGM
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: agm_identifier
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AlleleOriginAssociationDTO
    - AGMPhenotypeAnnotationDTO
    - AGMDiseaseAnnotationDTO
    - BioSampleGenomicInformationDTO
    range: string
    required: true
  inferred_gene_identifier:
    name: inferred_gene_identifier
    description: Identifier (curie/primary_external_id/mod_internal_id) of gene to
      which something is inferred to be associated via an automated pipeline
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: inferred_gene_identifier
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AllelePhenotypeAnnotationDTO
    - AGMPhenotypeAnnotationDTO
    - AlleleDiseaseAnnotationDTO
    - AGMDiseaseAnnotationDTO
    range: string
  asserted_gene_identifiers:
    name: asserted_gene_identifiers
    description: Identifiers (curie/primary_external_id/mod_internal_id) of the gene(s)
      to which something is manually asserted to be associated
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: asserted_gene_identifiers
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AllelePhenotypeAnnotationDTO
    - AGMPhenotypeAnnotationDTO
    - AlleleDiseaseAnnotationDTO
    - AGMDiseaseAnnotationDTO
    range: string
    multivalued: true
  inferred_allele_identifier:
    name: inferred_allele_identifier
    description: Identifier (curie/primary_external_id/mod_internal_id) of allele
      to which something is inferred to be associated via an automated pipeline
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: inferred_allele_identifier
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AGMPhenotypeAnnotationDTO
    - AGMDiseaseAnnotationDTO
    range: string
  asserted_allele_identifiers:
    name: asserted_allele_identifiers
    description: Identifier (curie/primary_external_id/mod_internal_id) of the allele
      to which something is manually asserted to be associated
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: asserted_allele_identifiers
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AGMPhenotypeAnnotationDTO
    - AGMDiseaseAnnotationDTO
    range: string
    multivalued: true
  phenotype_statement:
    name: phenotype_statement
    description: For MODs that curate to post-composed phenotype expressions composed
      of ontology terms and other necessary information (e.g. vocabularies, free text).
      This field captures the free text phenotype statement.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: phenotype_statement
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AlleleFunctionalImpactSlotAnnotation
    - AlleleFunctionalImpactSlotAnnotationDTO
    - AlleleInheritanceModeSlotAnnotation
    - AlleleInheritanceModeSlotAnnotationDTO
    - PhenotypeAnnotationDTO
    range: string
  phenotype_term_curies:
    name: phenotype_term_curies
    description: The string representation of the phenotype ontology term (PhenotypeTerm)
      curie
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: phenotype_term_curies
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - PhenotypeAnnotationDTO
    range: string
    multivalued: true
  cross_reference_dto:
    name: cross_reference_dto
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: cross_reference_dto
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - DataProviderDTO
    - PhenotypeAnnotationDTO
    range: CrossReferenceDTO
    multivalued: false
    inlined: true
  negated:
    name: negated
    description: if set to true, then the association is negated i.e. is not true
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: negated
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - ExpressionAnnotation
    - PhenotypeAnnotation
    - PhenotypeAnnotationDTO
    - DiseaseAnnotation
    - DiseaseAnnotationDTO
    range: boolean
  mod_internal_id:
    name: mod_internal_id
    description: The model organism database (MOD) internal identifier for the object
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: mod_internal_id
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
  primary_external_id:
    name: primary_external_id
    description: The primary external (non-Alliance) database identifier/curie for
      the object. Note that this may be an external (non-Alliance member) identifier
      for an object, like a UniProt ID for a protein, and may act as the MOD's/Alliance
      member's primary key for the entity.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: primary_external_id
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - Person
    - SubmittedObject
    - SubmittedObjectDTO
    - Annotation
    - AnnotationDTO
    range: string
  note_dtos:
    name: note_dtos
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: note_dtos
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - SubmittedObjectDTO
    - ConstructComponentSlotAnnotationDTO
    - ConstructGenomicEntityAssociationDTO
    - ConstructCassetteAssociationDTO
    - CassetteComponentSlotAnnotationDTO
    - CassetteGenomicEntityAssociationDTO
    - CassetteTransgenicToolAssociationDTO
    - CassetteStrAssociationDTO
    - AnnotationDTO
    - HTPExpressionDatasetSampleAnnotationDTO
    range: NoteDTO
    multivalued: true
    inlined: true
    inlined_as_list: true
  data_provider_dto:
    name: data_provider_dto
    description: Ingest object representing the organization (e.g. MOD) from which
      the data was sourced and a CrossReference to that organisation's site
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: data_provider_dto
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - SubmittedObjectDTO
    - ChromosomeDTO
    - AssemblyComponentDTO
    - AnnotationDTO
    range: DataProviderDTO
    required: true
    multivalued: false
    inlined: true
  condition_relation_dtos:
    name: condition_relation_dtos
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: condition_relation_dtos
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AnnotationDTO
    range: ConditionRelationDTO
    multivalued: true
    inlined: true
    inlined_as_list: true
  evidence_curie:
    name: evidence_curie
    description: Curie of InformationContentEntity object given as evidence
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: evidence_curie
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - SingleReferenceAssociationDTO
    range: string
  created_by_curie:
    name: created_by_curie
    description: Curie of the Person object representing the individual that created
      the entity
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObjectDTO
    alias: created_by_curie
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObjectDTO
    range: string
  date_created:
    name: date_created
    description: The date on which an entity was created. This can be applied to nodes
      or edges.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - creation_date
    exact_mappings:
    - dct:createdOn
    - WIKIDATA_PROPERTY:P577
    alias: date_created
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  updated_by_curie:
    name: updated_by_curie
    description: Curie of the Person object representing the individual that updated
      the entity
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    domain: AuditedObjectDTO
    alias: updated_by_curie
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObjectDTO
    range: string
  date_updated:
    name: date_updated
    description: Date on which an entity was last modified.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    aliases:
    - date_last_modified
    alias: date_updated
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_created:
    name: db_date_created
    description: The date on which an entity was created in the Alliance database.  This
      is distinct from date_created, which represents the date when the entity was
      originally created (i.e. at the MOD for imported data).
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_created
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  db_date_updated:
    name: db_date_updated
    description: Date on which an entity was last modified in the Alliance database.  This
      is distinct from date_updated, which represents the date when the entity was
      last modified and may predate import into the Alliance database.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: db_date_updated
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: datetime
  internal:
    name: internal
    description: Classifies the entity as private (for internal use) or not (for public
      use).
    notes:
    - Default value is true.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: internal
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean
    required: true
  obsolete:
    name: obsolete
    description: Entity is no longer current.
    notes:
    - Obsolete entities are preserved in the database for posterity but should not
      be publicly displayed.
    from_schema: https://github.com/alliance-genome/agr_curation_schema/alliance_schema
    alias: obsolete
    owner: AGMPhenotypeAnnotationDTO
    domain_of:
    - AuditedObject
    - AuditedObjectDTO
    range: boolean

</code></pre>
</details>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
<script src="../assets/javascripts/bundle.13a4f30d.min.js"></script>
<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js"></script><script>mermaid.initialize({});</script></body>
</html>